<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Andrea Mariossi">
<meta name="dcterms.date" content="2023-10-25">

<title>Analysis.Ciona_TMTproC_normalization</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="00.Ciona_prot_normalization_files/libs/clipboard/clipboard.min.js"></script>
<script src="00.Ciona_prot_normalization_files/libs/quarto-html/quarto.js"></script>
<script src="00.Ciona_prot_normalization_files/libs/quarto-html/popper.min.js"></script>
<script src="00.Ciona_prot_normalization_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="00.Ciona_prot_normalization_files/libs/quarto-html/anchor.min.js"></script>
<link href="00.Ciona_prot_normalization_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="00.Ciona_prot_normalization_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="00.Ciona_prot_normalization_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="00.Ciona_prot_normalization_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="00.Ciona_prot_normalization_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<style>html{ scroll-behavior: smooth; }</style>


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-full toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#tmtproc-ciona" id="toc-tmtproc-ciona" class="nav-link active" data-scroll-target="#tmtproc-ciona"><span class="header-section-number">1</span> TMTproC <em>Ciona</em></a>
  <ul>
  <li><a href="#loading-libraries" id="toc-loading-libraries" class="nav-link" data-scroll-target="#loading-libraries"><span class="header-section-number">1.0.0.1</span> Loading libraries</a></li>
  <li><a href="#functions-and-misc" id="toc-functions-and-misc" class="nav-link" data-scroll-target="#functions-and-misc"><span class="header-section-number">1.0.0.2</span> Functions and misc</a></li>
  <li><a href="#loading-data" id="toc-loading-data" class="nav-link" data-scroll-target="#loading-data"><span class="header-section-number">1.0.0.3</span> Loading data</a></li>
  <li><a href="#standard-normalisation" id="toc-standard-normalisation" class="nav-link" data-scroll-target="#standard-normalisation"><span class="header-section-number">1.0.1</span> Standard Normalisation</a>
  <ul>
  <li><a href="#normalisation-step-using-rowmean-1st-follow-by-columnmedian" id="toc-normalisation-step-using-rowmean-1st-follow-by-columnmedian" class="nav-link" data-scroll-target="#normalisation-step-using-rowmean-1st-follow-by-columnmedian"><span class="header-section-number">1.0.1.0.1</span> Normalisation step using rowMean 1st follow by columnMedian</a></li>
  <li><a href="#assessing-the-normalisation-step" id="toc-assessing-the-normalisation-step" class="nav-link" data-scroll-target="#assessing-the-normalisation-step"><span class="header-section-number">1.0.1.0.2</span> Assessing the normalisation step</a></li>
  </ul></li>
  <li><a href="#improving-the-normalization-process" id="toc-improving-the-normalization-process" class="nav-link" data-scroll-target="#improving-the-normalization-process"><span class="header-section-number">1.0.2</span> Improving the normalization process</a>
  <ul>
  <li><a href="#mitocarta" id="toc-mitocarta" class="nav-link" data-scroll-target="#mitocarta"><span class="header-section-number">1.0.2.1</span> Mitocarta</a>
  <ul>
  <li><a href="#genes-and-protein-annotation" id="toc-genes-and-protein-annotation" class="nav-link" data-scroll-target="#genes-and-protein-annotation"><span class="header-section-number">1.0.2.1.1</span> Genes and protein annotation</a></li>
  <li><a href="#mapping-ncbi-entrez-gene-id-of-mitocarta-to-hgnc-symbols" id="toc-mapping-ncbi-entrez-gene-id-of-mitocarta-to-hgnc-symbols" class="nav-link" data-scroll-target="#mapping-ncbi-entrez-gene-id-of-mitocarta-to-hgnc-symbols"><span class="header-section-number">1.0.2.1.2</span> Mapping NCBI Entrez Gene ID of Mitocarta to HGNC symbols</a></li>
  <li><a href="#visualise-mitocarta-genes" id="toc-visualise-mitocarta-genes" class="nav-link" data-scroll-target="#visualise-mitocarta-genes"><span class="header-section-number">1.0.2.1.3</span> Visualise mitocarta genes</a></li>
  <li><a href="#normalisation-step-using-rowmean-follow-by-flat-df-columnmedian" id="toc-normalisation-step-using-rowmean-follow-by-flat-df-columnmedian" class="nav-link" data-scroll-target="#normalisation-step-using-rowmean-follow-by-flat-df-columnmedian"><span class="header-section-number">1.0.2.1.4</span> Normalisation step using rowMean follow by flat df columnMedian</a></li>
  <li><a href="#assessing-the-mitocarta-normalisation-step" id="toc-assessing-the-mitocarta-normalisation-step" class="nav-link" data-scroll-target="#assessing-the-mitocarta-normalisation-step"><span class="header-section-number">1.0.2.1.5</span> Assessing the mitocarta normalisation step</a></li>
  <li><a href="#same-results-as-above" id="toc-same-results-as-above" class="nav-link" data-scroll-target="#same-results-as-above"><span class="header-section-number">1.0.2.1.6</span> Same results as above</a></li>
  <li><a href="#plotting-mitocarta-normalisation" id="toc-plotting-mitocarta-normalisation" class="nav-link" data-scroll-target="#plotting-mitocarta-normalisation"><span class="header-section-number">1.0.2.1.7</span> Plotting mitocarta normalisation</a></li>
  </ul></li>
  <li><a href="#ribosomes" id="toc-ribosomes" class="nav-link" data-scroll-target="#ribosomes"><span class="header-section-number">1.0.2.2</span> Ribosomes</a></li>
  </ul></li>
  <li><a href="#comparing-normalization-process" id="toc-comparing-normalization-process" class="nav-link" data-scroll-target="#comparing-normalization-process"><span class="header-section-number">1.0.3</span> Comparing normalization process</a></li>
  <li><a href="#relative-distribution" id="toc-relative-distribution" class="nav-link" data-scroll-target="#relative-distribution"><span class="header-section-number">1.0.4</span> Relative distribution</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content column-page-right" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Analysis.Ciona_TMTproC_normalization</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Andrea Mariossi </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 25, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="tmtproc-ciona" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> TMTproC <em>Ciona</em></h1>
<p>We conducted a quantitative complement tandem mass tag (TMTpro) mass spectrometry (MS) analysis (Johnson et al., 2021) of the proteome of <em>Ciona</em> embryos across eight developmental stages, spanning unfertilized eggs, MZT, neurulation, tail elongation, and larval stage. Notably, the use of a multiplexed proteomics method allows for straightforward interpretation of low signal as low protein abundance, presenting a significant advantage over label-free measurements, where the absence of signal may be due to either low abundance or the mass spectrometer’s failure to detect the protein of interest (Pappireddi et al., 2019).</p>
<section id="loading-libraries" class="level4" data-number="1.0.0.1">
<h4 data-number="1.0.0.1" class="anchored" data-anchor-id="loading-libraries"><span class="header-section-number">1.0.0.1</span> Loading libraries</h4>
</section>
<section id="functions-and-misc" class="level4" data-number="1.0.0.2">
<h4 data-number="1.0.0.2" class="anchored" data-anchor-id="functions-and-misc"><span class="header-section-number">1.0.0.2</span> Functions and misc</h4>
<div class="cell" data-layout-align="center" data-paged.print="true">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>(<span class="at">all=</span><span class="cn">TRUE</span>))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="do">#### Directories</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>directory <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">"/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/"</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>directory_save  <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">"/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># directory &lt;- file.choose(). # pick directory where the file is downloaded</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>raw_dir_proteome<span class="ot">=</span><span class="st">"b.raw_data/proteome/"</span> </span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>ciona_stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"unfE"</span>,<span class="st">"fertE"</span>,<span class="st">"cell16"</span>,<span class="st">"iniG"</span>,<span class="st">"latN"</span>,<span class="st">"midTII"</span>,<span class="st">"latTII"</span>,<span class="st">"larva"</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>ciona_stages_beautify <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"unfertilised"</span>,<span class="st">"1-cell"</span>,<span class="st">"16-cell"</span>,<span class="st">"110-cell"</span>,<span class="st">"late neurula"</span>,<span class="st">"mid tailbud"</span>,<span class="st">"late tailbud"</span>,<span class="st">"larva"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="do">#### Themes and colors</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>ciona_stages_palette <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#009392FF"</span>, <span class="st">"#39B185FF"</span>, <span class="st">"#9CCB86FF"</span>, <span class="st">"#E9E29CFF"</span>, <span class="st">"#EEB479FF"</span>, <span class="st">"#E88471FF"</span>, <span class="st">"#CF597EFF"</span>, <span class="st">"#A22E3B"</span>)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>ciona_cluster_palette <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#88A0DCFF"</span>, <span class="st">"#381A61FF"</span>, <span class="st">"#7C4B73FF"</span>, <span class="st">"#ED968CFF"</span>, <span class="st">"#AB3329FF"</span>, <span class="st">"#E78429FF"</span>, <span class="st">"#F9D14AFF"</span>, <span class="st">"#7FA074FF"</span>) <span class="co"># &gt; paletteer::paletteer_d("MetBrewer::Archambault") + MetBrewer::Cassatt2" - Colorblind-Friendly</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="do">#### Functions</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Parallelel plot</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>function_ggparallel <span class="ot">&lt;-</span> <span class="cf">function</span>(data, col_names, group_col, plot_title, <span class="at">alpha_lines =</span> <span class="dv">1</span>, <span class="at">show_points =</span> <span class="cn">TRUE</span>, <span class="at">plot_scale =</span> <span class="st">"globalminmax"</span>, ...) {</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot data</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggparcoord</span>(<span class="at">data =</span> data,</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>                     <span class="at">columns =</span> col_names,</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>                     <span class="at">groupColumn =</span> group_col,</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>                     <span class="at">title =</span> plot_title,</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>                     <span class="at">alphaLines =</span> alpha_lines,</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>                     <span class="at">showPoints =</span> show_points,</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>                     <span class="at">scale =</span> plot_scale,</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>                     ...)</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> plot <span class="sc">+</span> </span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">15</span>)) <span class="co"># Remove legend</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(plot)</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>function_plotGuides <span class="ot">=</span> <span class="fu">guides</span>(</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="st">"none"</span>, </span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">"Proteins number"</span>,</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>                      <span class="at">title.theme =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>,<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">direction =</span> <span class="st">"horizontal"</span>,</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>  <span class="at">title.position =</span> <span class="st">"top"</span>,</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">nrow=</span><span class="dv">1</span>)) </span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="co"># function for cluster plots</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>plot_cluster_norm <span class="ot">&lt;-</span> <span class="cf">function</span>(df, title) {</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span>  </span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>), <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">values_to =</span> <span class="st">"expression"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">str_remove</span>(stage, <span class="st">"_byRowCol"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">fct_relevel</span>(stage, ciona_stages), <span class="at">cluster =</span> <span class="fu">factor</span>(cluster)) <span class="sc">%&gt;%</span> </span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(stage, expression, <span class="at">color =</span> cluster)) <span class="sc">+</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="fu">interaction</span>(cluster, size), <span class="at">size =</span> <span class="fu">as.factor</span>(size))) <span class="sc">+</span></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> ciona_cluster_palette) <span class="sc">+</span></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>, <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> title, <span class="at">x =</span> <span class="st">"Developmental stages"</span>, <span class="at">y=</span><span class="st">"Relative expression"</span>) <span class="sc">+</span></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_size_discrete</span>(<span class="at">name =</span> <span class="st">"Cluster size"</span>) <span class="co"># rename the size legend</span></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a><span class="co"># function for cluster pivot</span></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>cluster_pivot_data <span class="ot">&lt;-</span> <span class="cf">function</span>(data, cluster, cols) {</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>  data <span class="sc">%&gt;%</span></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(.cluster <span class="sc">==</span> cluster) <span class="sc">%&gt;%</span></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(cols)) <span class="sc">%&gt;%</span></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">case_when</span>(name <span class="sc">==</span> <span class="st">"unfE_byRowCol"</span> <span class="sc">~</span> <span class="dv">1</span>, name <span class="sc">==</span> <span class="st">"fertE_byRowCol"</span> <span class="sc">~</span> <span class="dv">2</span>, name <span class="sc">==</span> <span class="st">"cell16_byRowCol"</span> <span class="sc">~</span> <span class="dv">3</span>, name <span class="sc">==</span> <span class="st">"iniG_byRowCol"</span> <span class="sc">~</span> <span class="dv">4</span>, name <span class="sc">==</span> <span class="st">"latN_byRowCol"</span> <span class="sc">~</span> <span class="dv">5</span>, name <span class="sc">==</span> <span class="st">"midTII_byRowCol"</span> <span class="sc">~</span> <span class="dv">6</span>, name <span class="sc">==</span> <span class="st">"latTII_byRowCol"</span> <span class="sc">~</span> <span class="dv">7</span>, name <span class="sc">==</span> <span class="st">"larva_byRowCol"</span> <span class="sc">~</span> <span class="dv">8</span>))</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a><span class="co"># function to linear regression</span></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a>plot_cluster_lr <span class="ot">&lt;-</span> <span class="cf">function</span>(data, title, norm, tag) {</span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> name, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">8</span>) <span class="sc">+</span></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> title,</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a>         <span class="at">subtitle =</span> <span class="fu">paste</span>(</span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>           <span class="st">"Adj R^2 = "</span>, <span class="fu">signif</span>(<span class="fu">summary</span>(<span class="fu">lm</span>(value <span class="sc">~</span> name, <span class="at">data =</span> data))<span class="sc">$</span>adj.r.squared, <span class="dv">5</span>), </span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>           <span class="st">"Intercept = "</span>, <span class="fu">signif</span>(<span class="fu">coef</span>(<span class="fu">lm</span>(value <span class="sc">~</span> name, <span class="at">data =</span> data))[<span class="dv">1</span>], <span class="dv">5</span>),</span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>           <span class="st">"Slope = "</span>, <span class="fu">signif</span>(<span class="fu">coef</span>(<span class="fu">lm</span>(value <span class="sc">~</span> name, <span class="at">data =</span> data))[<span class="dv">2</span>], <span class="dv">5</span>), <span class="at">size =</span> <span class="dv">5</span>),</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>         <span class="at">tag=</span> tag,</span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>         <span class="at">y=</span><span class="st">"Relative expression"</span>,</span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>         <span class="at">x=</span><span class="st">"Stage"</span>) <span class="sc">+</span></span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(norm) { <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">format</span>(x, <span class="at">scientific =</span> <span class="cn">FALSE</span>, <span class="at">big.mark =</span> <span class="st">","</span>))}</span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a><span class="co"># check HGNCmitocarta results with plot: individual proteins</span></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a>plot_HGNCmitocarta_dynamics <span class="ot">&lt;-</span> <span class="cf">function</span>(from, to) {</span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a>  ciona_rowMean_mitocartaHGNC_flatDynamics.df <span class="sc">%&gt;%</span></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename_with</span>(<span class="at">.fn =</span> <span class="sc">~</span><span class="fu">str_replace</span>(., <span class="st">"_by_rowMean$"</span>, <span class="st">""</span>), <span class="at">.cols =</span> <span class="fu">ends_with</span>(<span class="st">"_by_rowMean"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice</span>(from<span class="sc">:</span>to) <span class="sc">%&gt;%</span></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>),</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a>                 <span class="at">names_to =</span> <span class="st">"stage"</span>, </span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>                 <span class="at">values_to =</span> <span class="st">"expression"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">fct_relevel</span>(stage, ciona_stages)) <span class="sc">%&gt;%</span></span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(stage, expression, <span class="at">color =</span> Protein_Id_transcript)) <span class="sc">+</span></span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">group =</span> Protein_Id_transcript)) <span class="sc">+</span></span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> Protein_Id_transcript)) <span class="sc">+</span></span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>Protein_Id_transcript) <span class="sc">+</span></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_pubr</span>(<span class="at">legend =</span> <span class="st">"none"</span>,<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">6</span>)  </span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="loading-data" class="level4" data-number="1.0.0.3">
<h4 data-number="1.0.0.3" class="anchored" data-anchor-id="loading-data"><span class="header-section-number">1.0.0.3</span> Loading data</h4>
<div class="cell" data-layout-align="center" data-paged.print="true">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Loading the data</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_prenormal.df <span class="ot">&lt;-</span> </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>(</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="fu">paste0</span>(directory, raw_dir_proteome, <span class="st">"ky21_w_oxmet_protein_quant_pre.normalisation.tsv"</span>), <span class="at">col_names =</span> T))</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co"># cleaning up the names</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>lookup <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"unfE"</span> <span class="ot">=</span> <span class="st">"TMTPro126"</span>, <span class="st">"fertE"</span> <span class="ot">=</span> <span class="st">"TMTPro128C"</span>, <span class="st">"cell16"</span> <span class="ot">=</span> <span class="st">"TMTPro129N"</span>, <span class="st">"iniG"</span> <span class="ot">=</span> <span class="st">"TMTPro130C"</span>, <span class="st">"latN"</span> <span class="ot">=</span> <span class="st">"TMTPro131N"</span>, <span class="st">"midTII"</span> <span class="ot">=</span> <span class="st">"TMTPro131C"</span>, <span class="st">"latTII"</span> <span class="ot">=</span> <span class="st">"TMTPro133C"</span>, <span class="st">"larva"</span> <span class="ot">=</span> <span class="st">"TMTPro134N"</span>)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_prenormal.df <span class="ot">&lt;-</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_prenormal.df <span class="sc">%&gt;%</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="fu">all_of</span>(lookup)) <span class="sc">%&gt;%</span> <span class="co"># Rename the column</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"_contaminant"</span>, Protein_Id, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> <span class="co"># exclude human/pig/TEV contaminant, 42 entries</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"##"</span>, Protein_Id, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> <span class="co"># exclude scramble input for FDR correction</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"Gene_Symbol"</span>, <span class="st">"Description"</span>, <span class="st">"Group_ID"</span>, <span class="st">"Collapsed"</span>, <span class="st">"Parsimony"</span>, <span class="st">"default~ratios0 sum"</span>, <span class="st">"default~ratios9 sum"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Protein_Id_Gene =</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.v</span><span class="sc">\\</span><span class="st">S*</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="at">replacement =</span> <span class="st">""</span>, Protein_Id)) <span class="sc">%&gt;%</span> <span class="co"># transcript to gene ID</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Protein_Id_transcript =</span> Protein_Id) <span class="sc">%&gt;%</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Protein_Id_transcript) <span class="sc">%&gt;%</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(Protein_Id_transcript, Protein_Id_Gene, <span class="fu">everything</span>())</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(ciona_TMTproC_prenormal.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 7,095
Columns: 11
$ Protein_Id_transcript &lt;chr&gt; "KY21.Chr1.10.v1.nonSL2-1", "KY21.Chr1.1000.v1.SL1-1", "KY21.Chr1.1001.v1.SL1-1", "KY21.Chr1.1002.v1.SL1-1", "KY21.Chr1.1004.v1.nonSL4-1", "KY21.Chr1.1005.v1.ND1-1", "KY21.Chr1.1006.v1.SL1-1", "KY21.Chr1.1009.v1.ND1-1", "KY21.Chr1.101.v1.SL1-1", "KY21.Chr1.1010.v1.nonSL1-1", "KY21.Chr1.1011.v1.SL1-1", "KY21.Chr1.1012.v1.SL1-1", "KY21.Chr1.1013.v1.ND1-1", "KY21.Chr1.1014.v2.ND2-1", "KY21.Chr1.1015.v1.SL1-1", "KY21.Chr1.1016.v1.SL1-1", "KY21.Chr1.1018.v1.nonSL1-1", "KY21.Chr1.1019.v1.SL1-1", "KY21.Chr1.102.v1.nonSL2-1", "KY21.Chr1.1020.v1.SL1-1", "KY21.Chr1.1022.v1.ND1-1", "KY21.Chr1.1024.v1.SL1-1", "KY21.Chr1.1027.v2.nonSL3-1", "KY21.Chr1.1029.v1.SL1-1", "KY21.Chr1.1031.v1.ND1-1", "KY21.Chr1.1035.v1.SL1-1", "KY21.Chr1.1040.v1.nonSL7-1", "KY21.Chr1.1041.v1.nonSL2-1", "KY21.Chr1.1042.v1.nonSL4-1", "KY21.Chr1.1045.v1.SL1-1", "KY21.Chr1.1046.v1.SL1-1", "KY21.Chr1.1050.v1.SL1-1", "KY21.Chr1.1051.v1.SL1-1", "KY21.Chr1.1055.v1.SL1-1", "KY21.Chr1.1056.v1.nonSL3-1…
$ Protein_Id_Gene       &lt;chr&gt; "KY21.Chr1.10", "KY21.Chr1.1000", "KY21.Chr1.1001", "KY21.Chr1.1002", "KY21.Chr1.1004", "KY21.Chr1.1005", "KY21.Chr1.1006", "KY21.Chr1.1009", "KY21.Chr1.101", "KY21.Chr1.1010", "KY21.Chr1.1011", "KY21.Chr1.1012", "KY21.Chr1.1013", "KY21.Chr1.1014", "KY21.Chr1.1015", "KY21.Chr1.1016", "KY21.Chr1.1018", "KY21.Chr1.1019", "KY21.Chr1.102", "KY21.Chr1.1020", "KY21.Chr1.1022", "KY21.Chr1.1024", "KY21.Chr1.1027", "KY21.Chr1.1029", "KY21.Chr1.1031", "KY21.Chr1.1035", "KY21.Chr1.1040", "KY21.Chr1.1041", "KY21.Chr1.1042", "KY21.Chr1.1045", "KY21.Chr1.1046", "KY21.Chr1.1050", "KY21.Chr1.1051", "KY21.Chr1.1055", "KY21.Chr1.1056", "KY21.Chr1.1057", "KY21.Chr1.1058", "KY21.Chr1.1060", "KY21.Chr1.1061", "KY21.Chr1.1062", "KY21.Chr1.1063", "KY21.Chr1.1066", "KY21.Chr1.107", "KY21.Chr1.1072", "KY21.Chr1.1073", "KY21.Chr1.1077", "KY21.Chr1.1078", "KY21.Chr1.108", "KY21.Chr1.1085", "KY21.Chr1.1088", "KY21.Chr1.1089", "KY21.Chr1.109", "KY21.Chr1.1093", "KY21.Chr1.1095", "KY…
$ Number_of_peptides    &lt;dbl&gt; 5, 13, 15, 6, 21, 9, 1, 12, 9, 1, 2, 24, 5, 25, 1, 11, 3, 11, 2, 2, 2, 10, 7, 2, 3, 8, 2, 1, 8, 4, 8, 17, 6, 14, 3, 13, 1, 4, 23, 3, 5, 13, 9, 34, 7, 11, 1, 2, 2, 17, 12, 1, 12, 2, 39, 5, 6, 2, 4, 5, 5, 4, 7, 4, 30, 8, 2, 17, 7, 142, 23, 1, 1, 7, 6, 14, 15, 5, 5, 4, 13, 9, 4, 7, 10, 3, 17, 10, 10, 1, 5, 1, 6, 6, 54, 4, 14, 32, 5, 19, 3, 1, 8, 10, 1, 5, 10, 4, 2, 2, 25, 2, 5, 64, 3, 5, 7, 10, 1, 1, 2, 5, 9, 26, 6, 22, 5, 8, 2, 1, 4, 27, 1, 1, 33, 19, 1, 13, 11, 1, 4, 4, 2, 15, 2, 20, 8, 24, 3, 2, 3, 4, 24, 15, 5, 3, 1, 1, 6, 1, 3, 24, 19, 9, 1, 26, 7, 8, 5, 1, 42, 11, 1, 13, 2, 4, 29, 22, 3, 7, 6, 2, 1, 13, 19, 3, 28, 8, 1, 4, 15, 9, 3, 25, 11, 7, 4, 14, 2, 9, 6, 7, 8, 10, 3, 5, 6, 1, 6, 1, 3, 1, 20, 16, 28, 4, 2, 3, 11, 1, 18, 2, 2, 3, 6, 6, 9, 2, 10, 8, 4, 2, 1, 1, 4, 4, 3, 7, 1, 5, 24, 1, 5, 2, 23, 3, 2, 15, 18, 17, 3, 15, 2, 1, 18, 2, 4, 7, 4, 13, 9, 10, 1, 21, 2, 15, 8, 2, 2, 10, 8, 3, 4, 29, 10, 4, 10, 38, 3, 3, 4, 3, 12, 12, 4, 53, 15, 3, 4, 1, 8, …
$ unfE                  &lt;dbl&gt; 6.79550e-01, 1.80478e+00, 1.77061e+00, 2.66736e+00, 2.49546e+00, 1.18459e+00, 2.12080e-01, 2.15231e+00, 9.83251e-01, 2.93230e-01, 1.81664e-01, 2.67143e+00, 6.81070e-01, 3.21492e+00, 1.40200e-01, 1.19374e+00, 3.10652e-01, 1.16788e+00, 8.01760e-02, 3.84460e-01, 1.16219e-01, 1.41303e+00, 7.58105e-01, 2.03191e-01, 3.69990e-01, 6.47148e-01, 1.62202e-01, 2.97860e-01, 1.16880e+00, 3.64094e+00, 4.94889e-01, 2.19138e+00, 5.69127e-01, 1.96017e+00, 4.12340e-01, 2.42170e+00, 1.33990e-01, 3.06282e-01, 2.02815e+00, 3.20270e-01, 8.49990e-02, 1.45603e+00, 9.73567e-01, 4.39201e+00, 3.04738e-01, 1.29174e+00, 8.45240e-01, 9.60280e-02, 2.95710e-01, 1.86714e+00, 1.45689e+00, 8.31570e-03, 1.00654e+00, 1.60516e-01, 4.50932e+00, 2.96806e-01, 7.19776e-01, 2.56220e-01, 1.56429e-01, 6.51930e-01, 6.41110e-01, 5.46610e-01, 8.60690e-01, 2.85782e-01, 3.97738e+00, 8.18631e-01, 4.01320e-01, 2.18791e+00, 6.97508e-01, 1.92738e+01, 5.85748e-01, 1.95630e-01, 2.70560e-02, 2.27420e+00, 6.7773…
$ fertE                 &lt;dbl&gt; 0.57898000, 1.61365000, 1.47541000, 2.22937000, 2.12542000, 1.04231000, 0.16163000, 1.79585000, 0.83099500, 0.28886000, 0.15184300, 2.39866000, 0.51786800, 2.79024000, 0.11953000, 1.07551000, 0.34142300, 0.99176000, 0.06759600, 0.24660000, 0.10924900, 1.32116000, 0.65158100, 0.15684400, 0.29204400, 0.51322100, 0.15263800, 0.07427600, 0.95676400, 0.22614100, 0.54806600, 1.91668000, 0.33684600, 1.65419000, 0.33958000, 2.13430000, 0.10245000, 0.26434300, 1.88775000, 0.29613000, 0.06304520, 1.34814000, 0.89273200, 3.75227000, 0.26859700, 1.10863000, 0.05636300, 0.10972500, 0.22016300, 1.75429000, 1.14763000, 0.14042000, 0.80252200, 0.14261900, 3.81532000, 0.24705600, 0.64934400, 0.22233000, 0.28445900, 0.57858800, 0.58369100, 0.47278000, 0.77612000, 0.30220000, 3.58440000, 0.76841400, 0.20288900, 1.84208000, 0.81967000, 16.18490000, 0.62608200, 0.58697000, 0.10912000, 0.77289200, 0.58463500, 1.66089000, 1.53299000, 0.51176500, 0.37952900, 0.43241300, 1.30337…
$ cell16                &lt;dbl&gt; 6.30810e-01, 1.74379e+00, 1.84235e+00, 3.92345e-01, 2.61606e+00, 1.54636e+00, 2.81570e-02, 2.01416e+00, 9.19726e-01, 4.99240e-02, 1.96029e-01, 2.64079e+00, 5.92140e-01, 3.23076e+00, 1.21340e-01, 1.34959e+00, 4.17450e-01, 1.27442e+00, 1.59875e-01, 2.46230e-01, 2.00173e-01, 1.22588e+00, 8.85627e-01, 2.39350e-01, 3.57280e-01, 8.26071e-01, 1.94077e-01, 9.31880e-02, 1.02497e+00, 4.03112e-02, 1.14933e+00, 2.36967e+00, 6.11220e-01, 1.99817e+00, 4.15720e-01, 1.85064e+00, 1.27860e-01, 3.48907e-01, 2.67077e+00, 3.12358e-01, 5.07407e-02, 1.58694e+00, 1.20904e+00, 4.44341e+00, 3.90184e-01, 1.31081e+00, 4.67240e-04, 1.69883e-01, 2.13952e-01, 2.15117e+00, 1.54993e+00, 5.47300e-02, 1.12241e+00, 1.93090e-01, 4.62140e+00, 3.23287e-01, 7.08080e-01, 2.98280e-01, 2.00552e-01, 6.54220e-01, 6.74670e-01, 6.17570e-01, 9.20190e-01, 4.96919e-01, 3.99285e+00, 9.98534e-01, 4.39920e-01, 2.23781e+00, 8.37260e-01, 1.84059e+01, 1.26432e+00, 7.63360e-02, 7.70780e-02, 1.11694e+00, 7.0421…
$ iniG                  &lt;dbl&gt; 5.66530e-01, 1.40743e+00, 1.66590e+00, 8.85543e-02, 2.41560e+00, 1.25660e+00, 2.28600e-02, 1.09708e+00, 1.04635e+00, 4.16140e-02, 2.22320e-01, 2.66642e+00, 4.98090e-01, 3.00702e+00, 1.45100e-01, 1.23986e+00, 3.92330e-01, 1.35977e+00, 1.82828e-01, 1.73323e-01, 3.41980e-01, 6.05508e-01, 7.58633e-01, 2.23890e-01, 2.58228e-01, 9.31983e-01, 2.46000e-01, 4.29070e-02, 8.93905e-01, 3.44082e-03, 9.58860e-01, 1.84328e+00, 5.95189e-01, 1.82242e+00, 3.33750e-01, 1.41622e+00, 1.31940e-01, 4.51420e-01, 2.26438e+00, 3.32360e-01, 4.50053e-02, 1.68475e+00, 1.10516e+00, 3.85193e+00, 5.11094e-01, 1.07302e+00, 5.28620e-04, 2.51760e-01, 1.86955e-01, 2.08841e+00, 1.48001e+00, 1.38720e-01, 1.00068e+00, 2.70140e-01, 4.35132e+00, 8.75432e-02, 6.27003e-01, 2.27230e-01, 1.79769e-01, 5.23063e-01, 5.63302e-01, 6.11400e-01, 7.82620e-01, 6.78410e-01, 3.64809e+00, 1.19590e+00, 1.70195e-01, 1.59778e+00, 7.69533e-01, 1.61785e+01, 1.57272e+00, 6.54920e-02, 1.65220e-01, 1.15200e+00, 6.7515…
$ latN                  &lt;dbl&gt; 6.56020e-01, 1.60770e+00, 1.92901e+00, 1.48615e-01, 2.84228e+00, 1.15077e+00, 1.41600e-01, 1.24657e+00, 1.23647e+00, 5.97930e-02, 2.87670e-01, 3.11005e+00, 6.75230e-01, 3.52794e+00, 1.33690e-01, 1.46253e+00, 3.92100e-01, 1.55889e+00, 2.93490e-01, 2.01361e-01, 4.45370e-01, 8.74752e-01, 8.09143e-01, 2.62910e-01, 4.10610e-01, 1.09934e+00, 2.50590e-01, 1.83880e-02, 1.02141e+00, 2.00058e-03, 1.26086e+00, 2.12950e+00, 7.99010e-01, 2.13169e+00, 4.19640e-01, 1.45474e+00, 1.41710e-01, 5.24710e-01, 3.08728e+00, 4.23560e-01, 1.46835e+00, 1.96051e+00, 1.19747e+00, 4.43640e+00, 1.06858e+00, 1.43388e+00, 5.32300e-04, 3.27810e-01, 1.89183e-01, 2.30298e+00, 1.59574e+00, 7.38420e-02, 1.53803e+00, 3.23490e-01, 5.04323e+00, 1.04762e-01, 8.44640e-01, 2.39760e-01, 1.57163e-01, 6.84430e-01, 6.94280e-01, 6.07570e-01, 8.23100e-01, 6.68700e-01, 4.15992e+00, 1.31243e+00, 1.50844e-01, 1.99327e+00, 8.42364e-01, 1.75978e+01, 2.85044e+00, 5.03780e-02, 1.42170e-01, 9.48280e-01, 8.2858…
$ midTII                &lt;dbl&gt; 0.51649100, 1.47614000, 1.86640000, 0.14455300, 2.73108000, 1.00493000, 0.13914000, 1.19404000, 1.15203000, 0.05498100, 0.26129000, 3.13778000, 0.61660000, 3.08246000, 0.11558000, 1.39207000, 0.38579000, 1.39137000, 0.35927000, 0.27412000, 0.30976000, 1.06056000, 0.82864000, 0.28930000, 0.39583000, 1.10481000, 0.23575000, 0.06784000, 0.88612100, 0.00320937, 1.12370000, 1.84402000, 0.79298300, 1.58788000, 0.33307000, 1.19009000, 0.12321000, 0.59921000, 3.08591000, 0.38369000, 1.07596000, 1.66018000, 1.16994000, 4.11350000, 1.30775000, 1.49254000, 0.00027092, 0.31506000, 0.21978600, 2.24189000, 1.41341000, 0.10162000, 1.78241000, 0.34303000, 5.25463000, 0.33050300, 0.75703000, 0.20418100, 0.29036500, 0.60226000, 0.58384600, 0.43159000, 0.67521200, 0.61858000, 3.64306000, 1.22484000, 0.19632100, 1.80701000, 0.90437000, 16.52180000, 4.15660000, 0.02449100, 0.14110000, 0.37511500, 0.77145000, 1.70249000, 1.65501000, 0.58259000, 0.79794000, 0.46438000, 1.61223…
$ latTII                &lt;dbl&gt; 0.71847000, 1.81019000, 2.29295000, 0.19764400, 2.99856000, 0.95946700, 0.15448000, 1.39381000, 1.50655000, 0.13078000, 0.36377000, 3.98064000, 0.73932000, 3.19899000, 0.11578000, 1.73527000, 0.39115000, 1.74421000, 0.40071000, 0.24575000, 0.28959000, 1.66706000, 1.11664000, 0.32713000, 0.45108000, 1.43025000, 0.41443000, 0.13884000, 1.06728000, 0.00861615, 1.43945000, 2.44021000, 0.94058300, 1.60715000, 0.38654000, 1.46237000, 0.11509000, 0.70846000, 4.22869000, 0.49455000, 1.10235000, 1.72177000, 1.28422000, 4.71681000, 2.00193000, 1.79324000, 0.04356600, 0.38657000, 0.34050000, 2.29164000, 1.71192000, 0.24534000, 2.34237000, 0.31987000, 5.60966000, 1.32448000, 0.86305000, 0.26963000, 1.01290000, 0.65483000, 0.65851000, 0.40867400, 1.10525000, 0.51764000, 3.90885000, 1.04650000, 0.26919000, 2.76587000, 1.04617000, 20.00400000, 5.74928000, 0.00034559, 0.19598000, 0.22023100, 0.89548000, 1.21065000, 2.35680000, 0.76510000, 0.65684500, 0.51234000, 1.74198…
$ larva                 &lt;dbl&gt; 0.65318000, 1.53632000, 2.15739000, 0.13155000, 2.77555000, 0.85500400, 0.14004000, 1.10616000, 1.32461000, 0.08081900, 0.33542000, 3.39425000, 0.67965500, 2.94764000, 0.10878000, 1.55143000, 0.36908900, 1.51174000, 0.45606000, 0.22818000, 0.18767700, 1.83204000, 1.19164000, 0.29738000, 0.46495000, 1.44716000, 0.34432000, 0.26671000, 0.98072000, 0.07533700, 1.02489000, 2.26527000, 1.35504000, 1.23834000, 0.35935000, 1.06997000, 0.12374000, 0.79668000, 3.74706000, 0.43708000, 1.10956000, 1.58167000, 1.16787000, 4.29374000, 1.14713000, 1.49614000, 0.05302900, 0.34318000, 0.33375000, 2.30244000, 1.64451000, 0.23702000, 2.40503000, 0.24724000, 5.79510000, 2.28556000, 0.83109000, 0.28237000, 1.71836000, 0.65069000, 0.60056000, 0.30379800, 1.05683000, 0.43174600, 3.08550000, 0.63474200, 0.16932400, 2.56829000, 1.08313000, 17.83330000, 6.19484000, 0.00035054, 0.14228000, 0.14032400, 0.86275000, 0.81396900, 1.99099000, 0.64302500, 0.61435000, 0.43833100, 1.72909…</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ciona_Number_of_peptides.df &lt;- ciona_TMTproC_prenormal.df[,1:3]</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">vis_dat</span>(ciona_TMTproC_prenormal.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Loading%20data-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vis_miss</span>(ciona_TMTproC_prenormal.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Loading%20data-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">miss_case_summary</span>(ciona_TMTproC_prenormal.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 7,095 × 3
    case n_miss pct_miss
   &lt;int&gt;  &lt;int&gt;    &lt;dbl&gt;
 1     1      0        0
 2     2      0        0
 3     3      0        0
 4     4      0        0
 5     5      0        0
 6     6      0        0
 7     7      0        0
 8     8      0        0
 9     9      0        0
10    10      0        0
# ℹ 7,085 more rows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 7095 protein entries</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="standard-normalisation" class="level3" data-number="1.0.1">
<h3 data-number="1.0.1" class="anchored" data-anchor-id="standard-normalisation"><span class="header-section-number">1.0.1</span> Standard Normalisation</h3>
<section id="normalisation-step-using-rowmean-1st-follow-by-columnmedian" class="level5" data-number="1.0.1.0.1">
<h5 data-number="1.0.1.0.1" class="anchored" data-anchor-id="normalisation-step-using-rowmean-1st-follow-by-columnmedian"><span class="header-section-number">1.0.1.0.1</span> Normalisation step using rowMean 1st follow by columnMedian</h5>
<div class="cell" data-layout-align="center" data-paged.print="true">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Normalise each row by its average and then by the median columns</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_rowMean.df <span class="ot">&lt;-</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_prenormal.df <span class="sc">%&gt;%</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> <span class="co"># row-wise operations</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">row_mean =</span> <span class="fu">rowMeans</span>(<span class="fu">pick</span>(<span class="fu">c</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>)))) <span class="sc">%&gt;%</span> <span class="co"># calculate mean by row</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>), <span class="sc">~</span> . <span class="sc">/</span> row_mean)) <span class="sc">%&gt;%</span> <span class="co"># normalise by dividing each row by its mean</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>row_mean) <span class="sc">%&gt;%</span> <span class="co"># remove the row_mean column</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_at</span>(<span class="fu">c</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>), <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_by_rowMean"</span>)) <span class="co"># rename columns with _by_rowMean suffix</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_colMedian_normVector <span class="ot">&lt;-</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMean.df <span class="sc">%&gt;%</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="at">.cols =</span> <span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>, <span class="at">.fns =</span> median)) <span class="co"># summarize columns median</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="co"># # Double checking results</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co"># col_medians &lt;- apply(ciona_TMTproC_rowMean.df[,-c(1:3)], 2, median)</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="co"># col_medians_tbl &lt;- as_tibble(t(col_medians))</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="co"># normalised by column median and cleaning up the names</span></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_rowMean_colMedian.df <span class="ot">&lt;-</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMean.df <span class="sc">%&gt;%</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2_dfc</span>(ciona_TMTproC_colMedian_normVector, <span class="st">`</span><span class="at">/</span><span class="st">`</span>) <span class="sc">%&gt;%</span> <span class="co"># Divide each column of the selected df by the corresponding value in the colMedian_norm_.vector_flat vector</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">.fn =</span> <span class="sc">~</span> <span class="fu">str_replace</span>(., <span class="st">"_by_rowMean$"</span>, <span class="st">"_byRowCol"</span>),</span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">.cols =</span> <span class="fu">ends_with</span>(<span class="st">"_by_rowMean"</span>)</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(ciona_TMTproC_rowMean.df[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], ., <span class="at">.name_repair =</span> <span class="st">"minimal"</span>)</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(ciona_TMTproC_rowMean_colMedian.df, "normalized_counts_notGood_TMTproC_ciona.txt", quote=F, col.names=T, row.names=F, sep="\t")</span></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting raw data</span></span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>plot_original_ciona <span class="ot">&lt;-</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_prenormal.df <span class="sc">%&gt;%</span></span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>), <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> as.factor) <span class="sc">%&gt;%</span></span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> <span class="fu">rev</span>(ciona_stages)), <span class="at">x =</span> value, <span class="at">fill =</span> stage)) <span class="sc">+</span></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">setNames</span>(ciona_stages_palette, ciona_stages), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">setNames</span>(ciona_stages_palette, ciona_stages), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Raw data"</span>, <span class="at">y =</span> <span class="st">"Developmental stages"</span>, <span class="at">x =</span> <span class="st">" Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting normalised data</span></span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>plot_normRowCol_ciona <span class="ot">&lt;-</span></span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMean_colMedian.df <span class="sc">%&gt;%</span></span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>), <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> as.factor) <span class="sc">%&gt;%</span></span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(stage, <span class="st">"_byRowCol"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">str_remove</span>(stage, <span class="st">"_byRowCol"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> <span class="fu">rev</span>(ciona_stages)), <span class="at">x =</span> value, <span class="at">fill =</span> stage)) <span class="sc">+</span></span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">setNames</span>(ciona_stages_palette, ciona_stages), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">setNames</span>(ciona_stages_palette, ciona_stages), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Normalised not good"</span>, <span class="at">y =</span> <span class="st">"Developmental stages"</span>, <span class="at">x =</span> <span class="st">" Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-60"><a href="#cb9-60" aria-hidden="true" tabindex="-1"></a>plot_original_ciona <span class="sc">+</span> plot_normRowCol_ciona</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Normalisation%20step%20using%20rowMean%201st%20follow%20by%20columnMedian-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="assessing-the-normalisation-step" class="level5" data-number="1.0.1.0.2">
<h5 data-number="1.0.1.0.2" class="anchored" data-anchor-id="assessing-the-normalisation-step"><span class="header-section-number">1.0.1.0.2</span> Assessing the normalisation step</h5>
<p>CLuster 6 contains protein related to ribosomes, mitochondria, and housekeeping genes. This cluster’s expression decreases over time. Try a different normalization method to emove this technical variation and enhance the biological signals.</p>
<div class="cell" data-layout-align="center" data-paged.print="true">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Evaluating the normalization process through k-means clustering.</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">777</span>) <span class="co"># reproduce the clusters</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit and predict clusters with k = 8</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>ciona_kmeans <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(ciona_TMTproC_rowMean_colMedian.df[, <span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>], <span class="at">centers =</span> <span class="dv">8</span>, <span class="at">nstart =</span> <span class="dv">100</span>, <span class="at">iter.max =</span> <span class="dv">1000</span>)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="fu">glance</span>(ciona_kmeans) <span class="co"># extracts a single-row summary: totss, tot.withinss of clustering</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 4
   totss tot.withinss betweenss  iter
   &lt;dbl&gt;        &lt;dbl&gt;     &lt;dbl&gt; &lt;int&gt;
1 13307.        3127.    10179.     6</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># clean up the clustering and add cluster prediction to the original data set</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>ciona_results <span class="ot">&lt;-</span> <span class="fu">augment</span>(ciona_kmeans, ciona_TMTproC_rowMean_colMedian.df)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>ciona_results <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 12
  Protein_Id_transcript      Protein_Id_Gene Number_of_peptides unfE_byRowCol fertE_byRowCol cell16_byRowCol iniG_byRowCol latN_byRowCol midTII_byRowCol latTII_byRowCol larva_byRowCol .cluster
  &lt;chr&gt;                      &lt;chr&gt;                        &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;           &lt;dbl&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;          &lt;dbl&gt; &lt;fct&gt;   
1 KY21.Chr1.10.v1.nonSL2-1   KY21.Chr1.10                     5          1.21           1.21           1.07          1.02          0.999           0.838           0.942          0.935 6       
2 KY21.Chr1.1000.v1.SL1-1    KY21.Chr1.1000                  13          1.23           1.29           1.14          0.978         0.942           0.921           0.913          0.845 6       
3 KY21.Chr1.1001.v1.SL1-1    KY21.Chr1.1001                  15          1.05           1.03           1.04          1.00          0.980           1.01            1.00           1.03  4       
4 KY21.Chr1.1002.v1.SL1-1    KY21.Chr1.1002                   6          3.95           3.87           0.554         0.133         0.189           0.195           0.216          0.157 2       
5 KY21.Chr1.1004.v1.nonSL4-1 KY21.Chr1.1004                  21          1.06           1.06           1.06          1.04          1.03            1.05            0.937          0.946 4       </code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># summarizes  per-cluster level info eg how many point each cluster and within cluster sum of squares</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>ciona_cluster_avgSummary <span class="ot">&lt;-</span> <span class="fu">tidy</span>(ciona_kmeans)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>ciona_cluster_avgSummary</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 11
  unfE_byRowCol fertE_byRowCol cell16_byRowCol iniG_byRowCol latN_byRowCol midTII_byRowCol latTII_byRowCol larva_byRowCol  size withinss cluster
          &lt;dbl&gt;          &lt;dbl&gt;           &lt;dbl&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;          &lt;dbl&gt; &lt;int&gt;    &lt;dbl&gt; &lt;fct&gt;  
1         0.446         0.462            0.598        0.767         1.06            1.34             1.46           1.60    822    518.  1      
2         2.52          2.19             1.26         0.597         0.520           0.490            0.526          0.589   201    484.  2      
3         0.534         0.612            0.915        1.45          1.47            1.28             0.928          0.861   405    378.  3      
4         0.929         0.939            0.952        1.00          1.02            1.04             1.08           1.11   2429    419.  4      
5         7.30          0.604            0.104        0.0796        0.0713          0.0965           0.156          0.390    37     37.5 5      
6         1.22          1.23             1.17         1.05          0.995           0.928            0.861          0.824  2602    633.  6      
7         0.126         0.0906           0.114        0.0852        0.112           0.455            1.81           4.35    191    218.  7      
8         0.260         0.191            0.203        0.189         0.459           1.21             2.15           2.64    408    439.  8      </code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualise dynamics x cluster. Eg cluster 1 should be flatter</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">function_ggparallel</span>(</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> ciona_cluster_avgSummary <span class="sc">%&gt;%</span> <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">str_remove</span>(., <span class="st">"_byRowCol"</span>)),</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">col_names =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">group_col =</span> <span class="st">"cluster"</span>,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot_title =</span> <span class="st">"Ciona: cluster center dynamcis"</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color =</span> cluster, <span class="at">size =</span> <span class="fu">as.factor</span>(size))) <span class="sc">+</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> ciona_cluster_palette) <span class="sc">+</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>cluster) <span class="sc">+</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Visualise each protein behavior x cluster</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">function_ggparallel</span>(</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> ciona_results <span class="sc">%&gt;%</span> <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">str_remove</span>(., <span class="st">"_byRowCol"</span>)),</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">col_names =</span> <span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>,</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_col =</span> <span class="st">".cluster"</span>,</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot_title =</span> <span class="st">"Ciona: proteins dynamics x cluster"</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_line(aes(color=cluster, size=as.factor(size))) +</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> ciona_cluster_palette) <span class="sc">+</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>.cluster) <span class="sc">+</span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="dv">0</span>, <span class="fl">0.5</span>), <span class="st">"lines"</span>), <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Assessing_the_normalisation_step-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ciona_results %&gt;% rename_with(~ str_remove(., "_byRowCol")) %&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   pivot_longer(cols = c(4:11), names_to = "stage", values_to = "expression") %&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(stage = fct_relevel(stage, ciona_stages)) %&gt;%</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   ggplot(aes(stage,expression, color = .cluster)) +</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_point(aes(group = Protein_Id_transcript)) +</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_line(aes(group = Protein_Id_transcript)) +</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_color_manual(values = ciona_cluster_palette) +</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   facet_wrap(~.cluster) + theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust = 1))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="improving-the-normalization-process" class="level3" data-number="1.0.2">
<h3 data-number="1.0.2" class="anchored" data-anchor-id="improving-the-normalization-process"><span class="header-section-number">1.0.2</span> Improving the normalization process</h3>
<p>It seems that dividing the protein values by the median column values is not an effective method for normalizing this data because the “flat” proteins are decreasing over time. As an alternative, we try using ribosomal and/or mitochondrial proteins as internal standards to improve the normalization process and better capture the stable proteins.</p>
<section id="mitocarta" class="level4" data-number="1.0.2.1">
<h4 data-number="1.0.2.1" class="anchored" data-anchor-id="mitocarta"><span class="header-section-number">1.0.2.1</span> Mitocarta</h4>
<p>The mitochondrial protein annotation provided by the Mitocarta database (https://www.broadinstitute.org/mitocarta/mitocarta30-inventory-mammalian-mitochondrial-proteins-and-pathways) includes only Entrez IDs. However, the Ciona gff3 annotation utilizes HGCN IDs. To map between these two annotations, we first use biomart and then select only the mitochondrial proteins present in our TMTcPro dataframe.</p>
<section id="genes-and-protein-annotation" class="level5" data-number="1.0.2.1.1">
<h5 data-number="1.0.2.1.1" class="anchored" data-anchor-id="genes-and-protein-annotation"><span class="header-section-number">1.0.2.1.1</span> Genes and protein annotation</h5>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Annotating by gff3</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>ciona_annotation_gff3 <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(<span class="fu">read.delim</span>(<span class="at">file =</span> <span class="fu">paste0</span>(directory, <span class="st">"a.reference_annotation/annotation_ciona.KY21ensembl.109.MT.tsv"</span>), <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">dec =</span> <span class="st">"."</span>, <span class="at">strip.white =</span> T)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">10</span>, <span class="dv">11</span>, <span class="dv">12</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(ciona_annotation_gff3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 55,205
Columns: 5
$ GeneID         &lt;chr&gt; "KY21.Chr13.334", "KY21.Chr11.1168", "KY21.Chr11.1170", "KY21.Chr9.1164", "KY21.Chr7.558", "KY21.Chr11.343", "KY21.Chr12.128", "KY21.Chr2.200", "KY21.Chr3.56", "KY21.Chr1.156", "KY21.Chr1.157", "KY21.Chr1.157", "KY21.Chr1.157", "KY21.Chr1.157", "KY21.Chr1.157", "KY21.Chr1.157", "KY21.Chr1.157", "KY21.Chr1.157", "KY21.Chr9.577", "KY21.Chr9.577", "KY21.Chr12.699", "KY21.Chr12.699", "KY21.Chr9.398", "KY21.UAContig13.6", "KY21.UAContig13.6", "KY21.UAContig13.6", "KY21.UAContig13.6", "KY21.UAContig13.6", "KY21.UAContig32.28", "KY21.UAContig36.22", "KY21.UAContig49.9", "KY21.Chr8.497", "KY21.Chr7.192", "KY21.Chr9.981", "KY21.Chr9.1132", "KY21.Chr13.101", "KY21.Chr7.880", "KY21.Chr10.26", "KY21.Chr1.1883", "KY21.Chr2.1022", "KY21.Chr2.1023", "KY21.Chr2.1023", "KY21.Chr2.1023", "KY21.Chr2.1023", "KY21.Chr2.1023", "KY21.Chr3.114", "KY21.Chr3.115", "KY21.Chr3.115", "KY21.Chr14.562", "KY21.Chr14.562", "KY21.UAContig2.22", "KY21.Chr1.2027", "KY21.Chr5.488", "KY21.Chr7.427"…
$ TranscriptID   &lt;chr&gt; "KY21.Chr13.334.v1.ND1-1", "KY21.Chr11.1168.v2.SL1-1", "KY21.Chr11.1170.v1.SL1-1", "KY21.Chr9.1164.v1.ND1-1", "KY21.Chr7.558.v1.ND1-1", "KY21.Chr11.343.v1.nonSL1-1", "KY21.Chr12.128.v1.ND1-1", "KY21.Chr2.200.v1.SL1-1", "KY21.Chr3.56.v1.ND1-1", "KY21.Chr1.156.v1.SL1-1", "KY21.Chr1.157.v1.SL2-1", "KY21.Chr1.157.v1.SL1-1", "KY21.Chr1.157.v1.nonSL4-1", "KY21.Chr1.157.v1.nonSL6-1", "KY21.Chr1.157.v1.nonSL5-1", "KY21.Chr1.157.v1.nonSL8-1", "KY21.Chr1.157.v1.SL3-1", "KY21.Chr1.157.v1.nonSL7-1", "KY21.Chr9.577.v1.nonSL1-1", "KY21.Chr9.577.v1.nonSL2-1", "KY21.Chr12.699.v1.nonSL1-1", "KY21.Chr12.699.v1.nonSL2-1", "KY21.Chr9.398.v1.SL1-1", "KY21.UAContig13.6.v1.nonSL1-1", "KY21.UAContig13.6.v1.nonSL3-1", "KY21.UAContig13.6.v1.nonSL4-1", "KY21.UAContig13.6.v1.nonSL2-1", "KY21.UAContig13.6.v1.nonSL5-1", "KY21.UAContig32.28.v1.ND1-1", "KY21.UAContig36.22.v1.ND1-1", "KY21.UAContig49.9.v1.ND1-1", "KY21.Chr8.497.v1.SL1-1", "KY21.Chr7.192.v1.ND1-1", "KY21.Chr9.981.v1.ND1-1", "KY…
$ HumanID        &lt;chr&gt; "A2M", "AADAC", "AADAC", "AAMP", "AANAT", "AANAT", "AANAT", "AARS1", "AARS1", "ABAT", "ABAT", "ABAT", "ABAT", "ABAT", "ABAT", "ABAT", "ABAT", "ABAT", "ABCA10", "ABCA10", "ABCA2", "ABCA2", "ABCA3", "ABCA3", "ABCA3", "ABCA3", "ABCA3", "ABCA3", "ABCA3", "ABCA3", "ABCA3", "ABCA4", "ABCA5", "ABCA5", "ABCA5", "ABCA5", "ABCA8", "ABCA8", "ABCA9", "ABCB1", "ABCB1", "ABCB1", "ABCB1", "ABCB1", "ABCB1", "ABCB1", "ABCB1", "ABCB1", "ABCB1", "ABCB1", "ABCB10", "ABCB11", "TAP1", "ABCB5", "ABCB6", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB7", "ABCB8", "ABCB8", "ABCB8", "ABCB9", "ABCB9", "ABCB9", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "ABCC1", "AB…
$ HGNC           &lt;chr&gt; "HGNC:7", "HGNC:17", "HGNC:17", "HGNC:18", "HGNC:19", "HGNC:19", "HGNC:19", "HGNC:20", "HGNC:20", "HGNC:23", "HGNC:23", "HGNC:23", "HGNC:23", "HGNC:23", "HGNC:23", "HGNC:23", "HGNC:23", "HGNC:23", "HGNC:30", "HGNC:30", "HGNC:32", "HGNC:32", "HGNC:33", "HGNC:33", "HGNC:33", "HGNC:33", "HGNC:33", "HGNC:33", "HGNC:33", "HGNC:33", "HGNC:33", "HGNC:34", "HGNC:35", "HGNC:35", "HGNC:35", "HGNC:35", "HGNC:38", "HGNC:38", "HGNC:39", "HGNC:40", "HGNC:40", "HGNC:40", "HGNC:40", "HGNC:40", "HGNC:40", "HGNC:40", "HGNC:40", "HGNC:40", "HGNC:40", "HGNC:40", "HGNC:41", "HGNC:42", "HGNC:43", "HGNC:46", "HGNC:47", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:48", "HGNC:49", "HGNC:49…
$ Human_longname &lt;chr&gt; "alpha-2-macroglobulin", "arylacetamide deacetylase", "arylacetamide deacetylase", "angio associated migratory cell protein", "aralkylamine N-acetyltransferase", "aralkylamine N-acetyltransferase", "aralkylamine N-acetyltransferase", "alanyl-tRNA synthetase 1", "alanyl-tRNA synthetase 1", "4-aminobutyrate aminotransferase", "4-aminobutyrate aminotransferase", "4-aminobutyrate aminotransferase", "4-aminobutyrate aminotransferase", "4-aminobutyrate aminotransferase", "4-aminobutyrate aminotransferase", "4-aminobutyrate aminotransferase", "4-aminobutyrate aminotransferase", "4-aminobutyrate aminotransferase", "ATP binding cassette subfamily A member 10", "ATP binding cassette subfamily A member 10", "ATP binding cassette subfamily A member 2", "ATP binding cassette subfamily A member 2", "ATP binding cassette subfamily A member 3", "ATP binding cassette subfamily A member 3", "ATP binding cassette subfamily A member 3", "ATP binding cassette subfamily A member 3",…</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Annotating by HGNC</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>human_mitocarta <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">paste0</span>(directory, <span class="st">"a.reference_annotation/Human.MitoCarta3.0.xls"</span>), <span class="at">sheet =</span> <span class="st">"A Human MitoCarta3.0"</span>, <span class="at">trim_ws =</span> T, <span class="at">col_names =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">human_mitocarta =</span> <span class="st">"human_mitocarta"</span>)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># dplyr::select(1,3,5,15,16) $uniprot lenght and geneID</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(human_mitocarta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 1,136
Columns: 4
$ HumanGeneID     &lt;dbl&gt; 1537, 6390, 10229, 6389, 7384, 84274, 5160, 57017, 6182, 513, 9377, 122961, 9512, 7386, 498, 4967, 5162, 7385, 6392, 60488, 27089, 116540, 1629, 5166, 4191, 23107, 1431, 80273, 1737, 10128, 1743, 8050, 85476, 25874, 4719, 26589, 514, 5250, 51649, 2271, 23203, 506, 135154, 29796, 81689, 8803, 51805, 55699, 64960, 3419, 1353, 2110, 26519, 64981, 51069, 593, 7388, 192286, 539, 1892, 25875, 1337, 1355, 10939, 3030, 2108, 374291, 1376, 594, 3420, 23395, 35, 3954, 521, 4976, 549, 8802, 4729, 51004, 84545, 83451, 522, 4714, 9361, 1738, 9131, 79746, 139322, 124995, 92399, 4728, 34, 10989, 26520, 291, 25813, 4720, 4723, 50, 6832, 2235, 4528, 5245, 26275, 51116, 3313, 6834, 10935, 27069, 60558, 26517, 57128, 54948, 64976, 3421, 6391, 4711, 54949, 93058, 8209, 25828, 55168, 1345, 4715, 23788, 4700, 788, 65008, 10469, 10063, 440574, 6648, 1340, 7416, 8192, 3033, 33, 38, 115416, 10845, 4724, 51073, 708, 10531, 55245, 128308, 79922, 51102, 65080, 84263, 63931, 11194, 91647,…
$ Symbol          &lt;chr&gt; "CYC1", "SDHB", "COQ7", "SDHA", "UQCRC1", "COQ5", "PDHA1", "COQ9", "MRPL12", "ATP5F1D", "COX5A", "ISCA2", "PMPCB", "UQCRFS1", "ATP5F1A", "OGDH", "PDHB", "UQCRC2", "SDHD", "MRPS35", "UQCRQ", "MRPL53", "DBT", "PDK4", "MDH2", "MRPS27", "CS", "GRPEL1", "DLAT", "LRPPRC", "DLST", "PDHX", "GFM1", "MPC2", "NDUFS1", "MRPL46", "ATP5F1E", "SLC25A3", "MRPS23", "FH", "PMPCA", "ATP5F1B", "SDHAF4", "UQCR10", "ISCA1", "SUCLA2", "COQ3", "IARS2", "MRPS15", "IDH3A", "COX11", "ETFDH", "TIMM10", "MRPL34", "MRPL2", "BCKDHA", "UQCRH", "HIGD2A", "ATP5PO", "ECHS1", "LETMD1", "COX6A1", "COX15", "AFG3L2", "HADHA", "ETFA", "NDUFS7", "CPT2", "BCKDHB", "IDH3B", "LARS2", "ACADS", "LETM1", "ATP5ME", "OPA1", "AUH", "SUCLG1", "NDUFV2", "COQ6", "MRPL43", "ABHD11", "ATP5PF", "NDUFB8", "LONP1", "DLD", "AIFM1", "ECHDC3", "APOOL", "MRPL10", "MRRF", "NDUFS8", "ACADM", "IMMT", "TIMM9", "SLC25A4", "SAMM50", "NDUFS2", "NDUFV1", "ACO2", "SUPV3L1", "FECH", "MTIF2", "PHB", "HIBCH", "MRPS2", "HSPA9", "SURF…
$ Description     &lt;chr&gt; "cytochrome c1", "succinate dehydrogenase complex iron sulfur subunit B", "coenzyme Q7, hydroxylase", "succinate dehydrogenase complex flavoprotein subunit A", "ubiquinol-cytochrome c reductase core protein 1", "coenzyme Q5, methyltransferase", "pyruvate dehydrogenase E1 subunit alpha 1", "coenzyme Q9", "mitochondrial ribosomal protein L12", "ATP synthase F1 subunit delta", "cytochrome c oxidase subunit 5A", "iron-sulfur cluster assembly 2", "peptidase, mitochondrial processing subunit beta", "ubiquinol-cytochrome c reductase, Rieske iron-sulfur polypeptide 1", "ATP synthase F1 subunit alpha", "oxoglutarate dehydrogenase", "pyruvate dehydrogenase E1 subunit beta", "ubiquinol-cytochrome c reductase core protein 2", "succinate dehydrogenase complex subunit D", "mitochondrial ribosomal protein S35", "ubiquinol-cytochrome c reductase complex III subunit VII", "mitochondrial ribosomal protein L53", "dihydrolipoamide branched chain transacylase E2", "pyruvate dehydr…
$ human_mitocarta &lt;chr&gt; "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "human_mitocarta", "huma…</code></pre>
</div>
</div>
</section>
<section id="mapping-ncbi-entrez-gene-id-of-mitocarta-to-hgnc-symbols" class="level5" data-number="1.0.2.1.2">
<h5 data-number="1.0.2.1.2" class="anchored" data-anchor-id="mapping-ncbi-entrez-gene-id-of-mitocarta-to-hgnc-symbols"><span class="header-section-number">1.0.2.1.2</span> Mapping NCBI Entrez Gene ID of Mitocarta to HGNC symbols</h5>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Convert the NCBI Entrez Gene IDs from Mitocarta to their corresponding HGNC symbols.</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">"ensembl"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>) <span class="co"># use human genes dataset</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="co"># head(listAttributes(ensembl),20) # define the attributes to retrieve (HGNC symbol and Entrez Gene ID)</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="co"># split in two to handle the number of attributes</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>attributes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"entrezgene_id"</span>, <span class="st">"hgnc_symbol"</span>, <span class="st">"hgnc_id"</span>, <span class="st">"description"</span>)</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>attributes2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"entrezgene_id"</span>, <span class="st">"entrezgene_accession"</span>) <span class="co"># "entrezgene_description"</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="co"># convert Entrez Gene ID of mitocarta to HGNC symbols</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>entrez_ids <span class="ot">&lt;-</span> human_mitocarta<span class="sc">$</span>HumanGeneID <span class="co"># 1136</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>hgnc_symbols <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> attributes,</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">filters =</span> <span class="st">"entrezgene_id"</span>,</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">values =</span> entrez_ids,</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">mart =</span> ensembl</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>hgnc_symbols2 <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> attributes2,</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">filters =</span> <span class="st">"entrezgene_id"</span>,</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">values =</span> entrez_ids,</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">mart =</span> ensembl</span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>hgnc_symbols <span class="ot">&lt;-</span></span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>  hgnc_symbols <span class="sc">%&gt;%</span></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(hgnc_symbols2, <span class="at">by =</span> <span class="st">"entrezgene_id"</span>) <span class="sc">%&gt;%</span> <span class="co"># Merge the two dfs</span></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(hgnc_id, entrezgene_accession, hgnc_symbol, entrezgene_id, description)</span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(hgnc_symbols)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 1,137
Columns: 5
$ hgnc_id              &lt;chr&gt; "HGNC:40910", "HGNC:35410", "HGNC:40038", "HGNC:993", "HGNC:994", "HGNC:34528", "HGNC:35428", "HGNC:17312", "HGNC:40045", "HGNC:38844", "HGNC:47", "HGNC:2973", "HGNC:2264", "HGNC:44317", "HGNC:12367", "HGNC:9259", "HGNC:15714", "HGNC:16264", "HGNC:17366", "HGNC:10983", "HGNC:10985", "HGNC:50696", "HGNC:20567", "HGNC:18349", "HGNC:2244", "HGNC:16632", "HGNC:17310", "HGNC:16897", "HGNC:13734", "HGNC:16902", "HGNC:39", "HGNC:12730", "HGNC:1530", "HGNC:6737", "HGNC:17315", "HGNC:83", "HGNC:18001", "HGNC:14601", "HGNC:1329", "HGNC:17316", "HGNC:845", "HGNC:17663", "HGNC:17169", "HGNC:16637", "HGNC:14484", "HGNC:18155", "HGNC:7437", "HGNC:8587", "HGNC:14247", "HGNC:24639", "HGNC:7506", "HGNC:21062", "HGNC:9186", "HGNC:12843", "HGNC:344", "HGNC:7434", "HGNC:3978", "HGNC:2088", "HGNC:6985", "HGNC:8769", "HGNC:16985", "HGNC:24676", "HGNC:9354", "HGNC:315", "HGNC:15746", "HGNC:18431", "HGNC:53111", "HGNC:30862", "HGNC:6047", "HGNC:16429", "HGNC:11713", "HGNC:4907",…
$ entrezgene_accession &lt;chr&gt; "TSTD3", "TSTD1", "PET100", "BCL2L10", "BCL2L11", "TOMM6", "CMC4", "TIMM23", "PET117", "ATP5MF-PTCD1", "ABCB6", "DNM1L", "COX17", "PYURF", "TSFM", "PPIF", "LRPPRC", "TRAP1", "AASS", "SLC25A13", "SLC25A15", "PIGBOS1", "NME6", "DHRS2", "COQ7", "MRPS31", "TIMM17B", "RIDA", "GLYAT", "BCKDK", "ABCA9", "WARS2", "MICU1", "LYPLA1", "TIMM17A", "ACAA2", "TOMM40", "ECI2", "SLC30A9", "TIMM44", "ATP5PD", "PITRM1", "PRDX4", "HTATIP2", "MRPL28", "TXNRD2", "MTHFS", "PAICS", "ATP5MG", "PRELID3A", "MTX2", "FARS2", "POLQ", "YME1L1", "AHCYL1", "MTHFD2", "ALDH1L1", "CLPX", "ME3", "MRPS30", "DHRS4", "FASTK", "PRDX3", "AFG3L2", "TOMM34", "ACOT2", "HTD2", "UQCR11", "IMMT", "LIAS", "TDRKH", "HIBADH", "NUDT6", "NUDT5", "ABCB8", "PLPBP", "AKAP10", "MRPL3", "POLG2", "CA5B", "PXMP4", "RDH13", "FDX2", "HOGA1", "MTFR2", "PARK7", "PHB2", "ACOT7", "SDSL", "LACTB", "SLC25A25", "OSBPL1A", "PTPMT1", "OMA1", "SLC25A26", "MALSU1", "DHRS1", "CKMT1B", "CKMT2", "FAM210B", "COX20", "ACSM1", "TOP1…
$ hgnc_symbol          &lt;chr&gt; "TSTD3", "TSTD1", "PET100", "BCL2L10", "BCL2L11", "TOMM6", "CMC4", "TIMM23", "PET117", "ATP5MF-PTCD1", "ABCB6", "DNM1L", "COX17", "PYURF", "TSFM", "PPIF", "LRPPRC", "TRAP1", "AASS", "SLC25A13", "SLC25A15", "PIGBOS1", "NME6", "DHRS2", "COQ7", "MRPS31", "TIMM17B", "RIDA", "GLYAT", "BCKDK", "ABCA9", "WARS2", "MICU1", "LYPLA1", "TIMM17A", "ACAA2", "TOMM40", "ECI2", "SLC30A9", "TIMM44", "ATP5PD", "PITRM1", "PRDX4", "HTATIP2", "MRPL28", "TXNRD2", "MTHFS", "PAICS", "ATP5MG", "PRELID3A", "MTX2", "FARS2", "POLQ", "YME1L1", "AHCYL1", "MTHFD2", "ALDH1L1", "CLPX", "ME3", "MRPS30", "DHRS4", "FASTK", "PRDX3", "AFG3L2", "TOMM34", "ACOT2", "HTD2", "UQCR11", "IMMT", "LIAS", "TDRKH", "HIBADH", "NUDT6", "NUDT5", "ABCB8", "PLPBP", "AKAP10", "MRPL3", "POLG2", "CA5B", "PXMP4", "RDH13", "FDX2", "HOGA1", "MTFR2", "PARK7", "PHB2", "ACOT7", "SDSL", "LACTB", "SLC25A25", "OSBPL1A", "PTPMT1", "OMA1", "SLC25A26", "MALSU1", "DHRS1", "CKMT1B", "CKMT2", "FAM210B", "COX20", "ACSM1", "TOP1…
$ entrezgene_id        &lt;int&gt; 100130890, 100131187, 100131801, 10017, 10018, 100188893, 100272147, 100287932, 100303755, 100526740, 10058, 10059, 10063, 100996939, 10102, 10105, 10128, 10131, 10157, 10165, 10166, 101928527, 10201, 10202, 10229, 10240, 10245, 10247, 10249, 10295, 10350, 10352, 10367, 10434, 10440, 10449, 10452, 10455, 10463, 10469, 10476, 10531, 10549, 10553, 10573, 10587, 10588, 10606, 10632, 10650, 10651, 10667, 10721, 10730, 10768, 10797, 10840, 10845, 10873, 10884, 10901, 10922, 10935, 10939, 10953, 10965, 109703458, 10975, 10989, 11019, 11022, 11112, 11162, 11164, 11194, 11212, 11216, 11222, 11232, 11238, 11264, 112724, 112812, 112817, 113115, 11315, 11331, 11332, 113675, 114294, 114789, 114876, 114971, 115209, 115286, 115416, 115817, 1159, 1160, 116151, 116228, 116285, 116447, 116540, 116541, 117145, 118487, 118881, 118980, 119559, 122704, 122961, 123096, 123263, 123876, 124454, 124995, 125170, 125228, 125965, 125988, 126129, 126328, 126789, 127018, 128240, 12830…
$ description          &lt;chr&gt; "thiosulfate sulfurtransferase like domain containing 3 [Source:HGNC Symbol;Acc:HGNC:40910]", "thiosulfate sulfurtransferase like domain containing 1 [Source:HGNC Symbol;Acc:HGNC:35410]", "PET100 cytochrome c oxidase chaperone [Source:HGNC Symbol;Acc:HGNC:40038]", "BCL2 like 10 [Source:HGNC Symbol;Acc:HGNC:993]", "BCL2 like 11 [Source:HGNC Symbol;Acc:HGNC:994]", "translocase of outer mitochondrial membrane 6 [Source:HGNC Symbol;Acc:HGNC:34528]", "C-X9-C motif containing 4 [Source:HGNC Symbol;Acc:HGNC:35428]", "translocase of inner mitochondrial membrane 23 [Source:HGNC Symbol;Acc:HGNC:17312]", "PET117 cytochrome c oxidase chaperone [Source:HGNC Symbol;Acc:HGNC:40045]", "ATP5MF-PTCD1 readthrough [Source:HGNC Symbol;Acc:HGNC:38844]", "ATP binding cassette subfamily B member 6 (Langereis blood group) [Source:HGNC Symbol;Acc:HGNC:47]", "dynamin 1 like [Source:HGNC Symbol;Acc:HGNC:2973]", "cytochrome c oxidase copper chaperone COX17 [Source:HGNC Symbol;Acc:HGN…</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ENSG00000230623 not in HGNC</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(hgnc_symbols, "hgnc_symbols_to_match.txt", quote=F, col.names=T, row.names=F, sep="\t")</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co"># human_match_table &lt;- as_tibble(read.delim(file = paste0(directory,"a_reference_annotation/ciona_KY21/hgnc_symbols_to_match.txt"), header = TRUE, sep = "\t", dec = ".",strip.white = T))</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="co"># glimpse(human_match_table)</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>human_mitocarta_HGNC.df <span class="ot">&lt;-</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  human_mitocarta <span class="sc">%&gt;%</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(hgnc_symbols, <span class="at">multiple =</span> <span class="st">"all"</span>, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"HumanGeneID"</span> <span class="ot">=</span> <span class="st">"entrezgene_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">5</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">7</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(hgnc_id <span class="sc">==</span> <span class="st">"HGNC:1995"</span> <span class="sc">&amp;</span> HumanGeneID <span class="sc">==</span> <span class="dv">1159</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(hgnc_id) <span class="sc">%&gt;%</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="sc">-</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)) <span class="co"># 3 rows are empyty for hgnc_id and HumanGeneID</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a><span class="fu">vis_dat</span>(human_mitocarta_HGNC.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Mapping%20NCBI%20Entrez%20Gene%20ID%20of%20Mitocarta%20to%20HGNC%20symbols-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vis_miss</span>(human_mitocarta_HGNC.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Mapping%20NCBI%20Entrez%20Gene%20ID%20of%20Mitocarta%20to%20HGNC%20symbols-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">miss_case_summary</span>(human_mitocarta_HGNC.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1,133 × 3
    case n_miss pct_miss
   &lt;int&gt;  &lt;int&gt;    &lt;dbl&gt;
 1     1      0        0
 2     2      0        0
 3     3      0        0
 4     4      0        0
 5     5      0        0
 6     6      0        0
 7     7      0        0
 8     8      0        0
 9     9      0        0
10    10      0        0
# ℹ 1,123 more rows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># human_mitocarta_HGNC &lt;- human_mitocarta_HGNC[!apply(human_mitocarta_HGNC == " ", 1, all),]</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="co"># diff &lt;- human_mitocarta_HGNC$Symbol == human_mitocarta_HGNC$hgnc_symbol</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="co"># diff2 &lt;- human_mitocarta_HGNC$entrezgene_accession == human_mitocarta_HGNC$hgnc_symbol</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co"># human_mitocarta_HGNC[!diff, "entrezgene_accession"] # CKMT1A and CKMT1B are duplicates</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="co"># it look good you can drop the common columns</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="co"># PRORP HGNC:19958</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="co"># protein only RNase P catalytic subunit [Source:HGNC Symbol;Acc:HGNC:19958]</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="co"># protein only RNase P catalytic subunit</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="visualise-mitocarta-genes" class="level5" data-number="1.0.2.1.3">
<h5 data-number="1.0.2.1.3" class="anchored" data-anchor-id="visualise-mitocarta-genes"><span class="header-section-number">1.0.2.1.3</span> Visualise mitocarta genes</h5>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Visualise and select stable mitochondria proteins</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the mitochondrial proteins present in TMTcPro with less than 2 fold change in relative abundance over the time course.</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_rowMean_mitocartaHGNC.df <span class="ot">&lt;-</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMean.df <span class="sc">%&gt;%</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(ciona_annotation_gff3 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Protein_Id_transcript"</span> <span class="ot">=</span> <span class="st">"TranscriptID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">right_join</span>(human_mitocarta_HGNC.df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>, <span class="dv">5</span>), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"HGNC"</span> <span class="ot">=</span> <span class="st">"hgnc_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter_at</span>(<span class="fu">vars</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>), <span class="fu">all_vars</span>(. <span class="sc">&lt;=</span> <span class="dv">2</span>)) <span class="co"># fold change less than 2 over development</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 2nd filtering step</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>ciona_rowMean_mitocartaHGNC_flatDynamics.vector <span class="ot">&lt;-</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMean_mitocartaHGNC.df <span class="sc">%&gt;%</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="at">.fn =</span> <span class="sc">~</span> <span class="fu">str_replace</span>(., <span class="st">"_by_rowMean$"</span>, <span class="st">""</span>), <span class="at">.cols =</span> <span class="fu">ends_with</span>(<span class="st">"_by_rowMean"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transmute</span>(</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">Protein_Id_transcript =</span> Protein_Id_transcript,</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">diff_1cell_unfert =</span> <span class="fu">abs</span>(<span class="st">`</span><span class="at">fertE</span><span class="st">`</span> <span class="sc">-</span> unfE),</span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">diff_16cell_1cell =</span> <span class="fu">abs</span>(<span class="st">`</span><span class="at">cell16</span><span class="st">`</span> <span class="sc">-</span> <span class="st">`</span><span class="at">fertE</span><span class="st">`</span>),</span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">diff_110cell_16cell =</span> <span class="fu">abs</span>(<span class="st">`</span><span class="at">iniG</span><span class="st">`</span> <span class="sc">-</span> <span class="st">`</span><span class="at">cell16</span><span class="st">`</span>),</span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">diff_lateneurula_110cell =</span> <span class="fu">abs</span>(<span class="st">`</span><span class="at">latN</span><span class="st">`</span> <span class="sc">-</span> <span class="st">`</span><span class="at">iniG</span><span class="st">`</span>),</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">diff_midtailbud_lateneurula =</span> <span class="fu">abs</span>(<span class="st">`</span><span class="at">midTII</span><span class="st">`</span> <span class="sc">-</span> <span class="st">`</span><span class="at">latN</span><span class="st">`</span>),</span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">diff_latetailbud_midtailbud =</span> <span class="fu">abs</span>(<span class="st">`</span><span class="at">latTII</span><span class="st">`</span> <span class="sc">-</span> <span class="st">`</span><span class="at">midTII</span><span class="st">`</span>),</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">diff_larva_latetailbud =</span> <span class="fu">abs</span>(larva <span class="sc">-</span> <span class="st">`</span><span class="at">latTII</span><span class="st">`</span>)</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diff_sum =</span> <span class="fu">sum</span>(<span class="fu">c_across</span>(<span class="fu">starts_with</span>(<span class="st">"diff"</span>)))) <span class="sc">%&gt;%</span> <span class="co"># calculate sum difference</span></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(diff_sum <span class="sc">&lt;</span> <span class="dv">1</span>) <span class="co"># keep only proteinz with sum less than 2 sum difference</span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter step 1</span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a>toselect_flat <span class="ot">&lt;-</span></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>  ciona_rowMean_mitocartaHGNC_flatDynamics.vector<span class="sc">$</span>Protein_Id_transcript</span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter step 2</span></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a>ciona_rowMean_mitocartaHGNC_flatDynamics.df <span class="ot">&lt;-</span> ciona_TMTproC_rowMean_mitocartaHGNC.df[ciona_TMTproC_rowMean_mitocartaHGNC.df<span class="sc">$</span>Protein_Id_transcript <span class="sc">%in%</span> toselect_flat, ]</span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a><span class="co"># #  lag betwenn stages give the same resutls as above</span></span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a><span class="co"># df %&gt;%</span></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a><span class="co">#   rowwise() %&gt;%</span></span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(diff_all = sum(abs(diff(c_across(unfertilised:larva), lag = 1)))) %&gt;%</span></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(diff_all &lt; 1)</span></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a>HGNCmitocarta.dynamics<span class="fl">.1</span> <span class="ot">&lt;-</span> <span class="fu">plot_HGNCmitocarta_dynamics</span>(<span class="dv">0</span>, <span class="dv">140</span>)</span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a>HGNCmitocarta.dynamics<span class="fl">.2</span> <span class="ot">&lt;-</span> <span class="fu">plot_HGNCmitocarta_dynamics</span>(<span class="dv">141</span>, <span class="dv">270</span>)</span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a>HGNCmitocarta.dynamics<span class="fl">.1</span> <span class="co"># they look good</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Visualise%20mitocarta%20genes-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>HGNCmitocarta.dynamics<span class="fl">.2</span> <span class="co"># they look good</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Visualise%20mitocarta%20genes-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check result with plot: all proteins</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_rowMean_mitocartaHGNC.df <span class="sc">%&gt;%</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="at">.fn =</span> <span class="sc">~</span> <span class="fu">str_replace</span>(., <span class="st">"_by_rowMean$"</span>, <span class="st">""</span>), <span class="at">.cols =</span> <span class="fu">ends_with</span>(<span class="st">"_by_rowMean"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>), <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">values_to =</span> <span class="st">"expression"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">fct_relevel</span>(stage, ciona_stages)) <span class="sc">%&gt;%</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(stage, expression, <span class="at">color =</span> Protein_Id_transcript)) <span class="sc">+</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">group =</span> Protein_Id_transcript)) <span class="sc">+</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Some are still too dynamics and need to be exclude for normalization"</span>) <span class="sc">+</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> Protein_Id_transcript)) <span class="sc">+</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>  ciona_rowMean_mitocartaHGNC_flatDynamics.df <span class="sc">%&gt;%</span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="at">.fn =</span> <span class="sc">~</span> <span class="fu">str_replace</span>(., <span class="st">"_by_rowMean$"</span>, <span class="st">""</span>), <span class="at">.cols =</span> <span class="fu">ends_with</span>(<span class="st">"_by_rowMean"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>), <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">values_to =</span> <span class="st">"expression"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">fct_relevel</span>(stage, ciona_stages)) <span class="sc">%&gt;%</span></span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(stage, expression, <span class="at">color =</span> Protein_Id_transcript)) <span class="sc">+</span></span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">group =</span> Protein_Id_transcript)) <span class="sc">+</span></span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Good for normalization"</span>) <span class="sc">+</span></span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> Protein_Id_transcript)) <span class="sc">+</span></span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">2</span>) <span class="sc">&amp;</span> <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>,<span class="at">legend =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Visualise%20mitocarta%20genes-3.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="normalisation-step-using-rowmean-follow-by-flat-df-columnmedian" class="level5" data-number="1.0.2.1.4">
<h5 data-number="1.0.2.1.4" class="anchored" data-anchor-id="normalisation-step-using-rowmean-follow-by-flat-df-columnmedian"><span class="header-section-number">1.0.2.1.4</span> Normalisation step using rowMean follow by flat df columnMedian</h5>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Normalising using the median column of stable mitochondria proteins</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Median by column of mito protein</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>ciona_colMedian_normVector_mitocartaHGNC <span class="ot">&lt;-</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  ciona_rowMean_mitocartaHGNC_flatDynamics.df <span class="sc">%&gt;%</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize_all</span>(median)</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalise by median column</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df <span class="ot">&lt;-</span></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMean.df <span class="sc">%&gt;%</span></span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2_dfc</span>(ciona_colMedian_normVector_mitocartaHGNC, <span class="st">`</span><span class="at">/</span><span class="st">`</span>) <span class="sc">%&gt;%</span> <span class="co"># Divide each column of the selected df by the corresponding value in the colMedian_norm_.vector_flat vector</span></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(ciona_TMTproC_rowMean.df[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], ., <span class="at">.name_repair =</span> <span class="st">"minimal"</span>)</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df <span class="ot">&lt;-</span></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMean.df <span class="sc">%&gt;%</span></span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2_dfc</span>(ciona_colMedian_normVector_mitocartaHGNC, <span class="st">`</span><span class="at">/</span><span class="st">`</span>) <span class="sc">%&gt;%</span> <span class="co"># Divide each column of the selected df by the corresponding value in the colMedian_norm_.vector_flat vector</span></span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">.fn =</span> <span class="sc">~</span> <span class="fu">str_replace</span>(., <span class="st">"_by_rowMean$"</span>, <span class="st">"_byRowCol"</span>),</span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">.cols =</span> <span class="fu">ends_with</span>(<span class="st">"_by_rowMean"</span>)</span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(ciona_TMTproC_rowMean.df[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], ., <span class="at">.name_repair =</span> <span class="st">"minimal"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(ciona_annotation_gff3 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Protein_Id_transcript"</span> <span class="ot">=</span> <span class="st">"TranscriptID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">12</span>, <span class="dv">13</span>, <span class="dv">3</span>, <span class="fu">everything</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="assessing-the-mitocarta-normalisation-step" class="level5" data-number="1.0.2.1.5">
<h5 data-number="1.0.2.1.5" class="anchored" data-anchor-id="assessing-the-mitocarta-normalisation-step"><span class="header-section-number">1.0.2.1.5</span> Assessing the mitocarta normalisation step</h5>
<p>Normalization look good, with a cluster of protein ( ribosome and mitocondria) that are flat during embryogeneis . eg housekeeping</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Plot before after mitocarta only proteins dynamics </span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Before</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>ciona_avgNorm_std <span class="ot">&lt;-</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMean_colMedian.df <span class="sc">%&gt;%</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Protein_Id_transcript <span class="sc">%in%</span> toselect_flat) <span class="sc">%&gt;%</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ungroup() %&gt;%</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>), median)) <span class="sc">%&gt;%</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">str_remove</span>(name, <span class="st">"_byRowCol"</span>),</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">case_when</span>(</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"unfE"</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"fertE"</span> <span class="sc">~</span> <span class="dv">2</span>,</span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"cell16"</span> <span class="sc">~</span> <span class="dv">3</span>,</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"iniG"</span> <span class="sc">~</span> <span class="dv">4</span>,</span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"latN"</span> <span class="sc">~</span> <span class="dv">5</span>,</span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"midTII"</span> <span class="sc">~</span> <span class="dv">6</span>,</span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"latTII"</span> <span class="sc">~</span> <span class="dv">7</span>,</span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"larva"</span> <span class="sc">~</span> <span class="dv">8</span></span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a><span class="co"># After</span></span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>ciona_avgNorm_mitocarta <span class="ot">&lt;-</span></span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df <span class="sc">%&gt;%</span></span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Protein_Id_transcript <span class="sc">%in%</span> toselect_flat) <span class="sc">%&gt;%</span></span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>), median)) <span class="sc">%&gt;%</span></span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">str_remove</span>(name, <span class="st">"_byRowCol"</span>),</span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">case_when</span>(</span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"unfE"</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"fertE"</span> <span class="sc">~</span> <span class="dv">2</span>,</span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"cell16"</span> <span class="sc">~</span> <span class="dv">3</span>,</span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"iniG"</span> <span class="sc">~</span> <span class="dv">4</span>,</span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"latN"</span> <span class="sc">~</span> <span class="dv">5</span>,</span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"midTII"</span> <span class="sc">~</span> <span class="dv">6</span>,</span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"latTII"</span> <span class="sc">~</span> <span class="dv">7</span>,</span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a>      name <span class="sc">==</span> <span class="st">"larva"</span> <span class="sc">~</span> <span class="dv">8</span></span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb33-44"><a href="#cb33-44" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb33-45"><a href="#cb33-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-46"><a href="#cb33-46" aria-hidden="true" tabindex="-1"></a><span class="fu">function_ggparallel</span>(</span>
<span id="cb33-47"><a href="#cb33-47" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> ciona_TMTproC_rowMean_colMedian.df <span class="sc">%&gt;%</span></span>
<span id="cb33-48"><a href="#cb33-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Protein_Id_transcript <span class="sc">%in%</span> toselect_flat) <span class="sc">%&gt;%</span> <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">str_remove</span>(., <span class="st">"_byRowCol"</span>)),</span>
<span id="cb33-49"><a href="#cb33-49" aria-hidden="true" tabindex="-1"></a>  <span class="at">col_names =</span> <span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>,</span>
<span id="cb33-50"><a href="#cb33-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">group_col =</span> <span class="st">"Protein_Id_transcript"</span>,</span>
<span id="cb33-51"><a href="#cb33-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot_title =</span> <span class="st">"Ciona: std norm prot dynamcis"</span></span>
<span id="cb33-52"><a href="#cb33-52" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb33-53"><a href="#cb33-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb33-54"><a href="#cb33-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(ciona_avgNorm_std, <span class="fu">aes</span>(<span class="at">x =</span> name, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb33-55"><a href="#cb33-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb33-56"><a href="#cb33-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb33-57"><a href="#cb33-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb33-58"><a href="#cb33-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Linear Regression of Stage vs. Value"</span>) <span class="sc">+</span></span>
<span id="cb33-59"><a href="#cb33-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Stage"</span>) <span class="sc">+</span></span>
<span id="cb33-60"><a href="#cb33-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Value"</span>) <span class="sc">+</span></span>
<span id="cb33-61"><a href="#cb33-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-62"><a href="#cb33-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">function_ggparallel</span>(</span>
<span id="cb33-63"><a href="#cb33-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df <span class="sc">%&gt;%</span></span>
<span id="cb33-64"><a href="#cb33-64" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Protein_Id_transcript <span class="sc">%in%</span> toselect_flat) <span class="sc">%&gt;%</span> <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">str_remove</span>(., <span class="st">"_byRowCol"</span>)),</span>
<span id="cb33-65"><a href="#cb33-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">col_names =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>,</span>
<span id="cb33-66"><a href="#cb33-66" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_col =</span> <span class="st">"Protein_Id_transcript"</span>,</span>
<span id="cb33-67"><a href="#cb33-67" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot_title =</span> <span class="st">"Ciona: std mito prot dynamcis"</span></span>
<span id="cb33-68"><a href="#cb33-68" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb33-69"><a href="#cb33-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb33-70"><a href="#cb33-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(ciona_avgNorm_mitocarta, <span class="fu">aes</span>(<span class="at">x =</span> name, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb33-71"><a href="#cb33-71" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ggplot(ciona_avgNorm_mitocarta, aes(x = as.numeric(sub("Stage ", "", name)), y = value)) +</span></span>
<span id="cb33-72"><a href="#cb33-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb33-73"><a href="#cb33-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb33-74"><a href="#cb33-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Linear Regression of Stage vs. Value"</span>) <span class="sc">+</span></span>
<span id="cb33-75"><a href="#cb33-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb33-76"><a href="#cb33-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Stage"</span>) <span class="sc">+</span></span>
<span id="cb33-77"><a href="#cb33-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Value"</span>) <span class="sc">&amp;</span> <span class="fu">theme_pubr</span>(<span class="at">legend =</span> <span class="st">"none"</span>,<span class="at">base_family =</span> <span class="st">"Avenir"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Assessing%20the%20mitocarta%20normalisation%20step-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the normalised TMTproC data for Ciona</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">str_remove</span>(., <span class="st">"_byRowCol"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"protein_id_transcript"</span> <span class="ot">=</span> <span class="st">"Protein_Id_transcript"</span>, <span class="st">"protein_id_gene"</span> <span class="ot">=</span> <span class="st">"Protein_Id_Gene"</span>, <span class="st">"human_id"</span> <span class="ot">=</span> <span class="st">"HumanID"</span>, <span class="st">"hgnc"</span> <span class="ot">=</span> <span class="st">"HGNC"</span>, <span class="st">"number_of_peptides"</span> <span class="ot">=</span> <span class="st">"Number_of_peptides"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(., <span class="fu">paste0</span>(directory_save, <span class="st">"/ciona_proteome_norm.tsv"</span>), <span class="at">quote =</span> F, <span class="at">col.names =</span> T, <span class="at">row.names =</span> F, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="same-results-as-above" class="level5" data-number="1.0.2.1.6">
<h5 data-number="1.0.2.1.6" class="anchored" data-anchor-id="same-results-as-above"><span class="header-section-number">1.0.2.1.6</span> Same results as above</h5>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # divide each column of df by the corresponding column of vector median </span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co"># norm_ciona_mitocarta &lt;- ciona_TMTproC_rowMean.df</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="co"># norm_ciona_mitocarta[, 4:11] &lt;- mapply('/', norm_ciona_mitocarta[, 4:11], ciona_colMedian_normVector_mitocartaHGNC) </span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co"># norm_ciona_mitocarta</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plotting-mitocarta-normalisation" class="level5" data-number="1.0.2.1.7">
<h5 data-number="1.0.2.1.7" class="anchored" data-anchor-id="plotting-mitocarta-normalisation"><span class="header-section-number">1.0.2.1.7</span> Plotting mitocarta normalisation</h5>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Kmeans dynamics of Ciona protein dataset with mitocarta normalisation </span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">777</span>) <span class="co"># reproduce the clusters</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit and predict clusters with k = 8</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>ciona_kmeans_mito <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df[,<span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>], <span class="at">centers =</span> <span class="dv">8</span>, <span class="at">nstart =</span> <span class="dv">100</span>, <span class="at">iter.max =</span> <span class="dv">1000</span>)</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a><span class="co"># clean up the clustering and add cluster prediction to the original data set</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>ciona_results_mito <span class="ot">&lt;-</span> <span class="fu">augment</span>(ciona_kmeans_mito, ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df)</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>ciona_results_mito <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 14
  Protein_Id_transcript      Protein_Id_Gene HumanID HGNC       Number_of_peptides unfE_byRowCol fertE_byRowCol cell16_byRowCol iniG_byRowCol latN_byRowCol midTII_byRowCol latTII_byRowCol larva_byRowCol .cluster
  &lt;chr&gt;                      &lt;chr&gt;           &lt;chr&gt;   &lt;chr&gt;                   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;           &lt;dbl&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;          &lt;dbl&gt; &lt;fct&gt;   
1 KY21.Chr1.10.v1.nonSL2-1   KY21.Chr1.10    TAF9    HGNC:11542                  5         1.04           1.01            1.00          0.961         1.00            0.863           1.06           1.07  1       
2 KY21.Chr1.1000.v1.SL1-1    KY21.Chr1.1000  PSMB7   HGNC:9544                  13         1.06           1.08            1.06          0.918         0.943           0.948           1.03           0.970 1       
3 KY21.Chr1.1001.v1.SL1-1    KY21.Chr1.1001  LRRC47  HGNC:29207                 15         0.902          0.856           0.974         0.942         0.981           1.04            1.13           1.18  1       
4 KY21.Chr1.1002.v1.SL1-1    KY21.Chr1.1002  TDRD1   HGNC:11712                  6         3.40           3.23            0.519         0.125         0.189           0.201           0.243          0.180 3       
5 KY21.Chr1.1004.v1.nonSL4-1 KY21.Chr1.1004  DNAJC2  HGNC:13192                 21         0.908          0.881           0.988         0.976         1.03            1.09            1.05           1.08  1       </code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># summarizes  per-cluster level info eg how many point each cluster and within cluster sum of squares</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>ciona_cluster_avg_summary_mito <span class="ot">&lt;-</span> <span class="fu">tidy</span>(ciona_kmeans_mito)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>ciona_cluster_avg_summary_mito</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 11
  unfE_byRowCol fertE_byRowCol cell16_byRowCol iniG_byRowCol latN_byRowCol midTII_byRowCol latTII_byRowCol larva_byRowCol  size withinss cluster
          &lt;dbl&gt;          &lt;dbl&gt;           &lt;dbl&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;          &lt;dbl&gt; &lt;int&gt;    &lt;dbl&gt; &lt;fct&gt;  
1         1.01          0.983            1.06         0.993         1.01             0.979           0.999          0.981  3193     572. 1      
2         0.748         0.730            0.847        0.910         1.01             1.10            1.28           1.37   1998     430. 2      
3         1.90          1.86             1.33         0.684         0.611            0.552           0.563          0.552   237     403. 3      
4         0.111         0.0689           0.100        0.0740        0.0956           0.363           1.86           5.32    136     173. 4      
5         5.73          0.492            0.142        0.105         0.127            0.170           0.299          0.710    47     112. 5      
6         0.332         0.326            0.466        0.604         0.999            1.42            1.78           2.02    664     514. 6      
7         0.449         0.497            0.829        1.32          1.47             1.37            1.07           1.01    444     424. 7      
8         0.215         0.143            0.168        0.140         0.342            1.09            2.49           3.32    376     417. 8      </code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualise dynamics x cluster</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">function_ggparallel</span>(<span class="at">data =</span> ciona_cluster_avg_summary_mito <span class="sc">%&gt;%</span> <span class="fu">rename_with</span>(<span class="at">.fn =</span> <span class="sc">~</span><span class="fu">str_replace</span>(., <span class="st">"_byRowCol$"</span>, <span class="st">""</span>),<span class="at">.cols =</span> <span class="fu">ends_with</span>(<span class="st">"_byRowCol"</span>)),</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col_names =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_col =</span> <span class="st">"cluster"</span>,</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">plot_title =</span> <span class="st">"Ciona mito: cluster dynamcis"</span>) <span class="sc">+</span> </span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color=</span>cluster, <span class="at">size=</span><span class="fu">as.factor</span>(size))) <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> ciona_cluster_palette) <span class="sc">+</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span> cluster) <span class="sc">+</span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualise each protein behavior x cluster</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a><span class="fu">function_ggparallel</span>(<span class="at">data =</span> ciona_results_mito <span class="sc">%&gt;%</span> <span class="fu">rename_with</span>(<span class="at">.fn =</span> <span class="sc">~</span><span class="fu">str_replace</span>(., <span class="st">"_byRowCol$"</span>, <span class="st">""</span>),<span class="at">.cols =</span> <span class="fu">ends_with</span>(<span class="st">"_byRowCol"</span>)),</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col_names =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>,</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_col =</span> <span class="st">".cluster"</span>,</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">plot_title =</span> <span class="st">"Ciona mito: proteins dynamics x cluster"</span>) <span class="sc">+</span> </span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color=</span>.cluster)) <span class="sc">+</span></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> ciona_cluster_palette) <span class="sc">+</span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span> .cluster) <span class="sc">+</span></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">&amp;</span></span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="dv">0</span>, <span class="fl">0.5</span>), <span class="st">"lines"</span>), <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">theme_pubr</span>(<span class="at">legend =</span> <span class="st">"none"</span>,<span class="at">base_family =</span> <span class="st">"Avenir"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Plotting%20mitocarta%20normalisation-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># results_mito %&gt;%  pivot_longer(cols = c(6:13),</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                         names_to = "stage", </span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                         values_to = "expression") %&gt;% </span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(stage = str_remove(stage, "_byRowCol"),</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="co">#          stage = fct_relevel(stage, ciona_stages)) %&gt;% </span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   ggplot(aes(stage,expression, color = .cluster)) +</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_point(aes(group = Protein_Id_transcript)) +</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_line(aes(group = Protein_Id_transcript)) +</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   facet_wrap(~.cluster)</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Boxplot all entries</span></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>plot_normGood_ciona <span class="ot">&lt;-</span></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df <span class="sc">%&gt;%</span> </span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>), <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> as.factor) <span class="sc">%&gt;%</span>  </span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">str_detect</span>(stage, <span class="st">"_byRowCol"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">str_remove</span>(stage, <span class="st">"_byRowCol"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> <span class="fu">rev</span>(ciona_stages)), <span class="at">x =</span> value, <span class="at">fill =</span> stage)) <span class="sc">+</span>    </span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_line(aes(group = Protein_Id_transcript)) +</span></span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_point(aes(color=stage)) +</span></span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">setNames</span>(ciona_stages_palette, ciona_stages), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">setNames</span>(ciona_stages_palette, ciona_stages), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Normalised mitocarta good"</span>, <span class="at">y =</span> <span class="st">"Developmental stages"</span>, <span class="at">x =</span><span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> <span class="fu">theme_pubr</span>(<span class="at">legend =</span> <span class="st">"none"</span>,<span class="at">base_family =</span> <span class="st">"Avenir"</span>)</span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a>plot_original_ciona <span class="sc">+</span> plot_normRowCol_ciona <span class="sc">+</span> plot_normGood_ciona <span class="sc">&amp;</span> <span class="fu">theme_pubr</span>(<span class="at">legend =</span> <span class="st">"none"</span>,<span class="at">base_family =</span> <span class="st">"Avenir"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Plotting%20mitocarta%20normalisation-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="ribosomes" class="level4" data-number="1.0.2.2">
<h4 data-number="1.0.2.2" class="anchored" data-anchor-id="ribosomes"><span class="header-section-number">1.0.2.2</span> Ribosomes</h4>
<p>Normalising with ribosomes yield simialr results</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>ciona_ribosome <span class="ot">&lt;-</span>  <span class="fu">as_tibble</span>(<span class="fu">read_excel</span>(<span class="at">path =</span> <span class="fu">paste0</span>(directory,<span class="st">"a.reference_annotation/Supp Table - KY21_info.xlsx"</span>),<span class="at">col_names =</span> T,<span class="at">trim_ws =</span> T,<span class="at">sheet =</span> <span class="st">"ribosome_prot"</span>))</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(ciona_ribosome)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 3,307
Columns: 3
$ GeneID       &lt;chr&gt; "KY21.Chr1.702", "KY21.Chr1.702", "KY21.Chr1.702", "KY21.Chr1.702", "KY21.Chr1.702", "KY21.Chr1.705", "KY21.Chr1.705", "KY21.Chr1.705", "KY21.Chr1.705", "KY21.Chr1.705", "KY21.Chr1.705", "KY21.Chr1.705", "KY21.Chr1.705", "KY21.Chr1.720", "KY21.Chr1.720", "KY21.Chr1.720", "KY21.Chr1.720", "KY21.Chr1.720", "KY21.Chr1.754", "KY21.Chr1.754", "KY21.Chr1.754", "KY21.Chr1.770", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.791", "KY21.Chr1.795", "KY21.Chr1.795", "KY21.Chr1.797", "KY21.Chr1.824", "KY21.Chr1.884", "KY21.Chr1.904", "KY21.Chr1.909", "KY21.Chr1.909", "KY21.Chr1.909", "KY21.Chr1.909", "KY21.Chr1.909", "KY21.Chr1.979", "KY21.Chr1.979", "KY21.Chr1.979", "KY21.Chr1.979", "KY21.Chr1.1015", "KY21.Chr1.1071", "KY21.Chr1.1071", "KY21.Chr1.1072", "KY21.Chr1.1072", "KY…
$ TranscriptID &lt;chr&gt; "KY21.Chr1.702.v1.SL1-1", "KY21.Chr1.702.v1.SL2-1", "KY21.Chr1.702.v1.SL3-1", "KY21.Chr1.702.v1.nonSL4-1", "KY21.Chr1.702.v1.nonSL5-1", "KY21.Chr1.705.v1.ND1-1", "KY21.Chr1.705.v1.SL2-1", "KY21.Chr1.705.v1.SL3-1", "KY21.Chr1.705.v1.SL4-1", "KY21.Chr1.705.v1.SL5-1", "KY21.Chr1.705.v1.SL6-1", "KY21.Chr1.705.v1.SL7-1", "KY21.Chr1.705.v1.SL8-1", "KY21.Chr1.720.v1.nonSL1-1", "KY21.Chr1.720.v1.nonSL2-1", "KY21.Chr1.720.v1.nonSL3-1", "KY21.Chr1.720.v1.nonSL4-1", "KY21.Chr1.720.v1.nonSL5-1", "KY21.Chr1.754.v1.SL1-1", "KY21.Chr1.754.v1.nonSL2-1", "KY21.Chr1.754.v1.nonSL1-1", "KY21.Chr1.770.v1.ND1-1", "KY21.Chr1.791.v1.SL1-1", "KY21.Chr1.791.v1.nonSL2-1", "KY21.Chr1.791.v1.nonSL3-1", "KY21.Chr1.791.v2.SL1-2", "KY21.Chr1.791.v2.nonSL2-2", "KY21.Chr1.791.v2.nonSL3-2", "KY21.Chr1.791.v3.SL1-3", "KY21.Chr1.791.v3.nonSL2-3", "KY21.Chr1.791.v3.nonSL3-3", "KY21.Chr1.791.v4.SL1-4", "KY21.Chr1.791.v4.nonSL2-4", "KY21.Chr1.791.v4.nonSL3-4", "KY21.Chr1.791.v5.SL1-5", "KY21.Chr1.791.v…
$ annotation   &lt;chr&gt; "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "ribo", "…</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Annotated ribosomes</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_rowMean_ribosome.df <span class="ot">&lt;-</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMean.df <span class="sc">%&gt;%</span>  </span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">right_join</span>(ciona_ribosome, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Protein_Id_transcript"</span> <span class="ot">=</span> <span class="st">"TranscriptID"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(</span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">.fn =</span> <span class="sc">~</span><span class="fu">str_replace</span>(., <span class="st">"_by_rowMean$"</span>, <span class="st">""</span>),</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">.cols =</span> <span class="fu">ends_with</span>(<span class="st">"_by_rowMean"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter_at</span>(<span class="fu">vars</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">9</span>), <span class="fu">all_vars</span>(. <span class="sc">&lt;=</span> <span class="dv">2</span>)) <span class="co"># fc less than 2</span></span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>ciona_TMTproC_rowMean_ribosome.df <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(annotation) <span class="sc">%&gt;%</span>  <span class="fu">tally</span>() <span class="co"># 314 ribosome associated proteins ==&gt; 298 with cut off </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 2
  annotation     n
  &lt;chr&gt;      &lt;int&gt;
1 ribo         298</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table("HGNCmitocarta.dynamics.df.txt", quote=F, col.names=T, row.names=F, sep="\t")</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="do">#### Normalising using the median column of stable ribosomes proteins</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  Median by column</span></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>ciona_colMedian_normVector_Ribo <span class="ot">&lt;-</span> </span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMean_ribosome.df <span class="sc">%&gt;%</span> </span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">9</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize_all</span>(median)</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a><span class="co"># divide each column of df by the corresponding column of vector median </span></span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>ciona_normRibo <span class="ot">&lt;-</span> ciona_TMTproC_rowMean.df <span class="co">#%&gt;% drop_na()</span></span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>ciona_normRibo[, <span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>] <span class="ot">&lt;-</span> <span class="fu">mapply</span>(<span class="st">'/'</span>, ciona_normRibo[, <span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>], ciona_colMedian_normVector_Ribo)</span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ciona_normRibo) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Protein_Id_transcript"</span>, <span class="st">"Protein_Id_Gene"</span>, <span class="st">"Number_of_peptides"</span>, <span class="st">"unfE_byRowCol"</span>, <span class="st">"fertE_byRowCol"</span>, <span class="st">"cell16_byRowCol"</span>, <span class="st">"iniG_byRowCol"</span>, <span class="st">"latN_byRowCol"</span>, <span class="st">"midTII_byRowCol"</span>, <span class="st">"latTII_byRowCol"</span>, <span class="st">"larva_byRowCol"</span>)</span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a><span class="do">#### Kmeans dynamics of Ciona protein dataset with ribosomes normalisation </span></span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">777</span>) <span class="co"># reproduce the clusters</span></span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit and predict clusters with k = 8</span></span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a>ciona_kmeans_ribo <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(ciona_normRibo[,<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>], <span class="at">centers =</span> <span class="dv">8</span>, <span class="at">nstart =</span> <span class="dv">100</span>, <span class="at">iter.max =</span> <span class="dv">1000</span>)</span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a><span class="co"># clean up the clustering and add cluster prediction to the original data set</span></span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a>ciona_results_ribo <span class="ot">&lt;-</span> <span class="fu">augment</span>(ciona_kmeans_ribo, ciona_normRibo)</span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a><span class="co"># summarizes  per-cluster level info </span></span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a>ciona_cluster_avgSummary_ribo <span class="ot">&lt;-</span> <span class="fu">tidy</span>(ciona_kmeans_ribo)</span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a>ciona_cluster_avgSummary_ribo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 11
  unfE_byRowCol fertE_byRowCol cell16_byRowCol iniG_byRowCol latN_byRowCol midTII_byRowCol latTII_byRowCol larva_byRowCol  size withinss cluster
          &lt;dbl&gt;          &lt;dbl&gt;           &lt;dbl&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;          &lt;dbl&gt; &lt;int&gt;    &lt;dbl&gt; &lt;fct&gt;  
1         2.03          1.97             1.32         0.676         0.593            0.549           0.550          0.548   233     431. 1      
2         0.793         0.784            0.859        0.942         1.01             1.09            1.22           1.28   2023     427. 2      
3         1.06          1.05             1.07         1.03          1.01             0.979           0.949          0.922  3142     570. 3      
4         0.117         0.0732           0.101        0.0764        0.0958           0.364           1.77           5.01    136     159. 4      
5         0.349         0.344            0.473        0.627         1.01             1.43            1.69           1.90    666     499. 5      
6         0.226         0.152            0.169        0.144         0.345            1.09            2.37           3.13    378     407. 6      
7         0.471         0.526            0.838        1.34          1.46             1.37            1.03           0.969   470     440. 7      
8         6.02          0.524            0.143        0.108         0.128            0.171           0.285          0.670    47     116. 8      </code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ciona_cluster_avgSummary_ribo %&gt;% filter(cluster == 5) </span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>ciona_cluster_avgSummary_ribo <span class="sc">%&gt;%</span>  </span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>), <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">values_to =</span> <span class="st">"expression"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">str_remove</span>(stage, <span class="st">"_byRowCol"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">fct_relevel</span>(stage, ciona_stages), <span class="at">cluster =</span> <span class="fu">factor</span>(cluster)) <span class="sc">%&gt;%</span> </span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(stage, expression, <span class="at">color =</span> cluster)) <span class="sc">+</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="fu">interaction</span>(cluster, size), <span class="at">size =</span> <span class="fu">as.factor</span>(size))) <span class="sc">+</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> ciona_cluster_palette) <span class="sc">+</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>, <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Ribosome normalization"</span>, <span class="at">x =</span> <span class="st">""</span>) <span class="sc">&amp;</span> <span class="fu">theme_pubr</span>(<span class="at">legend =</span> <span class="st">"bottom"</span>,<span class="at">base_family =</span> <span class="st">"Avenir"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Ribosomes-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_size_discrete(name = "Cluster size") + # rename the size legend</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># guides(color = guide_none())</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="comparing-normalization-process" class="level3" data-number="1.0.3">
<h3 data-number="1.0.3" class="anchored" data-anchor-id="comparing-normalization-process"><span class="header-section-number">1.0.3</span> Comparing normalization process</h3>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### cluster plots</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>plot_std_norm <span class="ot">&lt;-</span> <span class="fu">plot_cluster_norm</span>(ciona_cluster_avgSummary, <span class="st">"Before good normalization"</span>) <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"#AB3329FF"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"#88A0DCFF"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"#ED968CFF"</span>,<span class="st">"4"</span> <span class="ot">=</span> <span class="st">"#381A61FF"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"#E78429FF"</span>, <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"#7FA074FF"</span>,<span class="st">"7"</span> <span class="ot">=</span> <span class="st">"#7C4B73FF"</span>, <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"#F9D14AFF"</span>))</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>plot_mito_norm <span class="ot">&lt;-</span> <span class="fu">plot_cluster_norm</span>(ciona_cluster_avg_summary_mito, <span class="st">"Mitocarta normalization"</span>) </span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>plot_ribo_norm <span class="ot">&lt;-</span> <span class="fu">plot_cluster_norm</span>(ciona_cluster_avgSummary_ribo, <span class="st">"Ribosome normalization"</span>) <span class="sc">+</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"#7FA074FF"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"#7C4B73FF"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"#88A0DCFF"</span>,<span class="st">"4"</span> <span class="ot">=</span> <span class="st">"#E78429FF"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"#381A61FF"</span>, <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"#ED968CFF"</span>,<span class="st">"7"</span> <span class="ot">=</span> <span class="st">"#F9D14AFF"</span>, <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"#AB3329FF"</span>)) </span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span>  plot_std_norm <span class="sc">+</span> plot_mito_norm <span class="sc">+</span> plot_ribo_norm</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a><span class="do">#### data pivot and linear regression</span></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a><span class="co"># function for cluster plots</span></span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a>ciona_results_std_pivot <span class="ot">&lt;-</span> <span class="fu">cluster_pivot_data</span>(ciona_results, <span class="dv">2</span>, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>))</span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a>ciona_results_mito_pivot <span class="ot">&lt;-</span> <span class="fu">cluster_pivot_data</span>(ciona_results_mito, <span class="dv">1</span>, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>))</span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a>ciona_results_ribo_pivot <span class="ot">&lt;-</span> <span class="fu">cluster_pivot_data</span>(ciona_results_ribo, <span class="dv">3</span>, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>))</span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot linear regression for each normalization method</span></span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a>bottom <span class="ot">&lt;-</span> </span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_cluster_lr</span>(ciona_results_std_pivot, <span class="st">"Standard normalisaton"</span>, <span class="cn">FALSE</span>, <span class="st">"B1"</span>) <span class="sc">+</span> </span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_cluster_lr</span>(ciona_results_mito_pivot, <span class="st">"Mitocarta normalisaton"</span>, <span class="cn">FALSE</span>,<span class="st">"B2"</span>) <span class="sc">+</span></span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_cluster_lr</span>(ciona_results_ribo_pivot, <span class="st">"Ribosome normalisaton"</span>, <span class="cn">FALSE</span>, <span class="st">"B3"</span>) <span class="sc">&amp;</span> <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>)</span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a>top <span class="sc">/</span> bottom <span class="sc">+</span> <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">'Normalization comparison'</span>) <span class="sc">+</span> <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span></span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="st">'Avenir'</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">30</span>), <span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/Comparing%20normalization%20process-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="relative-distribution" class="level3" data-number="1.0.4">
<h3 data-number="1.0.4" class="anchored" data-anchor-id="relative-distribution"><span class="header-section-number">1.0.4</span> Relative distribution</h3>
<p>Counts per protein scaled to integrate to 1. This is useful, to compare the proteome between species.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### reshaping the data to span 0-1 and/or to the number of multiplext TMT channels (8 for ciona; 10 in THao et al.;11 in Itallie et al.)</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Sum of channels/ stages is 8 (8 channels - 8 development stages)</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>ciona_proteome <span class="ot">&lt;-</span> </span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df <span class="sc">%&gt;%</span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">str_remove</span>(., <span class="st">"_byRowCol"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"protein_id_transcript"</span> <span class="ot">=</span> <span class="st">"Protein_Id_transcript"</span>, <span class="st">"protein_id_gene"</span> <span class="ot">=</span> <span class="st">"Protein_Id_Gene"</span>, <span class="st">"human_id"</span> <span class="ot">=</span> <span class="st">"HumanID"</span>, <span class="st">"hgnc"</span> <span class="ot">=</span> <span class="st">"HGNC"</span>, <span class="st">"number_of_peptides"</span> <span class="ot">=</span> <span class="st">"Number_of_peptides"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>ciona_proteome_normChan_tidy_wide <span class="ot">&lt;-</span> </span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>  ciona_TMTproC_rowMeancolMedian_flat_goodNorm.df <span class="sc">%&gt;%</span></span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">str_remove</span>(., <span class="st">"_byRowCol"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"protein_id_transcript"</span> <span class="ot">=</span> <span class="st">"Protein_Id_transcript"</span>, <span class="st">"protein_id_gene"</span> <span class="ot">=</span> <span class="st">"Protein_Id_Gene"</span>, <span class="st">"human_id"</span> <span class="ot">=</span> <span class="st">"HumanID"</span>, <span class="st">"hgnc"</span> <span class="ot">=</span> <span class="st">"HGNC"</span>, <span class="st">"number_of_peptides"</span> <span class="ot">=</span> <span class="st">"Number_of_peptides"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>), <span class="sc">~</span> .x <span class="sc">/</span> <span class="fu">sum</span>(<span class="fu">c_across</span>(<span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>))) <span class="sc">*</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() </span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a><span class="co"># mutate(row_sum = rowSums(across(c(6:13)), na.rm = TRUE))  # checking rowSum protein is 1</span></span>
<span id="cb51-20"><a href="#cb51-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-21"><a href="#cb51-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Sum of channels/stages is 1</span></span>
<span id="cb51-22"><a href="#cb51-22" aria-hidden="true" tabindex="-1"></a>ciona_proteome_normTo1_tidy_wide <span class="ot">&lt;-</span> ciona_proteome_normChan_tidy_wide <span class="sc">|&gt;</span> </span>
<span id="cb51-23"><a href="#cb51-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dplyr::select(-number_of_peptides) |&gt; </span></span>
<span id="cb51-24"><a href="#cb51-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>), <span class="sc">~</span>.<span class="sc">/</span><span class="dv">8</span> ))</span>
<span id="cb51-25"><a href="#cb51-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># mutate(row_sum = rowSums(across(c(6:13)), na.rm = TRUE))  # checking rowSum protein is 1</span></span>
<span id="cb51-26"><a href="#cb51-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-27"><a href="#cb51-27" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(ciona_proteome_normTo1_tidy_wide, <span class="fu">paste0</span>(directory_save, <span class="st">"/ciona_proteome_normTo1.tsv"</span>), <span class="at">quote =</span> F, <span class="at">col.names =</span> T, <span class="at">row.names =</span> F, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb51-28"><a href="#cb51-28" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(ciona_proteome_normChan_tidy_wide, <span class="fu">paste0</span>(directory_save, <span class="st">"/ciona_proteome_normToChan.tsv"</span>), <span class="at">quote =</span> F, <span class="at">col.names =</span> T, <span class="at">row.names =</span> F, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Checking that the pattern is the same</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Picking the protein Thr</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>a1 <span class="ot">&lt;-</span> ciona_proteome <span class="sc">|&gt;</span> </span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(human_id <span class="sc">==</span><span class="st">"THRB"</span>) <span class="sc">|&gt;</span>   </span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> unfE<span class="sc">:</span>larva, <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"unfE"</span>, <span class="st">"fertE"</span>, <span class="st">"cell16"</span>, <span class="st">"iniG"</span>, <span class="st">"latN"</span>, <span class="st">"midTII"</span>, <span class="st">"latTII"</span>, <span class="st">"larva"</span>))) <span class="sc">|&gt;</span> </span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> stage, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Scatterplot of Protein Expression Across Stages"</span>,</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Stage"</span>,</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Expression Level"</span>) <span class="sc">+</span></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) </span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a>a2 <span class="ot">&lt;-</span> ciona_proteome_normChan_tidy_wide <span class="sc">|&gt;</span> </span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(human_id <span class="sc">==</span><span class="st">"THRB"</span>) <span class="sc">|&gt;</span>   </span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> unfE<span class="sc">:</span>larva, <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"unfE"</span>, <span class="st">"fertE"</span>, <span class="st">"cell16"</span>, <span class="st">"iniG"</span>, <span class="st">"latN"</span>, <span class="st">"midTII"</span>, <span class="st">"latTII"</span>, <span class="st">"larva"</span>))) <span class="sc">|&gt;</span> </span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> stage, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Scatterplot of Protein Expression Across Stages"</span>,</span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Stage"</span>,</span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Expression Level"</span>) <span class="sc">+</span></span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>)</span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a>a3 <span class="ot">&lt;-</span> ciona_proteome_normTo1_tidy_wide <span class="sc">|&gt;</span> </span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(human_id <span class="sc">==</span><span class="st">"THRB"</span>) <span class="sc">|&gt;</span>   </span>
<span id="cb52-26"><a href="#cb52-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> unfE<span class="sc">:</span>larva, <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb52-27"><a href="#cb52-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"unfE"</span>, <span class="st">"fertE"</span>, <span class="st">"cell16"</span>, <span class="st">"iniG"</span>, <span class="st">"latN"</span>, <span class="st">"midTII"</span>, <span class="st">"latTII"</span>, <span class="st">"larva"</span>))) <span class="sc">|&gt;</span> </span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> stage, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb52-30"><a href="#cb52-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Scatterplot of Protein Expression Across Stages"</span>,</span>
<span id="cb52-31"><a href="#cb52-31" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Stage"</span>,</span>
<span id="cb52-32"><a href="#cb52-32" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Expression Level"</span>) <span class="sc">+</span></span>
<span id="cb52-33"><a href="#cb52-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) </span>
<span id="cb52-34"><a href="#cb52-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-35"><a href="#cb52-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-36"><a href="#cb52-36" aria-hidden="true" tabindex="-1"></a>a1 <span class="sc">/</span> a2 <span class="sc">/</span>a3 </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/00.Ciona_prot_normalization/thr%20profile%20check-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># &amp; coord_fixed(ratio = 1)</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="co"># a1 / a2 /a3 &amp; coord_equal()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



<script src="00.Ciona_prot_normalization_files/libs/quarto-html/zenscroll-min.js"></script>
</body></html>