<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Andrea Mariossi">
<meta name="dcterms.date" content="2023-10-25">

<title>Analysis.Ciona_TMTproC_part1</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="01.Ciona_prot_QC_files/libs/clipboard/clipboard.min.js"></script>
<script src="01.Ciona_prot_QC_files/libs/quarto-html/quarto.js"></script>
<script src="01.Ciona_prot_QC_files/libs/quarto-html/popper.min.js"></script>
<script src="01.Ciona_prot_QC_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="01.Ciona_prot_QC_files/libs/quarto-html/anchor.min.js"></script>
<link href="01.Ciona_prot_QC_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="01.Ciona_prot_QC_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="01.Ciona_prot_QC_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="01.Ciona_prot_QC_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="01.Ciona_prot_QC_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<style>html{ scroll-behavior: smooth; }</style>


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-full toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#ciona-proteome" id="toc-ciona-proteome" class="nav-link active" data-scroll-target="#ciona-proteome"><span class="header-section-number">1</span> <em>Ciona</em> proteome</a>
  <ul>
  <li><a href="#loading-libraries" id="toc-loading-libraries" class="nav-link" data-scroll-target="#loading-libraries"><span class="header-section-number">1.0.1</span> Loading libraries</a></li>
  <li><a href="#functions-misc-loading-data" id="toc-functions-misc-loading-data" class="nav-link" data-scroll-target="#functions-misc-loading-data"><span class="header-section-number">1.0.2</span> Functions &amp; misc &amp; loading data</a></li>
  <li><a href="#qc-ms-data-with-lda" id="toc-qc-ms-data-with-lda" class="nav-link" data-scroll-target="#qc-ms-data-with-lda"><span class="header-section-number">1.1</span> QC MS data with LDA</a>
  <ul>
  <li><a href="#all-fractions-together" id="toc-all-fractions-together" class="nav-link" data-scroll-target="#all-fractions-together"><span class="header-section-number">1.1.1</span> All fractions together</a></li>
  <li><a href="#example-of-a-fraction" id="toc-example-of-a-fraction" class="nav-link" data-scroll-target="#example-of-a-fraction"><span class="header-section-number">1.1.2</span> Example of a fraction</a></li>
  </ul></li>
  <li><a href="#descriptive-statistics" id="toc-descriptive-statistics" class="nav-link" data-scroll-target="#descriptive-statistics"><span class="header-section-number">1.2</span> Descriptive statistics</a></li>
  <li><a href="#peptides-proteins-quantification" id="toc-peptides-proteins-quantification" class="nav-link" data-scroll-target="#peptides-proteins-quantification"><span class="header-section-number">1.3</span> Peptides &amp; proteins quantification</a></li>
  <li><a href="#mapping-isoforms" id="toc-mapping-isoforms" class="nav-link" data-scroll-target="#mapping-isoforms"><span class="header-section-number">1.4</span> Mapping isoforms</a></li>
  <li><a href="#top-and-bottom-proteins-expressed-by-stage" id="toc-top-and-bottom-proteins-expressed-by-stage" class="nav-link" data-scroll-target="#top-and-bottom-proteins-expressed-by-stage"><span class="header-section-number">1.5</span> Top and bottom proteins expressed by stage</a></li>
  <li><a href="#protein-distribution" id="toc-protein-distribution" class="nav-link" data-scroll-target="#protein-distribution"><span class="header-section-number">1.6</span> Protein distribution</a>
  <ul>
  <li><a href="#boxplots-density" id="toc-boxplots-density" class="nav-link" data-scroll-target="#boxplots-density"><span class="header-section-number">1.6.1</span> Boxplots &amp; density</a></li>
  <li><a href="#distribution-per-chromosome" id="toc-distribution-per-chromosome" class="nav-link" data-scroll-target="#distribution-per-chromosome"><span class="header-section-number">1.6.2</span> Distribution per chromosome</a></li>
  <li><a href="#distribution-over-development" id="toc-distribution-over-development" class="nav-link" data-scroll-target="#distribution-over-development"><span class="header-section-number">1.6.3</span> Distribution over development</a></li>
  <li><a href="#protein-distribution-by-selecting-10-quantile-cutoff-x-protein" id="toc-protein-distribution-by-selecting-10-quantile-cutoff-x-protein" class="nav-link" data-scroll-target="#protein-distribution-by-selecting-10-quantile-cutoff-x-protein"><span class="header-section-number">1.6.4</span> Protein distribution by selecting &gt;10% quantile cutoff x protein</a></li>
  <li><a href="#high-abundance-proteins" id="toc-high-abundance-proteins" class="nav-link" data-scroll-target="#high-abundance-proteins"><span class="header-section-number">1.6.5</span> High-Abundance Proteins</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content column-page-right" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Analysis.Ciona_TMTproC_part1</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Andrea Mariossi </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 25, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="ciona-proteome" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="ciona-proteome"><span class="header-section-number">1</span> <em>Ciona</em> proteome</h2>
<section id="loading-libraries" class="level4" data-number="1.0.1">
<h4 data-number="1.0.1" class="anchored" data-anchor-id="loading-libraries"><span class="header-section-number">1.0.1</span> Loading libraries</h4>
</section>
<section id="functions-misc-loading-data" class="level4" data-number="1.0.2">
<h4 data-number="1.0.2" class="anchored" data-anchor-id="functions-misc-loading-data"><span class="header-section-number">1.0.2</span> Functions &amp; misc &amp; loading data</h4>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### directories</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>directory_figures <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Vector with stages in develpmental order</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>ciona_stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"unfE"</span>, <span class="st">"fertE"</span>, <span class="st">"cell16"</span>, <span class="st">"iniG"</span>, <span class="st">"latN"</span>, <span class="st">"midTII"</span>, <span class="st">"latTII"</span>, <span class="st">"larva"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="do">#### color palette</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>ciona_stages_palette <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#009392FF"</span>, <span class="st">"#39B185FF"</span>, <span class="st">"#9CCB86FF"</span>, <span class="st">"#E9E29CFF"</span>, <span class="st">"#EEB479FF"</span>, <span class="st">"#E88471FF"</span>, <span class="st">"#CF597EFF"</span>, <span class="st">"#D82526FF"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>ciona_cluster_palette <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#88A0DCFF"</span>, <span class="st">"#381A61FF"</span>, <span class="st">"#7C4B73FF"</span>, <span class="st">"#ED968CFF"</span>, <span class="st">"#AB3329FF"</span>, <span class="st">"#E78429FF"</span>, <span class="st">"#F9D14AFF"</span>, <span class="st">"#7FA074FF"</span>) <span class="co"># &gt; paletteer::paletteer_d("MetBrewer::Archambault") + MetBrewer::Cassatt2" - Colorblind-Friendly</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>isoform_palette <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#FBE426"</span>, <span class="st">"#36b39c"</span>, <span class="st">"#f3c300"</span>, <span class="st">"#875692"</span>, <span class="st">"#f38400"</span>, <span class="st">"#a1caf1"</span>, <span class="st">"#be0032"</span>, <span class="st">"#c2b280"</span>, <span class="st">"#848482"</span>, <span class="st">"#008856"</span>, <span class="st">"#e68fac"</span>, <span class="st">"#0067a5"</span>, <span class="st">"#f99379"</span>, <span class="st">"#604e97"</span>, <span class="st">"#f6a600"</span>, <span class="st">"#b3446c"</span>, <span class="st">"#dcd300"</span>, <span class="st">"#882d17"</span>, <span class="st">"#8db600"</span>, <span class="st">"#654522"</span>, <span class="st">"#e25822"</span>, <span class="st">"#2b3d26"</span>, <span class="st">"#AA0DFE"</span>, <span class="st">"#3283FE"</span>, <span class="st">"#85660D"</span>, <span class="st">"#782AB6"</span>, <span class="st">"#565656"</span>, <span class="st">"#1C8356"</span>, <span class="st">"#16FF32"</span>, <span class="st">"#F7E1A0"</span>, <span class="st">"#E2E2E2"</span>, <span class="st">"#1CBE4F"</span>, <span class="st">"#C4451C"</span>, <span class="st">"#DEA0FD"</span>, <span class="st">"#FE00FA"</span>, <span class="st">"#325A9B"</span>, <span class="st">"#FEAF16"</span>, <span class="st">"#F8A19F"</span>, <span class="st">"#90AD1C"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="qc-ms-data-with-lda" class="level3" data-number="1.1">
<h3 data-number="1.1" class="anchored" data-anchor-id="qc-ms-data-with-lda"><span class="header-section-number">1.1</span> QC MS data with LDA</h3>
<p>Starting with checking the quality of MS</p>
<section id="all-fractions-together" class="level4" data-number="1.1.1">
<h4 data-number="1.1.1" class="anchored" data-anchor-id="all-fractions-together"><span class="header-section-number">1.1.1</span> All fractions together</h4>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### There are multiple fractions, first combine all</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># forward = target</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># reverse = decoy</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Files </span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"LDA_1696944876.csv"</span>, <span class="st">"LDA_1696944628.csv"</span>, <span class="st">"LDA_1696944697.csv"</span>, <span class="st">"LDA_1696944803.csv"</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>full_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(directory_figures, files)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="co"># list of df</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">&lt;-</span> <span class="fu">map</span>(full_paths, <span class="sc">~</span><span class="fu">read_csv</span>(.x, <span class="at">col_names =</span> T))</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co"># column names </span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>col_names <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(full_paths[<span class="dv">1</span>], <span class="at">n_max =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">names</span>()</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">&lt;-</span> <span class="fu">map</span>(data_list, set_names, <span class="at">nm =</span> col_names)</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Bind df together</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>LDA_data <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(data_list) <span class="sc">|&gt;</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">clean_names</span>() <span class="sc">|&gt;</span> </span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"delta_corr"</span> <span class="ot">=</span> <span class="st">"number_916_corr"</span>)</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>LDA_data <span class="sc">|&gt;</span>  <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 17
  scan_f scan_l     z  rank x_corr delta_corr uniq_number_916_corr number_ions number_ions_link                                                                                                         reference                  reference_link                                                                                                                            peptide                           peptide_link                                                                                                                                                       pept_length gene_symbol annotation lda_score
   &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;      &lt;dbl&gt;                &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;                                                                                                                    &lt;chr&gt;                      &lt;chr&gt;                                                                                                                                     &lt;chr&gt;                             &lt;chr&gt;                                                                                                                                                                    &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;          &lt;dbl&gt;
1    130    130     2     1  1.80       0.345                0.345 9/34        /gfy/www/modules//sdig/index.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=1&amp;scanf=130&amp;charge=2&amp;value=9/34  ##KY21.Chr4.75.v1.nonSL1-1 /gfy/www/modules//consensus/?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=1&amp;scanf=130&amp;charge=2&amp;value=##KY21.Chr4.75.v1.nonSL1-1 K.MFIGFVAMVWTQSFKFFK.R            /gfy/www/modules/show_out/show_out_wrapper.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=1&amp;scanf=130&amp;charge=2&amp;value=K.MFIGFVAMVWTQSFKFFK.R                     18 &lt;NA&gt;        ####            1.12
2    131    131     2     1  3.83       0.691                0.691 22/26       /gfy/www/modules//sdig/index.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=2&amp;scanf=131&amp;charge=2&amp;value=22/26 KY21.Chr1.2009.v1.ND1-1    /gfy/www/modules//consensus/?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=2&amp;scanf=131&amp;charge=2&amp;value=KY21.Chr1.2009.v1.ND1-1    K.PNLVDWLSDWSMEK.S                /gfy/www/modules/show_out/show_out_wrapper.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=2&amp;scanf=131&amp;charge=2&amp;value=K.PNLVDWLSDWSMEK.S                         14 &lt;NA&gt;        &lt;NA&gt;            2.06
3    132    132     2     1  0.363      0.235                0.235 2/26        /gfy/www/modules//sdig/index.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=3&amp;scanf=132&amp;charge=2&amp;value=2/26  KY21.Chr5.42.v1.nonSL5-1   /gfy/www/modules//consensus/?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=3&amp;scanf=132&amp;charge=2&amp;value=KY21.Chr5.42.v1.nonSL5-1   R.DFM*ETLNTLKYANR.A               /gfy/www/modules/show_out/show_out_wrapper.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=3&amp;scanf=132&amp;charge=2&amp;value=R.DFM*ETLNTLKYANR.A                        14 &lt;NA&gt;        &lt;NA&gt;           -2.10
4    133    133     2     1  0.658      0.037                0.037 4/26        /gfy/www/modules//sdig/index.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=4&amp;scanf=133&amp;charge=2&amp;value=4/26  ##KY21.Chr12.421.v1.ND1-1  /gfy/www/modules//consensus/?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=4&amp;scanf=133&amp;charge=2&amp;value=##KY21.Chr12.421.v1.ND1-1  R.PQSYDIILVIDDLK.L                /gfy/www/modules/show_out/show_out_wrapper.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=4&amp;scanf=133&amp;charge=2&amp;value=R.PQSYDIILVIDDLK.L                         14 &lt;NA&gt;        ####           -1.68
5    134    134     3     1  0.374      0.015                0.015 4/112       /gfy/www/modules//sdig/index.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=5&amp;scanf=134&amp;charge=3&amp;value=4/112 KY21.Chr10.1078.v1.SL1-1   /gfy/www/modules//consensus/?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=5&amp;scanf=134&amp;charge=3&amp;value=KY21.Chr10.1078.v1.SL1-1   K.LGSHYLVMVSLPFAEDVRPLSFESLFSGK.N /gfy/www/modules/show_out/show_out_wrapper.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=5&amp;scanf=134&amp;charge=3&amp;value=K.LGSHYLVMVSLPFAEDVRPLSFESLFSGK.N          29 &lt;NA&gt;        &lt;NA&gt;           -1.64
6    135    135     2     1  0.222      0                    0.04  1/28        /gfy/www/modules//sdig/index.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=6&amp;scanf=135&amp;charge=2&amp;value=1/28  ##KY21.Chr4.577.v1.ND1-1   /gfy/www/modules//consensus/?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=6&amp;scanf=135&amp;charge=2&amp;value=##KY21.Chr4.577.v1.ND1-1   R.LYKSLGM*ELQLTILR.H              /gfy/www/modules/show_out/show_out_wrapper.php?run_id=21229&amp;search_id=28317&amp;scans_id=21229&amp;peptide_id=6&amp;scanf=135&amp;charge=2&amp;value=R.LYKSLGM*ELQLTILR.H                       15 &lt;NA&gt;        ####           -1.72</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the $ in front of the protein are the reverse/decoy ones</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="do">#### plots</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>LDA_data <span class="sc">|&gt;</span> </span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction =</span> <span class="fu">if_else</span>(<span class="fu">str_detect</span>(reference, <span class="st">"^##"</span>), <span class="st">"reverse"</span>, <span class="st">"forward"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(reference, <span class="st">'_HUMAN_contaminant'</span>) <span class="sc">&amp;</span> </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>        x_corr <span class="sc">&gt;</span> <span class="dv">1</span> ) <span class="sc">|&gt;</span> </span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>lda_score, <span class="at">fill=</span>direction)) <span class="sc">+</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="fl">0.1</span>, <span class="at">position=</span><span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#BDE1C2"</span>, <span class="st">"#D20076FF"</span>)) <span class="sc">+</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#BDE1C2"</span>, <span class="st">"#D20076FF"</span>)) <span class="sc">+</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"LDA score"</span>, <span class="at">y=</span><span class="st">"Count"</span>, <span class="at">fill=</span><span class="st">"direction"</span>) <span class="sc">+</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>LDA_data <span class="sc">|&gt;</span> </span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction =</span> <span class="fu">if_else</span>(<span class="fu">str_detect</span>(reference, <span class="st">"^##"</span>), <span class="st">"reverse"</span>, <span class="st">"forward"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(reference, <span class="st">'_HUMAN_contaminant'</span>) <span class="sc">&amp;</span> </span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>        x_corr <span class="sc">&gt;</span> <span class="dv">1</span> ) <span class="sc">|&gt;</span> </span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>lda_score, <span class="at">fill=</span>direction,<span class="at">color=</span>direction)) <span class="sc">+</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_freqpoly</span>(<span class="at">binwidth =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#BDE1C2"</span>, <span class="st">"#D20076FF"</span>)) <span class="sc">+</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#BDE1C2"</span>, <span class="st">"#D20076FF"</span>)) <span class="sc">+</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"LDA Score"</span>, <span class="at">y=</span><span class="st">"Count"</span>, <span class="at">fill=</span><span class="st">"direction"</span>) <span class="sc">+</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/lda%20all%20franctions-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="example-of-a-fraction" class="level4" data-number="1.1.2">
<h4 data-number="1.1.2" class="anchored" data-anchor-id="example-of-a-fraction"><span class="header-section-number">1.1.2</span> Example of a fraction</h4>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### LDA fitlering at 1% FDR change for each fraction - plotting just one as an example</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Score Cutoff: 1.7402</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>lda <span class="ot">&lt;-</span>  <span class="fu">as_tibble</span>(</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_excel</span>(<span class="at">path =</span> <span class="fu">paste0</span>(directory_figures, <span class="st">"lda_single_fraction.xlsx"</span>),</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">sheet =</span> <span class="st">"Mariossi_1696882402"</span>,</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>             <span class="co"># range = cell_cols("A:K"),</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">col_names=</span><span class="cn">TRUE</span>))  <span class="sc">|&gt;</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">clean_names</span>()</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co"># https://stackoverflow.com/questions/74778096/customize-the-position-of-geom-rug</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>limits <span class="ot">&lt;-</span> <span class="fu">range</span>(lda<span class="sc">$</span>lda_score)</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>step   <span class="ot">&lt;-</span> <span class="fu">diff</span>(limits) <span class="sc">*</span> <span class="fl">0.1</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>size   <span class="ot">&lt;-</span> <span class="fl">0.45</span> <span class="sc">*</span> step</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>plot_lda <span class="ot">&lt;-</span> lda <span class="sc">%&gt;%</span> </span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">direction =</span> <span class="fu">if_else</span>(<span class="fu">str_detect</span>(reference, <span class="st">"^##"</span>), <span class="st">"Decoy"</span>, <span class="st">"target"</span>),</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">expression=</span> <span class="fu">if_else</span>(direction <span class="sc">==</span> <span class="st">"decoy"</span>, <span class="st">"5"</span>,<span class="st">"16"</span>),</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">lda_group =</span> <span class="fu">case_when</span>(</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>      lda_score <span class="sc">&gt;</span> <span class="fl">1.7402</span> <span class="sc">&amp;</span> direction <span class="sc">==</span> <span class="st">"target"</span> <span class="sc">~</span> <span class="st">"Target pass"</span>,</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>      lda_score <span class="sc">&lt;</span> <span class="fl">1.7402</span> <span class="sc">&amp;</span> direction <span class="sc">==</span> <span class="st">"target"</span> <span class="sc">~</span> <span class="st">"Target fail"</span>,</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">as.character</span>(direction)</span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">lda_group =</span> <span class="fu">factor</span>(lda_group, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Target fail"</span>, <span class="st">"Target pass"</span>, <span class="st">"Decoy"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(x_corr <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> lda_score <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">fill=</span>lda_group)) <span class="sc">+</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>lda_score, <span class="at">alpha=</span>lda_group), <span class="at">color=</span><span class="st">"black"</span>, <span class="at">binwidth=</span><span class="fl">0.1</span>, <span class="at">position=</span><span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> lda_group,</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">x=</span>lda_score,</span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>      <span class="at">xend =</span> lda_score,</span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>      <span class="at">y    =</span> limits[<span class="dv">1</span>] <span class="sc">-</span> <span class="fu">as.numeric</span>(expression) <span class="sc">*</span> step <span class="sc">+</span> size<span class="sc">*</span><span class="dv">10</span>,</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>      <span class="at">yend =</span> limits[<span class="dv">1</span>] <span class="sc">-</span> <span class="fu">as.numeric</span>(expression) <span class="sc">*</span> step <span class="sc">-</span> size<span class="sc">*</span><span class="dv">10</span></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">Target fail</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"#BDE1C2"</span>, <span class="st">`</span><span class="at">Target pass</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"#00A087FF"</span>, <span class="at">Decoy =</span> <span class="st">"#D20076FF"</span>)) <span class="sc">+</span></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">Target fail</span><span class="st">`</span> <span class="ot">=</span> <span class="fl">0.1</span>, <span class="st">`</span><span class="at">Target pass</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="at">Decoy =</span> <span class="dv">1</span>), <span class="at">guide=</span><span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">Target fail</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"#BDE1C2"</span>, <span class="st">`</span><span class="at">Target pass</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"#00A087FF"</span>, <span class="at">Decoy =</span> <span class="st">"#D20076FF"</span>), <span class="at">guide=</span><span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">1.2</span>, <span class="at">y =</span> <span class="dv">150</span>, <span class="at">size=</span><span class="dv">5</span>,<span class="at">label =</span> <span class="st">"XCorr &gt; 1"</span>) <span class="sc">+</span></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"LDA score"</span>, <span class="at">y=</span><span class="st">"Counts"</span>, <span class="at">fill=</span><span class="st">"LDA Group"</span>) <span class="sc">+</span></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>,<span class="at">base_size =</span> <span class="dv">13</span>) <span class="sc">+</span> </span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position=</span><span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.83</span>), </span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title=</span><span class="fu">element_blank</span>()</span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>plot_lda</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/lda%20single%20franction-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cairo_pdf</span>(<span class="st">"plot_lda.pdf"</span>, <span class="at">width=</span><span class="fl">8.5</span>, <span class="at">height=</span><span class="fl">5.6</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot_lda)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="descriptive-statistics" class="level3" data-number="1.2">
<h3 data-number="1.2" class="anchored" data-anchor-id="descriptive-statistics"><span class="header-section-number">1.2</span> Descriptive statistics</h3>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### loading the data</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>ciona_prot.df <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(<span class="fu">read_delim</span>(<span class="fu">paste0</span>(directory_figures, <span class="st">"ciona_proteome_normTo1.tsv"</span>)), <span class="at">col_names =</span> T, <span class="at">trim_ws =</span> T)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(ciona_prot.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 7,095
Columns: 13
$ protein_id_transcript &lt;chr&gt; "KY21.Chr1.10.v1.nonSL2-1", "KY21.Chr1.1000.v1.SL1-1", "KY21.Chr1.1001.v1.SL1-1", "KY21.Chr1.1002.v1.SL1-1", "KY21.Chr1.1004.v1.nonSL4-1", "KY21.Chr1.1005.v1.ND1-1", "KY21.Chr1.1006.v1.SL1-1", "KY21.Chr1.1009.v1.ND1-1", "KY21.Chr1.101.v1.SL1-1", "KY21.Chr1.1010.v1.nonSL1-1", "KY21.Chr1.1011.v1.SL1-1", "KY21.Chr1.1012.v1.SL1-1", "KY21.Chr1.1013.v1.ND1-1", "KY21.Chr1.1014.v2.ND2-1", "KY21.Chr1.1015.v1.SL1-1", "KY21.Chr1.1016.v1.SL1-1", "KY21.Chr1.1018.v1.nonSL1-1", "KY21.Chr1.1019.v1.SL1-1", "KY21.Chr1.102.v1.nonSL2-1", "KY21.Chr1.1020.v1.SL1-1", "KY21.Chr1.1022.v1.ND1-1", "KY21.Chr1.1024.v1.SL1-1", "KY21.Chr1.1027.v2.nonSL3-1", "KY21.Chr1.1029.v1.SL1-1", "KY21.Chr1.1031.v1.ND1-1", "KY21.Chr1.1035.v1.SL1-1", "KY21.Chr1.1040.v1.nonSL7-1", "KY21.Chr1.1041.v1.nonSL2-1", "KY21.Chr1.1042.v1.nonSL4-1", "KY21.Chr1.1045.v1.SL1-1", "KY21.Chr1.1046.v1.SL1-1", "KY21.Chr1.1050.v1.SL1-1", "KY21.Chr1.1051.v1.SL1-1", "KY21.Chr1.1055.v1.SL1-1", "KY21.Chr1.1056.v1.nonSL3-1…
$ protein_id_gene       &lt;chr&gt; "KY21.Chr1.10", "KY21.Chr1.1000", "KY21.Chr1.1001", "KY21.Chr1.1002", "KY21.Chr1.1004", "KY21.Chr1.1005", "KY21.Chr1.1006", "KY21.Chr1.1009", "KY21.Chr1.101", "KY21.Chr1.1010", "KY21.Chr1.1011", "KY21.Chr1.1012", "KY21.Chr1.1013", "KY21.Chr1.1014", "KY21.Chr1.1015", "KY21.Chr1.1016", "KY21.Chr1.1018", "KY21.Chr1.1019", "KY21.Chr1.102", "KY21.Chr1.1020", "KY21.Chr1.1022", "KY21.Chr1.1024", "KY21.Chr1.1027", "KY21.Chr1.1029", "KY21.Chr1.1031", "KY21.Chr1.1035", "KY21.Chr1.1040", "KY21.Chr1.1041", "KY21.Chr1.1042", "KY21.Chr1.1045", "KY21.Chr1.1046", "KY21.Chr1.1050", "KY21.Chr1.1051", "KY21.Chr1.1055", "KY21.Chr1.1056", "KY21.Chr1.1057", "KY21.Chr1.1058", "KY21.Chr1.1060", "KY21.Chr1.1061", "KY21.Chr1.1062", "KY21.Chr1.1063", "KY21.Chr1.1066", "KY21.Chr1.107", "KY21.Chr1.1072", "KY21.Chr1.1073", "KY21.Chr1.1077", "KY21.Chr1.1078", "KY21.Chr1.108", "KY21.Chr1.1085", "KY21.Chr1.1088", "KY21.Chr1.1089", "KY21.Chr1.109", "KY21.Chr1.1093", "KY21.Chr1.1095", "KY…
$ human_id              &lt;chr&gt; "TAF9", "PSMB7", "LRRC47", "TDRD1", "DNAJC2", "MYO18B", "NUAK1", "SUMF1", "NAA10", "GAS2", "SLC9A9", "SPEN", "SPEN", "XPO7", "RFK", "CPOX", "ABRAXAS2", "PPM1G", "DNAJB9", "MGME1", "CCDC15", "CEP85", "GNAI1", "UVRAG", "EPC1", "DDX58", "APIP", "GPR155", "SH3GLB1", "TXN", "RBM33", "GSTM3", "CYRIB", "PRIM2", "NSMCE1", "TACC1", "ZNF143", "SCYL3", "DDX39B", "KIFAP3", "NR6A1", "COG7", "WDR33", "RPL8", "ZSWIM8", "TTC21B", "GRB2", "CYB5D1", "DEPDC5", "SRPRB", "DCUN1D1", "ENKUR", "ACAP2", "AMBRA1", "COPB2", "GRAMD1A", "TFDP1", "VAMP3", "TCEA3", "COP1", "ERI1", "NPHP4", "BICD2", "PTPMT1", "MCM4", "NINL", "GNB1L", "PUS7L", "PRKAR2A", "HSP90B1", "AGL", "SYCE2", "TEKT2", "GMNN", "VPS28", "CTSC", "TTPAL", "C12orf4", "UVSSA", "CIB1", "XPNPEP1", "KIF3B", "ZNF830", "CYB5R2", "CASK", "RAB23", "DDX19B", "EXOC4", "TCEA1", "NR2C2AP", "MRRF", "BARD1", "MFAP1", "IRAK1", "DLD", "MRPL52", "RAB7A", "USP15", "MMAB", "SLC25A11", "GREB1", "MYT1L", "SNX33", "GCC1", "MED4", "NUP98", "B…
$ hgnc                  &lt;chr&gt; "HGNC:11542", "HGNC:9544", "HGNC:29207", "HGNC:11712", "HGNC:13192", "HGNC:18150", "HGNC:14311", "HGNC:20376", "HGNC:18704", "HGNC:4167", "HGNC:20653", "HGNC:17575", "HGNC:17575", "HGNC:14108", "HGNC:30324", "HGNC:2321", "HGNC:28975", "HGNC:9278", "HGNC:6968", "HGNC:16205", "HGNC:25798", "HGNC:25309", "HGNC:4384", "HGNC:12640", "HGNC:19876", "HGNC:19102", "HGNC:17581", "HGNC:22951", "HGNC:10833", "HGNC:12435", "HGNC:27223", "HGNC:4635", "HGNC:25216", "HGNC:9370", "HGNC:29897", "HGNC:11522", "HGNC:12928", "HGNC:19285", "HGNC:13917", "HGNC:17060", "HGNC:7985", "HGNC:18622", "HGNC:25651", "HGNC:10368", "HGNC:23528", "HGNC:25660", "HGNC:4566", "HGNC:26516", "HGNC:18423", "HGNC:24085", "HGNC:18184", "HGNC:28388", "HGNC:16469", "HGNC:25990", "HGNC:2232", "HGNC:29305", "HGNC:11749", "HGNC:12644", "HGNC:11615", "HGNC:17440", "HGNC:23994", "HGNC:19104", "HGNC:17208", "HGNC:26965", "HGNC:6947", "HGNC:29163", "HGNC:4397", "HGNC:25276", "HGNC:9391", "HGNC:12028", "H…
$ number_of_peptides    &lt;dbl&gt; 5, 13, 15, 6, 21, 9, 1, 12, 9, 1, 2, 24, 5, 25, 1, 11, 3, 11, 2, 2, 2, 10, 7, 2, 3, 8, 2, 1, 8, 4, 8, 17, 6, 14, 3, 13, 1, 4, 23, 3, 5, 13, 9, 34, 7, 11, 1, 2, 2, 17, 12, 1, 12, 2, 39, 5, 6, 2, 4, 5, 5, 4, 7, 4, 30, 8, 2, 17, 7, 142, 23, 1, 1, 7, 6, 14, 15, 5, 5, 4, 13, 9, 4, 7, 10, 3, 17, 10, 10, 1, 5, 1, 6, 6, 54, 4, 14, 32, 5, 19, 3, 1, 8, 10, 1, 5, 10, 4, 2, 2, 25, 2, 5, 64, 3, 5, 7, 10, 1, 1, 2, 5, 9, 26, 6, 22, 5, 8, 2, 1, 4, 27, 1, 1, 33, 19, 1, 13, 11, 1, 4, 4, 2, 15, 2, 20, 8, 24, 3, 2, 3, 4, 24, 15, 5, 3, 1, 1, 6, 1, 3, 24, 19, 9, 1, 26, 7, 8, 5, 1, 42, 11, 1, 13, 2, 4, 29, 22, 3, 7, 6, 2, 1, 13, 19, 3, 28, 8, 1, 4, 15, 9, 3, 25, 11, 7, 4, 14, 2, 9, 6, 7, 8, 10, 3, 5, 6, 1, 6, 1, 3, 1, 20, 16, 28, 4, 2, 3, 11, 1, 18, 2, 2, 3, 6, 6, 9, 2, 10, 8, 4, 2, 1, 1, 4, 4, 3, 7, 1, 5, 24, 1, 5, 2, 23, 3, 2, 15, 18, 17, 3, 15, 2, 1, 18, 2, 4, 7, 4, 13, 9, 10, 1, 21, 2, 15, 8, 2, 2, 10, 8, 3, 4, 29, 10, 4, 10, 38, 3, 3, 4, 3, 12, 12, 4, 53, 15, 3, 4, 1, 8, …
$ unfE                  &lt;dbl&gt; 1.298049e-01, 1.324528e-01, 1.127597e-01, 4.200519e-01, 1.134175e-01, 1.252194e-01, 2.032697e-01, 1.710183e-01, 1.044630e-01, 2.785118e-01, 8.699938e-02, 1.063616e-01, 1.302725e-01, 1.226167e-01, 1.333502e-01, 1.036764e-01, 9.848785e-02, 1.014728e-01, 3.841175e-02, 1.834680e-01, 5.557100e-02, 1.349891e-01, 1.034271e-01, 9.713907e-02, 1.179749e-01, 7.741159e-02, 7.770750e-02, 2.865548e-01, 1.394184e-01, 9.014801e-01, 5.926869e-02, 1.230879e-01, 9.073042e-02, 1.335176e-01, 1.312359e-01, 1.773455e-01, 1.276592e-01, 7.315110e-02, 8.445418e-02, 1.020476e-01, 1.649247e-02, 1.067782e-01, 1.031701e-01, 1.232679e-01, 4.209236e-02, 1.123042e-01, 8.366374e-01, 4.596535e-02, 1.413864e-01, 1.046384e-01, 1.159262e-01, 7.884029e-03, 8.040522e-02, 7.663698e-02, 1.102994e-01, 5.699762e-02, 1.145377e-01, 1.222085e-01, 3.739114e-02, 1.243875e-01, 1.223155e-01, 1.300714e-01, 1.174618e-01, 6.797665e-02, 1.264099e-01, 9.751044e-02, 1.925553e-01, 1.231327e-01, 9.484711e-02, 1.…
$ fertE                 &lt;dbl&gt; 0.1259673628, 0.1348872849, 0.1070208546, 0.3998787072, 0.1100268958, 0.1254946261, 0.1764491823, 0.1625296473, 0.1005591244, 0.3124981464, 0.0828260298, 0.1087763791, 0.1128248903, 0.1212120565, 0.1294932728, 0.1063921248, 0.1232894869, 0.0981483299, 0.0368863364, 0.1340376814, 0.0594995071, 0.1437564947, 0.1012506983, 0.0854047794, 0.1060651322, 0.0699248526, 0.0832902489, 0.0813895791, 0.1299898756, 0.0637744491, 0.0747610127, 0.1226230278, 0.0611645675, 0.1283378832, 0.1231017306, 0.1780246180, 0.1111774146, 0.0719104431, 0.0895344843, 0.1074715510, 0.0139331301, 0.1126086854, 0.1077540860, 0.1199514854, 0.0422573796, 0.1097822787, 0.0635442331, 0.0598222959, 0.1198976977, 0.1119800015, 0.1040114955, 0.1516362856, 0.0730188450, 0.0775572143, 0.1062961899, 0.0540386207, 0.1176930579, 0.1207845253, 0.0774454593, 0.1257390141, 0.1268401582, 0.1281410426, 0.1206433908, 0.0818736458, 0.1297553262, 0.1042516577, 0.1108786442, 0.1180802420, 0.1269517666, 0.…
$ cell16                &lt;dbl&gt; 0.1250456449, 0.1328101843, 0.1217596402, 0.0641194480, 0.1233892497, 0.1696345681, 0.0280065481, 0.1660856178, 0.1014044612, 0.0492090452, 0.0974244402, 0.1091126988, 0.1175400224, 0.1278746571, 0.1197704838, 0.1216388784, 0.1373451292, 0.1149117477, 0.0794878241, 0.1219411609, 0.0993291963, 0.1215333709, 0.1253880041, 0.1187471576, 0.1182247980, 0.1025463113, 0.0964897295, 0.0930370244, 0.1268794777, 0.0103578272, 0.1428441909, 0.1381293030, 0.1011210562, 0.1412464172, 0.1373088402, 0.1406442782, 0.1264197227, 0.0864787410, 0.1154138351, 0.1032854792, 0.0102171236, 0.1207738544, 0.1329623983, 0.1294206162, 0.0559302019, 0.1182662662, 0.0004799518, 0.0843884353, 0.1061593830, 0.1251092266, 0.1279873533, 0.0538486900, 0.0930475763, 0.0956709580, 0.1173102106, 0.0644276984, 0.1169321320, 0.1476430094, 0.0497483641, 0.1295388302, 0.1335797736, 0.1525073846, 0.1303249884, 0.1226622137, 0.1316943812, 0.1234315218, 0.2190476699, 0.1306975610, 0.1181505037, 0.…
$ iniG                  &lt;dbl&gt; 1.201025e-01, 1.146366e-01, 1.177441e-01, 1.547713e-02, 1.218467e-01, 1.474212e-01, 2.431692e-02, 9.674655e-02, 1.233771e-01, 4.386662e-02, 1.181640e-01, 1.178227e-01, 1.057373e-01, 1.272844e-01, 1.531695e-01, 1.195095e-01, 1.380446e-01, 1.311223e-01, 9.721245e-02, 9.179619e-02, 1.814810e-01, 6.419875e-02, 1.148672e-01, 1.187910e-01, 9.138235e-02, 1.237285e-01, 1.307980e-01, 4.581240e-02, 1.183398e-01, 9.455054e-04, 1.274477e-01, 1.149075e-01, 1.053072e-01, 1.377693e-01, 1.178903e-01, 1.151039e-01, 1.395133e-01, 1.196574e-01, 1.046477e-01, 1.175316e-01, 9.691588e-03, 1.371220e-01, 1.299788e-01, 1.199843e-01, 7.834961e-02, 1.035352e-01, 5.807113e-04, 1.337454e-01, 9.920607e-02, 1.298941e-01, 1.307009e-01, 1.459647e-01, 8.871721e-02, 1.431424e-01, 1.181252e-01, 1.865804e-02, 1.107338e-01, 1.202856e-01, 4.768982e-02, 1.107616e-01, 1.192751e-01, 1.614690e-01, 1.185387e-01, 1.790921e-01, 1.286794e-01, 1.580947e-01, 9.062976e-02, 9.979766e-02, 1.161346e-01, 1.…
$ latN                  &lt;dbl&gt; 1.249958e-01, 1.176930e-01, 1.225389e-01, 2.334494e-02, 1.288561e-01, 1.213391e-01, 1.353770e-01, 9.880142e-02, 1.310360e-01, 5.664927e-02, 1.374202e-01, 1.235143e-01, 1.288313e-01, 1.342176e-01, 1.268391e-01, 1.267021e-01, 1.239978e-01, 1.351063e-01, 1.402561e-01, 9.585023e-02, 2.124226e-01, 8.335679e-02, 1.101131e-01, 1.253734e-01, 1.305984e-01, 1.311726e-01, 1.197510e-01, 1.764569e-02, 1.215315e-01, 4.940914e-04, 1.506236e-01, 1.193120e-01, 1.270588e-01, 1.448363e-01, 1.332241e-01, 1.062659e-01, 1.346756e-01, 1.250050e-01, 1.282348e-01, 1.346201e-01, 2.841909e-01, 1.434135e-01, 1.265789e-01, 1.242013e-01, 1.472287e-01, 1.243490e-01, 5.255602e-04, 1.565178e-01, 9.022618e-02, 1.287399e-01, 1.266559e-01, 6.983314e-02, 1.225538e-01, 1.540599e-01, 1.230494e-01, 2.006765e-02, 1.340700e-01, 1.140707e-01, 3.747232e-02, 1.302608e-01, 1.321272e-01, 1.442147e-01, 1.120498e-01, 1.586591e-01, 1.318796e-01, 1.559365e-01, 7.219405e-02, 1.118971e-01, 1.142572e-01, 1.…
$ midTII                &lt;dbl&gt; 0.1077732493, 0.1183431058, 0.1298416904, 0.0248672053, 0.1355945589, 0.1160426877, 0.1456811696, 0.1036418491, 0.1337028314, 0.0570461601, 0.1366937320, 0.1364715547, 0.1288377428, 0.1284267381, 0.1200899658, 0.1320717241, 0.1336096975, 0.1320604291, 0.1880265698, 0.1428987075, 0.1617986810, 0.1106779343, 0.1234949652, 0.1510832931, 0.1378754043, 0.1443671826, 0.1233777760, 0.0712951320, 0.1154652964, 0.0008680424, 0.1470098516, 0.1131466747, 0.1380976013, 0.1181519524, 0.1158007347, 0.0952046470, 0.1282343305, 0.1563352621, 0.1403727657, 0.1335504432, 0.2280586429, 0.1329982158, 0.1354347087, 0.1261179185, 0.1973239848, 0.1417507407, 0.0002929388, 0.1647420625, 0.1147942878, 0.1372483475, 0.1228574088, 0.1052463644, 0.1555390226, 0.1789083710, 0.1404050126, 0.0693326818, 0.1315960484, 0.1063854656, 0.0758183433, 0.1255273590, 0.1216818575, 0.1121900324, 0.1006626954, 0.1607308671, 0.1264820198, 0.1593752228, 0.1028987043, 0.1110920667, 0.1343382230, 0.…
$ latTII                &lt;dbl&gt; 0.1323855417, 0.1281513763, 0.1408600114, 0.0300239069, 0.1314632254, 0.0978353289, 0.1428260352, 0.1068325396, 0.1543988512, 0.1198226208, 0.1680491776, 0.1528819631, 0.1364129942, 0.1176940707, 0.1062285554, 0.1453782815, 0.1196228233, 0.1461882747, 0.1851877048, 0.1131266179, 0.1335724529, 0.1536245883, 0.1469536031, 0.1508592978, 0.1387443682, 0.1650351834, 0.1915226408, 0.1288464160, 0.1228062631, 0.0020578708, 0.1662939088, 0.1322169302, 0.1446448857, 0.1055998584, 0.1186735515, 0.1033045166, 0.1057741735, 0.1632213358, 0.1698592292, 0.1520053098, 0.2063258368, 0.1218006127, 0.1312772641, 0.1277019046, 0.2667397474, 0.1503908711, 0.0415974944, 0.1784937759, 0.1570438987, 0.1238861960, 0.1314015102, 0.2243778449, 0.1804974007, 0.1473179842, 0.1323612046, 0.2453531835, 0.1324797239, 0.1240563168, 0.2335502275, 0.1205220858, 0.1211919003, 0.0938088059, 0.1455032273, 0.1187722384, 0.1198381711, 0.1202442661, 0.1245907415, 0.1501543622, 0.1372269508, 0.…
$ larva                 &lt;dbl&gt; 0.1339249741, 0.1210256914, 0.1474750562, 0.0222367440, 0.1354058320, 0.0970131460, 0.1440734634, 0.0943440902, 0.1510585525, 0.0823963035, 0.1724230172, 0.1450587562, 0.1395431677, 0.1206737802, 0.1110589490, 0.1446310143, 0.1256025887, 0.1409897847, 0.2345312541, 0.1168814599, 0.0963254769, 0.1878629887, 0.1745054139, 0.1526020050, 0.1591346739, 0.1858137758, 0.1770630576, 0.2754189458, 0.1255694383, 0.0200221148, 0.1317509598, 0.1365767000, 0.2318754915, 0.0905406482, 0.1227648170, 0.0841066657, 0.1265461499, 0.2042406962, 0.1674829964, 0.1494879759, 0.2310902780, 0.1245050561, 0.1328438229, 0.1293544904, 0.1700780469, 0.1396214249, 0.0563416675, 0.1763249112, 0.1712860526, 0.1385037780, 0.1404591979, 0.2412089287, 0.2062209235, 0.1267061709, 0.1521534767, 0.4711245047, 0.1419574824, 0.1445659451, 0.4408843193, 0.1332628136, 0.1229884698, 0.0775975991, 0.1548153468, 0.1102331479, 0.1052612427, 0.0811557235, 0.0872051517, 0.1551483014, 0.1580936807, 0.…</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vis_dat</span>(ciona_prot.df) <span class="co"># not all Proteins are mapped to Human orthologs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/descriptive%20stats-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vis_miss</span>(ciona_prot.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/descriptive%20stats-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ## loading annoatation from gff3 table  - TFs, signalling molecules, genes names and symbols etc.</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co"># annotation.gff3.df &lt;-</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># as_tibble(read_excel("/Users/mariossi/Desktop/Annotation/final/df_selex&amp;gff3.xlsx", sheet = "df_selex&amp;gff3", range = cell_cols("A:I"), col_names=TRUE))</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co"># # Merging protein dynamics with annotaion table</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co"># KY.TMTproC.df[,which(KY.TMTproC.df$Protein.Id.gene %in% annotation.gff3.df$KY.gene)]</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># KY.TMTproC.df &lt;- left_join(KY.TMTproC.df,annotation.df, by = c("Protein.Id.gene" = "KY.gene.model"),keep =T)</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="do">#### Summary univariate stats</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="fu">descr</span>(ciona_prot.df,</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">headings =</span> <span class="cn">FALSE</span>, <span class="co"># remove headings</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">stats =</span> <span class="st">"common"</span> <span class="co"># most common descriptive statistics</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
                   cell16     fertE      iniG     larva      latN    latTII    midTII
--------------- --------- --------- --------- --------- --------- --------- ---------
           Mean      0.11      0.10      0.11      0.17      0.12      0.15      0.13
        Std.Dev      0.04      0.05      0.04      0.11      0.04      0.06      0.04
            Min      0.00      0.00      0.00      0.00      0.00      0.00      0.00
         Median      0.12      0.10      0.12      0.14      0.13      0.14      0.13
            Max      0.91      0.61      0.78      0.97      0.67      0.70      0.57
        N.Valid   7095.00   7095.00   7095.00   7095.00   7095.00   7095.00   7095.00
      Pct.Valid    100.00    100.00    100.00    100.00    100.00    100.00    100.00

Table: Table continues below

 

                  number_of_peptides      unfE
--------------- -------------------- ---------
           Mean                 8.80      0.10
        Std.Dev                15.85      0.07
            Min                 1.00      0.00
         Median                 5.00      0.11
            Max               841.00      0.93
        N.Valid              7095.00   7095.00
      Pct.Valid               100.00    100.00</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">view</span>(<span class="fu">dfSummary</span>(ciona_prot.df[, <span class="dv">5</span><span class="sc">:</span><span class="dv">13</span>]))</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co"># check quantiles</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(protein_id_transcript, number_of_peptides) <span class="sc">%&gt;%</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">group =</span> <span class="st">"same"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># group_by(group) %&gt;%</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean =</span> <span class="fu">mean</span>(number_of_peptides),</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(number_of_peptides),</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">min =</span> <span class="fu">min</span>(number_of_peptides),</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">max =</span> <span class="fu">max</span>(number_of_peptides),</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">range =</span> <span class="fu">diff</span>(<span class="fu">range</span>(number_of_peptides, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)),</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">n_distinct =</span> <span class="fu">n_distinct</span>(number_of_peptides),</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">quantile =</span> <span class="fu">quantile</span>(number_of_peptides),</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  ) <span class="co"># |&gt; unnest_wider(quantile)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 7
   mean median   min   max range n_distinct quantile
  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;      &lt;int&gt;    &lt;dbl&gt;
1  8.80      5     1   841   840         96        1
2  8.80      5     1   841   840         96        2
3  8.80      5     1   841   840         96        5
4  8.80      5     1   841   840         96       11
5  8.80      5     1   841   840         96      841</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sum values per stage =&gt; higher in late developing</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>ciona_prot.df <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), sum))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 9
  number_of_peptides  unfE fertE cell16  iniG  latN midTII latTII larva
               &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;
1              62471  744.  695.   768.  775.  857.   934.  1095. 1227.</code></pre>
</div>
</div>
</section>
<section id="peptides-proteins-quantification" class="level3" data-number="1.3">
<h3 data-number="1.3" class="anchored" data-anchor-id="peptides-proteins-quantification"><span class="header-section-number">1.3</span> Peptides &amp; proteins quantification</h3>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### number of peptides per protein</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co"># descriptive statistics for peptides</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>ciona_prot.df <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">tot_peptides =</span> <span class="fu">sum</span>(number_of_peptides))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 1
  tot_peptides
         &lt;dbl&gt;
1        62471</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we identified 62471 peptides</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co"># % peptites distribution table</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>peptides_freqTable <span class="ot">&lt;-</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># drop_na(Protein.Id.transcript) %&gt;%</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(number_of_peptides)) <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(number_of_peptides) <span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count =</span> number_of_peptides) <span class="sc">%&gt;%</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="at">count =</span> <span class="fu">case_when</span>(count <span class="sc">&gt;=</span> <span class="dv">50</span> <span class="sc">~</span> <span class="st">"&gt;=50"</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">as.character</span>(count))) <span class="sc">%&gt;%</span> <span class="co"># sum protein count if peptides higher than 50</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> <span class="co"># mutate(value = sum(n), .groups = 'drop') %&gt;%</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(number_of_peptides <span class="sc">&lt;=</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">freq =</span> <span class="fu">prop.table</span>(n),</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">freq =</span> value <span class="sc">/</span> <span class="fu">sum</span>(value), <span class="co"># same way</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">rel.freq =</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> n <span class="sc">/</span> <span class="fu">sum</span>(n), <span class="dv">1</span>), <span class="st">"%"</span>)</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>peptides_freqTable</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 50 × 6
   number_of_peptides     n count value   freq rel.freq
                &lt;dbl&gt; &lt;int&gt; &lt;chr&gt; &lt;int&gt;  &lt;dbl&gt; &lt;chr&gt;   
 1                  1  1122 1      1122 0.158  16%     
 2                  2   909 2       909 0.128  13%     
 3                  3   627 3       627 0.0884 8.9%    
 4                  4   575 4       575 0.0810 8.2%    
 5                  5   494 5       494 0.0696 7.1%    
 6                  6   414 6       414 0.0584 5.9%    
 7                  7   372 7       372 0.0524 5.3%    
 8                  8   303 8       303 0.0427 4.3%    
 9                  9   247 9       247 0.0348 3.5%    
10                 10   249 10      249 0.0351 3.6%    
# ℹ 40 more rows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Plotting peptides distribution</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>plot_peptides_freqTable <span class="ot">&lt;-</span> peptides_freqTable <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ggplot(aes(reorder(as.factor(n),-n),freq)) +</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(number_of_peptides, freq)) <span class="sc">+</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_bar(stat="identity",position = "identity", alpha= .50, fill = "grey78") +</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">fill =</span> n), <span class="at">width =</span> <span class="fl">0.9</span>, <span class="at">show.legend =</span> F) <span class="sc">+</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fl">8.80</span>), <span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">size =</span> <span class="fl">0.7</span>) <span class="sc">+</span> <span class="co"># mean</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">5</span>), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="dv">3</span>, <span class="at">size =</span> <span class="fl">0.7</span>) <span class="sc">+</span> <span class="co"># median</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> value), <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>percent, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.001</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.17</span>)) <span class="sc">+</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">breaks =</span> peptides_freqTable<span class="sc">$</span>number_of_peptides,</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> peptides_freqTable<span class="sc">$</span>count,</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="dv">0</span>)</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span>  <span class="fu">wes_palette</span>(<span class="st">"Darjeeling1"</span>, <span class="dv">100</span>, <span class="at">type =</span> <span class="st">"continuous"</span>)) <span class="sc">+</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="co"># title= "Number of peptides per quantified protein",</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Number of peptides per quantified protein "</span>,</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Frequency"</span>,</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># subtitle = "Total peptides: 62.471 \nTotal protein: 7.095"</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">6.5</span>, <span class="at">y =</span> <span class="fl">0.15</span>, <span class="at">label =</span> <span class="st">"median"</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">angle =</span> <span class="sc">-</span><span class="dv">90</span>) <span class="sc">+</span></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">10</span>, <span class="at">y =</span> <span class="fl">0.15</span>, <span class="at">label =</span> <span class="st">"mean"</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">angle =</span> <span class="sc">-</span><span class="dv">90</span>) <span class="sc">+</span></span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">40</span>, <span class="at">y =</span> <span class="fl">0.16</span>, <span class="at">label =</span> <span class="st">"Total protein: 7.095"</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">40</span>, <span class="at">y =</span> <span class="fl">0.15</span>, <span class="at">label =</span> <span class="st">"Total unique protein: 7.057"</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">40</span>, <span class="at">y =</span> <span class="fl">0.14</span>, <span class="at">label =</span> <span class="st">"Total peptides: 62.471"</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">x.text.angle =</span> <span class="dv">45</span>)</span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>plot_peptides_freqTable</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/distribution%20Number%20of%20peptides%20per%20protein-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### number of times a peptide per protein</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="co"># number of peptides per quantified protein</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(protein_id_transcript) <span class="sc">%&gt;%</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(number_of_peptides)) <span class="sc">%&gt;%</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(number_of_peptides) <span class="sc">%&gt;%</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(number_of_peptides) <span class="sc">%&gt;%</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter(n &lt; 50) %&gt;%</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">number_of_peptides =</span> <span class="fu">as.factor</span>(number_of_peptides)) <span class="sc">%&gt;%</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_reorder</span>(number_of_peptides, <span class="sc">-</span>n), <span class="at">y =</span> n)) <span class="sc">+</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">fill =</span> n), <span class="at">width =</span> <span class="fl">0.9</span>, <span class="at">show.legend =</span> F) <span class="sc">+</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.02</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span>  <span class="fu">wes_palette</span>(<span class="st">"Darjeeling1"</span>, <span class="dv">100</span>, <span class="at">type =</span> <span class="st">"continuous"</span>)) <span class="sc">+</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="at">label=</span><span class="st">""</span>,</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>          <span class="at">subtitle =</span> <span class="st">"Peptides per protein"</span>) <span class="sc">+</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="fl">0.1</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.length =</span> <span class="fu">unit</span>(<span class="dv">0</span>, <span class="st">"cm"</span>)</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Peptides #"</span>,</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Protein #"</span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">x.text.angle =</span> <span class="dv">45</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/distribution%20Number%20of%20peptides%20per%20protein-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Circular plot - top peptides x protein</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>plot_peptides_per_protein <span class="ot">&lt;-</span> ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(number_of_peptides)) <span class="sc">%&gt;%</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(protein_id_transcript, human_id, number_of_peptides) <span class="sc">%&gt;%</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">nrow</span>(.))) <span class="sc">%&gt;%</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(id <span class="sc">&lt;</span> <span class="dv">25</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(<span class="fu">as.factor</span>(human_id), number_of_peptides), <span class="at">y =</span> number_of_peptides)) <span class="sc">+</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_col(aes(fill = reorder(as.factor(Protein.Id.gene),number_of_peptides))) +</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_fill_viridis_d(option = "A") +</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">fill =</span> id)) <span class="sc">+</span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">rev</span>(<span class="fu">wes_palette</span>(<span class="st">"Royal1"</span>, <span class="dv">25</span>, <span class="at">type =</span> <span class="st">"continuous"</span>))) <span class="sc">+</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">70</span>, <span class="dv">100</span>, <span class="dv">150</span>, <span class="dv">200</span>, <span class="dv">400</span>, <span class="dv">500</span>, <span class="dv">800</span>, <span class="dv">900</span>)) <span class="sc">+</span></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"# peptides per protein"</span>) <span class="sc">+</span></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">legend =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linewidth =</span> <span class="fl">0.4</span>, <span class="at">linetype =</span> <span class="dv">2</span>),</span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>)</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="at">start =</span> <span class="dv">0</span>)</span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>plot_peptides_per_protein</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/distribution%20Number%20of%20peptides%20per%20protein-3.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cairo_pdf</span>(<span class="st">"plot_peptides_freqTable.pdf"</span>, <span class="at">width=</span><span class="fl">8.5</span>, <span class="at">height=</span><span class="fl">5.6</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot_peptides_freqTable)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cairo_pdf</span>(<span class="st">"plot_peptides_per_protein.pdf"</span>, <span class="at">width=</span><span class="fl">8.5</span>, <span class="at">height=</span><span class="fl">5.6</span>)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot_peptides_per_protein)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mapping-isoforms" class="level3" data-number="1.4">
<h3 data-number="1.4" class="anchored" data-anchor-id="mapping-isoforms"><span class="header-section-number">1.4</span> Mapping isoforms</h3>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Plotting isoforms : checking multiple peptides matching same protein</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="fu">len_uniq</span>(ciona_prot.df<span class="sc">$</span>protein_id_gene) <span class="co"># Number of unique protein - 7057</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 7057</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>protein_isoform.list <span class="ot">&lt;-</span> <span class="fu">unique</span>(ciona_prot.df<span class="sc">$</span>protein_id_gene[<span class="fu">duplicated</span>(ciona_prot.df<span class="sc">$</span>protein_id_gene)]) <span class="co"># list of isoforms: 35 protein have multiple isoforms</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting isoforms dynamics</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co"># protein_isoform_only &lt;-  left_join(protein_isoform_only,annotation.df, by = c("Protein.Id.gene" = "KY.gene.model"),keep =T) # not needed</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>protein_isoform_only <span class="ot">&lt;-</span> ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_dupes</span>(protein_id_gene) <span class="sc">%&gt;%</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(dupe_count))</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a><span class="co"># unite("protein_id_gene", protein_id_gene, gene.name.KY, extra.info, sep= "-") %&gt;%</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="co"># mutate(protein_id_gene = gsub("-NA",replacement = "", protein_id_gene), protein_id_gene = gsub("KY21.",replacement = "", protein_id_gene))</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Summary with number of isoforms per protein</span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>protein_isoform_only <span class="sc">%&gt;%</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(protein_id_gene) <span class="sc">%&gt;%</span></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">dupe_count =</span> <span class="fu">length</span>(protein_id_gene)) <span class="sc">%&gt;%</span></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(dupe_count)) <span class="co"># |&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 35 × 2
   protein_id_gene dupe_count
   &lt;chr&gt;                &lt;int&gt;
 1 KY21.Chr10.215           4
 2 KY21.Chr5.818            3
 3 KY21.Chr1.1682           2
 4 KY21.Chr1.2263           2
 5 KY21.Chr1.326            2
 6 KY21.Chr1.670            2
 7 KY21.Chr10.1001          2
 8 KY21.Chr11.1071          2
 9 KY21.Chr11.568           2
10 KY21.Chr11.747           2
# ℹ 25 more rows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pull(dupe_count) %&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sum()</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the duplicates isoform matching different human genes</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>protein_isoform_duplicates.human_id <span class="ot">&lt;-</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  protein_isoform_only <span class="sc">%&gt;%</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(protein_id_gene) <span class="sc">%&gt;%</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">duplicated</span>(protein_id_gene) <span class="sc">&amp;</span> human_id <span class="sc">!=</span> <span class="fu">lag</span>(human_id)) <span class="sc">%&gt;%</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="co"># clean the name</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>protein_isoform_duplicates_all.human_id <span class="ot">&lt;-</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  protein_isoform_only <span class="sc">%&gt;%</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(protein_id_gene <span class="sc">%in%</span> protein_isoform_duplicates.human_id<span class="sc">$</span>protein_id_gene) <span class="sc">|&gt;</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(protein_id_gene) <span class="sc">%&gt;%</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">human_id =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(human_id), <span class="at">collapse =</span> <span class="st">" - "</span>)) <span class="sc">|&gt;</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">protein_id_gene_human_id =</span> <span class="fu">paste0</span>(protein_id_gene, <span class="st">" - "</span>, human_id))</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> protein_isoform_only <span class="sc">%&gt;%</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">7</span><span class="sc">:</span><span class="dv">14</span>, <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> <span class="fu">list</span>(<span class="at">stage =</span> as.factor), <span class="at">values_to =</span> <span class="st">"concentration"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hpf =</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(<span class="fu">recode</span>(stage, <span class="st">`</span><span class="at">unfE</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"0"</span>, <span class="st">`</span><span class="at">fertE</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"24"</span>, <span class="st">`</span><span class="at">cell16</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"141"</span>, <span class="st">`</span><span class="at">iniG</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"260"</span>, <span class="st">`</span><span class="at">latN</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"434"</span>, <span class="st">`</span><span class="at">midTII</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"632"</span>, <span class="st">`</span><span class="at">latTII</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"796"</span>, <span class="st">`</span><span class="at">larva</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"1038"</span>)))) <span class="sc">%&gt;%</span></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">protein_id_gene_human_id =</span> <span class="fu">paste0</span>(protein_id_gene, <span class="st">" - "</span>, human_id)) <span class="sc">|&gt;</span></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_isoform_duplicates_all.human_id, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"protein_id_gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">protein_id_gene_human_id.x =</span> <span class="fu">if_else</span>(<span class="sc">!</span><span class="fu">is.na</span>(human_id.y),</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>    protein_id_gene_human_id.y,</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>    protein_id_gene_human_id.x</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">|&gt;</span></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">protein_id_gene_human_id =</span> protein_id_gene_human_id.x) <span class="sc">|&gt;</span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># slice_head(n = 50) %&gt;%</span></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(hpf, concentration)) <span class="sc">+</span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(protein_id_gene <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">"free_y"</span>, ) <span class="sc">+</span></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">group =</span> protein_id_transcript, <span class="at">colour =</span> protein_id_gene)) <span class="sc">+</span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> protein_id_transcript, <span class="at">colour =</span> protein_id_gene)) <span class="sc">+</span></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> isoform_palette) <span class="sc">+</span></span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Protein expression over development"</span>, <span class="at">x =</span> <span class="st">"Time (hr)"</span>, <span class="at">y =</span> <span class="st">"Relative expression"</span>) <span class="sc">+</span></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Hours post fertilization"</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"0.4"</span>, <span class="st">"2.65"</span>, <span class="st">"4.5"</span>, <span class="st">"7.4"</span>, <span class="st">"10.9"</span>, <span class="st">"13.5"</span>, <span class="st">"17.5"</span>), <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">24</span>, <span class="dv">141</span>, <span class="dv">260</span>, <span class="dv">434</span>, <span class="dv">632</span>, <span class="dv">796</span>, <span class="dv">1038</span>)) <span class="sc">+</span></span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>), <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">colour =</span> <span class="st">"black"</span>), <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">size =</span> <span class="dv">8</span>))</span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a>p_fin <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">facet_rep_wrap</span>(<span class="sc">~</span>protein_id_gene_human_id, <span class="co">#scales = "free_y", </span></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a>                   <span class="at">repeat.tick.labels =</span> <span class="st">"left"</span>) <span class="sc">&amp;</span> </span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">legend =</span> <span class="st">"none"</span>, <span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">x.text.angle =</span> <span class="dv">45</span>)</span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a>p_fin</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/isoforms-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cairo_pdf</span>(<span class="st">"plot_isofomsDynamics.pdf"</span>, <span class="at">width=</span><span class="dv">12</span>, <span class="at">height=</span><span class="dv">8</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p_fin)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="top-and-bottom-proteins-expressed-by-stage" class="level3" data-number="1.5">
<h3 data-number="1.5" class="anchored" data-anchor-id="top-and-bottom-proteins-expressed-by-stage"><span class="header-section-number">1.5</span> Top and bottom proteins expressed by stage</h3>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Top and bottom proteins expressed by stage</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co"># top 10 most abundant entries for each stage.</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>top_entries <span class="ot">&lt;-</span> ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>, <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> <span class="fu">list</span>(<span class="at">stage =</span> as.factor), <span class="at">values_to =</span> <span class="st">"concentration"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">entry =</span> <span class="fu">coalesce</span>(human_id, protein_id_gene)) <span class="sc">%&gt;%</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(stage, <span class="fu">desc</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="at">order_by =</span> concentration, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">with_ties =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>bottom_entries <span class="ot">&lt;-</span> ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>, <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> <span class="fu">list</span>(<span class="at">stage =</span> as.factor), <span class="at">values_to =</span> <span class="st">"concentration"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">entry =</span> <span class="fu">coalesce</span>(human_id, protein_id_gene)) <span class="sc">%&gt;%</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(stage, concentration) <span class="sc">%&gt;%</span></span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span></span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_min</span>(<span class="at">order_by =</span> concentration, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">with_ties =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>top_entries <span class="sc">%&gt;%</span></span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> ciona_stages)) <span class="sc">%&gt;%</span></span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(entry, concentration), <span class="at">y =</span> concentration, <span class="at">fill =</span> stage)) <span class="sc">+</span></span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"dodge"</span>, <span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> stage)) <span class="sc">+</span></span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>stage, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Top 10 most abundant"</span>, <span class="at">x =</span> <span class="st">"Relative abundance"</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a>  bottom_entries <span class="sc">%&gt;%</span></span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> ciona_stages)) <span class="sc">%&gt;%</span></span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(entry, concentration), <span class="at">y =</span> concentration, <span class="at">fill =</span> stage)) <span class="sc">+</span></span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"dodge"</span>, <span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> stage)) <span class="sc">+</span></span>
<span id="cb31-33"><a href="#cb31-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Top 10 less abundant"</span>, <span class="at">y =</span> <span class="st">"Relative abundance"</span>, <span class="at">x =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb31-34"><a href="#cb31-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>stage, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb31-35"><a href="#cb31-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">&amp;</span> <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/top&amp;bottom%20protein%20per%20stage-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>), <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> as.factor) <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter(str_detect(stage, "_byRowCol")) %&gt;%</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(stage = str_remove(stage, "_byRowCol")) %&gt;%</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> ciona_stages), <span class="at">y =</span> value, <span class="at">fill =</span> stage, <span class="at">color =</span> stage)) <span class="sc">+</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> protein_id_transcript)) <span class="sc">+</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_line(aes(group = protein_id_transcript)) +</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>(<span class="fu">aes</span>(<span class="at">group =</span> protein_id_transcript)) <span class="sc">+</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">setNames</span>(ciona_stages_palette, ciona_stages), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">setNames</span>(ciona_stages_palette, ciona_stages), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Developmental stages"</span>, <span class="at">y =</span> <span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">legend =</span> <span class="st">"none"</span>, <span class="at">base_family =</span> <span class="st">"Avenir"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/top&amp;bottom%20protein%20per%20stage-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Top 30 changing per stage</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>stage_pairs <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"unfE"</span>, <span class="st">"fertE"</span>),</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"fertE"</span>, <span class="st">"cell16"</span>),</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"cell16"</span>, <span class="st">"iniG"</span>),</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"iniG"</span>, <span class="st">"latN"</span>),</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"latN"</span>, <span class="st">"midTII"</span>),</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"midTII"</span>, <span class="st">"latTII"</span>),</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"latTII"</span>, <span class="st">"larva"</span>)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>plots_up <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pair <span class="cf">in</span> stage_pairs) {</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>  selected_stages <span class="ot">&lt;-</span> pair</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>  df_filtered <span class="ot">&lt;-</span> ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter(get(paste0(selected_stages[2], "_byRowCol")) &gt; get(paste0(selected_stages[1], "_byRowCol"))) %&gt;%</span></span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">get</span>(<span class="fu">paste0</span>(selected_stages[<span class="dv">2</span>])) <span class="sc">&gt;</span> <span class="fu">get</span>(<span class="fu">paste0</span>(selected_stages[<span class="dv">1</span>]))) <span class="sc">%&gt;%</span></span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>), <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> as.factor) <span class="sc">%&gt;%</span></span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter(str_detect(stage, "_byRowCol")) %&gt;%</span></span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># mutate(stage = str_remove(stage, "_byRowCol")) %&gt;%</span></span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter(str_detect(stage)) %&gt;%</span></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># mutate(stage = str_remove(stage)) %&gt;%</span></span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(stage <span class="sc">%in%</span> selected_stages) <span class="sc">%&gt;%</span></span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(protein_id_transcript) <span class="sc">%&gt;%</span></span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">reframe</span>(</span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a>      <span class="at">protein_id_transcript =</span> protein_id_transcript, <span class="at">human_id =</span> human_id, <span class="at">stage =</span> stage, <span class="at">value =</span> value,</span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a>      <span class="at">diff =</span> value[stage <span class="sc">==</span> selected_stages[<span class="dv">2</span>]] <span class="sc">-</span> value[stage <span class="sc">==</span> selected_stages[<span class="dv">1</span>]]</span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>diff) <span class="sc">|&gt;</span></span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">60</span>)</span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a>  plot_title <span class="ot">&lt;-</span> <span class="fu">paste</span>(selected_stages[<span class="dv">1</span>], selected_stages[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">" - "</span>)</span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_filtered, <span class="fu">aes</span>(</span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> ciona_stages), <span class="at">y =</span> value,</span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> protein_id_transcript, <span class="at">color =</span> stage, <span class="at">label =</span> human_id</span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_path</span>() <span class="sc">+</span></span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="fu">setNames</span>(ciona_stages_palette, ciona_stages), <span class="at">selected_stages =</span> <span class="fu">c</span>(<span class="st">"#009392FF"</span>, <span class="st">"#39B185FF"</span>, <span class="st">"#9CCB86FF"</span>, <span class="st">"#E9E29CFF"</span>, <span class="st">"#EEB479FF"</span>, <span class="st">"#E88471FF"</span>, <span class="st">"#CF597EFF"</span>)), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb33-44"><a href="#cb33-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb33-45"><a href="#cb33-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(plot_title) <span class="sc">+</span></span>
<span id="cb33-46"><a href="#cb33-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_pubr</span>(<span class="at">legend =</span> <span class="st">"none"</span>, <span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb33-47"><a href="#cb33-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(</span>
<span id="cb33-48"><a href="#cb33-48" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> <span class="cf">function</span>(df) {</span>
<span id="cb33-49"><a href="#cb33-49" aria-hidden="true" tabindex="-1"></a>        df <span class="sc">%&gt;%</span></span>
<span id="cb33-50"><a href="#cb33-50" aria-hidden="true" tabindex="-1"></a>          <span class="co"># group_by(stage) %&gt;%</span></span>
<span id="cb33-51"><a href="#cb33-51" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">human_id =</span> <span class="fu">ifelse</span>(human_id <span class="sc">==</span> <span class="st">""</span>, protein_id_transcript, human_id)) <span class="sc">|&gt;</span></span>
<span id="cb33-52"><a href="#cb33-52" aria-hidden="true" tabindex="-1"></a>          <span class="fu">arrange</span>(<span class="sc">-</span>value) <span class="sc">%&gt;%</span></span>
<span id="cb33-53"><a href="#cb33-53" aria-hidden="true" tabindex="-1"></a>          <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">30</span>) <span class="sc">|&gt;</span></span>
<span id="cb33-54"><a href="#cb33-54" aria-hidden="true" tabindex="-1"></a>          <span class="fu">distinct</span>(human_id, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-55"><a href="#cb33-55" aria-hidden="true" tabindex="-1"></a>      }, <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb33-56"><a href="#cb33-56" aria-hidden="true" tabindex="-1"></a>      <span class="at">color =</span> <span class="st">"grey"</span>,</span>
<span id="cb33-57"><a href="#cb33-57" aria-hidden="true" tabindex="-1"></a>      <span class="at">box.padding =</span> <span class="fl">0.2</span>,</span>
<span id="cb33-58"><a href="#cb33-58" aria-hidden="true" tabindex="-1"></a>      <span class="at">nudge_x =</span> <span class="dv">1</span>,</span>
<span id="cb33-59"><a href="#cb33-59" aria-hidden="true" tabindex="-1"></a>      <span class="at">direction =</span> <span class="st">"y"</span>,</span>
<span id="cb33-60"><a href="#cb33-60" aria-hidden="true" tabindex="-1"></a>      <span class="at">verbose =</span> F,</span>
<span id="cb33-61"><a href="#cb33-61" aria-hidden="true" tabindex="-1"></a>      <span class="co"># hjust = "left",</span></span>
<span id="cb33-62"><a href="#cb33-62" aria-hidden="true" tabindex="-1"></a>      <span class="at">max.overlaps =</span> <span class="cn">Inf</span>,</span>
<span id="cb33-63"><a href="#cb33-63" aria-hidden="true" tabindex="-1"></a>      <span class="at">max.time =</span> <span class="dv">1</span>,</span>
<span id="cb33-64"><a href="#cb33-64" aria-hidden="true" tabindex="-1"></a>      <span class="at">max.iter =</span> <span class="fl">1e5</span>, <span class="at">segment.curvature =</span> <span class="sc">-</span><span class="fl">0.1</span>,</span>
<span id="cb33-65"><a href="#cb33-65" aria-hidden="true" tabindex="-1"></a>      <span class="at">segment.ncp =</span> <span class="dv">3</span>,</span>
<span id="cb33-66"><a href="#cb33-66" aria-hidden="true" tabindex="-1"></a>      <span class="at">segment.angle =</span> <span class="dv">20</span></span>
<span id="cb33-67"><a href="#cb33-67" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb33-68"><a href="#cb33-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-69"><a href="#cb33-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-70"><a href="#cb33-70" aria-hidden="true" tabindex="-1"></a>  plots_up[[plot_title]] <span class="ot">&lt;-</span> plot</span>
<span id="cb33-71"><a href="#cb33-71" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-72"><a href="#cb33-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-73"><a href="#cb33-73" aria-hidden="true" tabindex="-1"></a><span class="co"># combined_plots_up &lt;- wrap_plots(plotlist = plots_up, nrow = 1, align = "v")</span></span>
<span id="cb33-74"><a href="#cb33-74" aria-hidden="true" tabindex="-1"></a><span class="co"># combined_plots_up</span></span>
<span id="cb33-75"><a href="#cb33-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-76"><a href="#cb33-76" aria-hidden="true" tabindex="-1"></a>top_plot_2nd <span class="ot">&lt;-</span> plots_up<span class="sc">$</span><span class="st">`</span><span class="at">fertE - cell16</span><span class="st">`</span> <span class="sc">+</span> plots_up<span class="sc">$</span><span class="st">`</span><span class="at">cell16 - iniG</span><span class="st">`</span> <span class="sc">+</span> plots_up<span class="sc">$</span><span class="st">`</span><span class="at">iniG - latN</span><span class="st">`</span> <span class="sc">+</span> plots_up<span class="sc">$</span><span class="st">`</span><span class="at">latN - midTII</span><span class="st">`</span> <span class="sc">+</span> plots_up<span class="sc">$</span><span class="st">`</span><span class="at">midTII - latTII</span><span class="st">`</span> <span class="sc">+</span> plots_up<span class="sc">$</span><span class="st">`</span><span class="at">latTII - larva</span><span class="st">`</span> <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.line.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb33-77"><a href="#cb33-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-78"><a href="#cb33-78" aria-hidden="true" tabindex="-1"></a>plot_combined_up <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(plots_up<span class="sc">$</span><span class="st">`</span><span class="at">unfE - fertE</span><span class="st">`</span>, top_plot_2nd, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">widths =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="dv">5</span>))</span>
<span id="cb33-79"><a href="#cb33-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-80"><a href="#cb33-80" aria-hidden="true" tabindex="-1"></a>plots_down <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb33-81"><a href="#cb33-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-82"><a href="#cb33-82" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pair <span class="cf">in</span> stage_pairs) {</span>
<span id="cb33-83"><a href="#cb33-83" aria-hidden="true" tabindex="-1"></a>  selected_stages <span class="ot">&lt;-</span> pair</span>
<span id="cb33-84"><a href="#cb33-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-85"><a href="#cb33-85" aria-hidden="true" tabindex="-1"></a>  df_filtered <span class="ot">&lt;-</span> ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb33-86"><a href="#cb33-86" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter(get(paste0(selected_stages[1], "_byRowCol")) &gt; get(paste0(selected_stages[2], "_byRowCol"))) %&gt;%</span></span>
<span id="cb33-87"><a href="#cb33-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">get</span>(selected_stages[<span class="dv">1</span>]) <span class="sc">&gt;</span> <span class="fu">get</span>(selected_stages[<span class="dv">2</span>])) <span class="sc">%&gt;%</span></span>
<span id="cb33-88"><a href="#cb33-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>), <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> as.factor) <span class="sc">%&gt;%</span></span>
<span id="cb33-89"><a href="#cb33-89" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(stage <span class="sc">%in%</span> selected_stages) <span class="sc">%&gt;%</span></span>
<span id="cb33-90"><a href="#cb33-90" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(protein_id_transcript) <span class="sc">%&gt;%</span></span>
<span id="cb33-91"><a href="#cb33-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">reframe</span>(</span>
<span id="cb33-92"><a href="#cb33-92" aria-hidden="true" tabindex="-1"></a>      <span class="at">protein_id_transcript =</span> protein_id_transcript, <span class="at">human_id =</span> human_id, <span class="at">stage =</span> stage, <span class="at">value =</span> value,</span>
<span id="cb33-93"><a href="#cb33-93" aria-hidden="true" tabindex="-1"></a>      <span class="at">diff =</span> value[stage <span class="sc">==</span> selected_stages[<span class="dv">1</span>]] <span class="sc">-</span> value[stage <span class="sc">==</span> selected_stages[<span class="dv">2</span>]]</span>
<span id="cb33-94"><a href="#cb33-94" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb33-95"><a href="#cb33-95" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>diff) <span class="sc">|&gt;</span></span>
<span id="cb33-96"><a href="#cb33-96" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">60</span>)</span>
<span id="cb33-97"><a href="#cb33-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-98"><a href="#cb33-98" aria-hidden="true" tabindex="-1"></a>  plot_title <span class="ot">&lt;-</span> <span class="fu">paste</span>(selected_stages[<span class="dv">1</span>], selected_stages[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">" - "</span>)</span>
<span id="cb33-99"><a href="#cb33-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-100"><a href="#cb33-100" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_filtered, <span class="fu">aes</span>(</span>
<span id="cb33-101"><a href="#cb33-101" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> ciona_stages), <span class="at">y =</span> value,</span>
<span id="cb33-102"><a href="#cb33-102" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> protein_id_transcript, <span class="at">color =</span> stage, <span class="at">label =</span> human_id</span>
<span id="cb33-103"><a href="#cb33-103" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb33-104"><a href="#cb33-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_path</span>() <span class="sc">+</span></span>
<span id="cb33-105"><a href="#cb33-105" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb33-106"><a href="#cb33-106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb33-107"><a href="#cb33-107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="fu">setNames</span>(ciona_stages_palette, ciona_stages), <span class="at">selected_stages =</span> <span class="fu">c</span>(<span class="st">"#009392FF"</span>, <span class="st">"#39B185FF"</span>, <span class="st">"#9CCB86FF"</span>, <span class="st">"#E9E29CFF"</span>, <span class="st">"#EEB479FF"</span>, <span class="st">"#E88471FF"</span>, <span class="st">"#CF597EFF"</span>)), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb33-108"><a href="#cb33-108" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb33-109"><a href="#cb33-109" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(plot_title) <span class="sc">+</span></span>
<span id="cb33-110"><a href="#cb33-110" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_pubr</span>(<span class="at">legend =</span> <span class="st">"none"</span>, <span class="at">base_family =</span> <span class="st">"Avenir"</span>) <span class="sc">+</span></span>
<span id="cb33-111"><a href="#cb33-111" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(</span>
<span id="cb33-112"><a href="#cb33-112" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> <span class="cf">function</span>(df) {</span>
<span id="cb33-113"><a href="#cb33-113" aria-hidden="true" tabindex="-1"></a>        df <span class="sc">%&gt;%</span></span>
<span id="cb33-114"><a href="#cb33-114" aria-hidden="true" tabindex="-1"></a>          <span class="co"># group_by(stage) %&gt;%</span></span>
<span id="cb33-115"><a href="#cb33-115" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">human_id =</span> <span class="fu">ifelse</span>(human_id <span class="sc">==</span> <span class="st">""</span>, protein_id_transcript, human_id)) <span class="sc">|&gt;</span></span>
<span id="cb33-116"><a href="#cb33-116" aria-hidden="true" tabindex="-1"></a>          <span class="fu">arrange</span>(diff) <span class="sc">%&gt;%</span></span>
<span id="cb33-117"><a href="#cb33-117" aria-hidden="true" tabindex="-1"></a>          <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">30</span>) <span class="sc">|&gt;</span></span>
<span id="cb33-118"><a href="#cb33-118" aria-hidden="true" tabindex="-1"></a>          <span class="fu">distinct</span>(human_id, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-119"><a href="#cb33-119" aria-hidden="true" tabindex="-1"></a>      }, <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb33-120"><a href="#cb33-120" aria-hidden="true" tabindex="-1"></a>      <span class="at">box.padding =</span> <span class="fl">0.2</span>,</span>
<span id="cb33-121"><a href="#cb33-121" aria-hidden="true" tabindex="-1"></a>      <span class="at">nudge_x =</span> <span class="dv">1</span>,</span>
<span id="cb33-122"><a href="#cb33-122" aria-hidden="true" tabindex="-1"></a>      <span class="at">direction =</span> <span class="st">"y"</span>,</span>
<span id="cb33-123"><a href="#cb33-123" aria-hidden="true" tabindex="-1"></a>      <span class="at">verbose =</span> F,</span>
<span id="cb33-124"><a href="#cb33-124" aria-hidden="true" tabindex="-1"></a>      <span class="co"># hjust = "left",</span></span>
<span id="cb33-125"><a href="#cb33-125" aria-hidden="true" tabindex="-1"></a>      <span class="at">max.overlaps =</span> <span class="cn">Inf</span>,</span>
<span id="cb33-126"><a href="#cb33-126" aria-hidden="true" tabindex="-1"></a>      <span class="at">color =</span> <span class="st">"grey"</span>,</span>
<span id="cb33-127"><a href="#cb33-127" aria-hidden="true" tabindex="-1"></a>      <span class="at">max.time =</span> <span class="dv">1</span>,</span>
<span id="cb33-128"><a href="#cb33-128" aria-hidden="true" tabindex="-1"></a>      <span class="at">max.iter =</span> <span class="fl">1e5</span>,</span>
<span id="cb33-129"><a href="#cb33-129" aria-hidden="true" tabindex="-1"></a>      <span class="at">segment.curvature =</span> <span class="sc">-</span><span class="fl">0.1</span></span>
<span id="cb33-130"><a href="#cb33-130" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb33-131"><a href="#cb33-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-132"><a href="#cb33-132" aria-hidden="true" tabindex="-1"></a>  plots_down[[plot_title]] <span class="ot">&lt;-</span> plot</span>
<span id="cb33-133"><a href="#cb33-133" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-134"><a href="#cb33-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-135"><a href="#cb33-135" aria-hidden="true" tabindex="-1"></a><span class="co"># combined_plots_down &lt;- wrap_plots(plotlist = plots_down, nrow = 1, align = "v")</span></span>
<span id="cb33-136"><a href="#cb33-136" aria-hidden="true" tabindex="-1"></a><span class="co"># combined_plots_down</span></span>
<span id="cb33-137"><a href="#cb33-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-138"><a href="#cb33-138" aria-hidden="true" tabindex="-1"></a>bottom_plot_2nd <span class="ot">&lt;-</span> plots_down<span class="sc">$</span><span class="st">`</span><span class="at">fertE - cell16</span><span class="st">`</span> <span class="sc">+</span> plots_down<span class="sc">$</span><span class="st">`</span><span class="at">cell16 - iniG</span><span class="st">`</span> <span class="sc">+</span> plots_down<span class="sc">$</span><span class="st">`</span><span class="at">iniG - latN</span><span class="st">`</span> <span class="sc">+</span> plots_down<span class="sc">$</span><span class="st">`</span><span class="at">latN - midTII</span><span class="st">`</span> <span class="sc">+</span> plots_down<span class="sc">$</span><span class="st">`</span><span class="at">midTII - latTII</span><span class="st">`</span> <span class="sc">+</span> plots_down<span class="sc">$</span><span class="st">`</span><span class="at">latTII - larva</span><span class="st">`</span> <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.line.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb33-139"><a href="#cb33-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-140"><a href="#cb33-140" aria-hidden="true" tabindex="-1"></a>plot_combined_down <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(plots_down<span class="sc">$</span><span class="st">`</span><span class="at">unfE - fertE</span><span class="st">`</span>, bottom_plot_2nd, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">widths =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="dv">5</span>))</span>
<span id="cb33-141"><a href="#cb33-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-142"><a href="#cb33-142" aria-hidden="true" tabindex="-1"></a>plot_TopBottom_protein <span class="ot">&lt;-</span>  plot_combined_up <span class="sc">/</span> plot_combined_down</span>
<span id="cb33-143"><a href="#cb33-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-144"><a href="#cb33-144" aria-hidden="true" tabindex="-1"></a>plot_TopBottom_protein</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/top&amp;bottom%20protein%20per%20stage-3.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cairo_pdf</span>(<span class="st">"plot_TopBottom_protein.pdf"</span>, <span class="at">width=</span><span class="fl">8.5</span>, <span class="at">height=</span><span class="fl">5.6</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot_TopBottom_protein)</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="protein-distribution" class="level3" data-number="1.6">
<h3 data-number="1.6" class="anchored" data-anchor-id="protein-distribution"><span class="header-section-number">1.6</span> Protein distribution</h3>
<ul>
<li>calculation of quantile x stage</li>
<li>Below 10% from max value each protein: we consider the protein not being expressed</li>
</ul>
<section id="boxplots-density" class="level4" data-number="1.6.1">
<h4 data-number="1.6.1" class="anchored" data-anchor-id="boxplots-density"><span class="header-section-number">1.6.1</span> Boxplots &amp; density</h4>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Quantile summary</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>, <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> <span class="fu">list</span>(<span class="at">stage =</span> as.factor), <span class="at">values_to =</span> <span class="st">"concentration"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">min =</span> <span class="fu">quantile</span>(concentration, <span class="at">probs =</span> <span class="dv">0</span>),</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">qnt_10 =</span> <span class="fu">quantile</span>(concentration, <span class="at">probs =</span> <span class="fl">0.1</span>),</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">qnt_25 =</span> <span class="fu">quantile</span>(concentration, <span class="at">probs =</span> <span class="fl">0.25</span>),</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">qnt_50 =</span> <span class="fu">quantile</span>(concentration, <span class="at">probs =</span> <span class="fl">0.5</span>),</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">qnt_75 =</span> <span class="fu">quantile</span>(concentration, <span class="at">probs =</span> <span class="fl">0.75</span>),</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">qnt_90 =</span> <span class="fu">quantile</span>(concentration, <span class="at">probs =</span> <span class="fl">0.9</span>),</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">max =</span> <span class="fu">quantile</span>(concentration, <span class="at">probs =</span> <span class="dv">1</span>),</span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean =</span> <span class="fu">mean</span>(concentration),</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(concentration),</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">sd =</span> <span class="fu">sd</span>(concentration)</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(min, qnt_10, qnt_25, qnt_50, qnt_75, qnt_90, max, mean, median, sd)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 11
# Groups:   stage [8]
  stage          min qnt_10 qnt_25 qnt_50 qnt_75 qnt_90   max   mean median     sd
  &lt;fct&gt;        &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
1 unfE   0.000000524 0.0276 0.0761  0.108  0.125  0.145 0.931 0.105   0.108 0.0740
2 fertE  0.000000408 0.0286 0.0741  0.104  0.121  0.139 0.611 0.0979  0.104 0.0481
3 cell16 0.00000135  0.0449 0.0928  0.117  0.130  0.146 0.914 0.108   0.117 0.0439
4 iniG   0.000000527 0.0477 0.100   0.117  0.129  0.145 0.780 0.109   0.117 0.0425
5 latN   0.000000299 0.0752 0.112   0.125  0.138  0.155 0.670 0.121   0.125 0.0397
6 midTII 0.00000795  0.0980 0.116   0.129  0.146  0.171 0.573 0.132   0.129 0.0402
7 latTII 0.0000489   0.106  0.125   0.141  0.168  0.224 0.703 0.154   0.141 0.0605
8 larva  0.0000548   0.0983 0.123   0.143  0.180  0.279 0.971 0.173   0.143 0.107 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Protein distribution</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>, <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">names_transform =</span> <span class="fu">list</span>(<span class="at">stage =</span> as.factor), <span class="at">values_to =</span> <span class="st">"concentration"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> <span class="fu">rev</span>(ciona_stages))) <span class="sc">%&gt;%</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> stage, <span class="at">y =</span> concentration)) <span class="sc">+</span></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>  ggdist<span class="sc">::</span><span class="fu">stat_halfeye</span>(<span class="fu">aes</span>(<span class="at">colour =</span> stage, <span class="at">fill =</span> <span class="fu">after_scale</span>(<span class="fu">lighten</span>(colour, .<span class="dv">5</span>))), <span class="at">adjust =</span> .<span class="dv">5</span>, <span class="at">scale =</span> <span class="fl">0.7</span>, <span class="at">width =</span> .<span class="dv">75</span>, <span class="at">.width =</span> <span class="dv">0</span>, <span class="at">justification =</span> <span class="sc">-</span>.<span class="dv">4</span>, <span class="at">point_colour =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> stage, <span class="at">colour =</span> <span class="fu">after_scale</span>(<span class="fu">darken</span>(colour, .<span class="dv">1</span>, <span class="at">space =</span> <span class="st">"HLS"</span>))), <span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">stroke =</span> .<span class="dv">4</span>, <span class="at">size =</span> <span class="fl">1.5</span>, <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">seed =</span> <span class="dv">1</span>, <span class="at">width =</span> .<span class="dv">12</span>)) <span class="sc">+</span></span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">colour =</span> stage, <span class="at">colour =</span> <span class="fu">after_scale</span>(<span class="fu">darken</span>(colour, .<span class="dv">1</span>, <span class="at">space =</span> <span class="st">"HLS"</span>)), <span class="at">fill =</span> <span class="fu">after_scale</span>(<span class="fu">desaturate</span>(<span class="fu">lighten</span>(colour, .<span class="dv">8</span>), .<span class="dv">4</span>))), <span class="at">width =</span> .<span class="dv">4</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">geom =</span> <span class="st">"point"</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">fun =</span> <span class="st">"mean"</span>, <span class="fu">aes</span>(<span class="at">colour =</span> stage, <span class="at">colour =</span> <span class="fu">after_scale</span>(<span class="fu">darken</span>(colour, .<span class="dv">3</span>, <span class="at">space =</span> <span class="st">"HLS"</span>)))) <span class="sc">+</span></span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">rev</span>(ciona_stages_palette), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_fill_manual(values = rev(ciona_stages_palette), guide = "none") +</span></span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="at">by =</span> <span class="dv">1</span>), <span class="at">expand =</span> <span class="fu">c</span>(.<span class="dv">01</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb37-22"><a href="#cb37-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb37-23"><a href="#cb37-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb37-24"><a href="#cb37-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb37-25"><a href="#cb37-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">darken</span>(ciona_stages_palette, .<span class="dv">1</span>, <span class="at">space =</span> <span class="st">"HLS"</span>)), <span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb37-26"><a href="#cb37-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">10</span>), <span class="at">size =</span> <span class="dv">16</span>)</span>
<span id="cb37-27"><a href="#cb37-27" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/boxplots%20&amp;%20density-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="distribution-per-chromosome" class="level4" data-number="1.6.2">
<h4 data-number="1.6.2" class="anchored" data-anchor-id="distribution-per-chromosome"><span class="header-section-number">1.6.2</span> Distribution per chromosome</h4>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Distrubtion of proteins by chromosome</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co"># total genes per chromosome</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>genes.tot <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">"/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/a.reference_annotation/annotation_ciona.KY21.tsv"</span>)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>genesTot_chr <span class="ot">&lt;-</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  genes.tot <span class="sc">%&gt;%</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">|&gt;</span></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">GeneID_chr =</span> <span class="fu">str_remove</span>(GeneID, <span class="st">"</span><span class="sc">\\</span><span class="st">.[^.]+$"</span>), <span class="co"># Extract everything after last .</span></span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">gsub</span>(<span class="st">"*KY21."</span>, <span class="st">""</span>, GeneID_chr),</span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">chr =</span> <span class="fu">gsub</span>(<span class="st">"*Chr"</span>, <span class="st">""</span>, group)</span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"UAContig"</span>, chr)) <span class="sc">%&gt;%</span></span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(chr) <span class="sc">%&gt;%</span></span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">as.numeric</span>(chr))</span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a><span class="co"># pull(n) |&gt;  sum()</span></span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Stats by chromosome in TMTproC</span></span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>prot.per.chr.df <span class="ot">&lt;-</span></span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>  ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(protein_id_transcript) <span class="sc">%&gt;%</span></span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">protein_id_transcript =</span> <span class="fu">gsub</span>(<span class="st">".v.*"</span>, <span class="st">""</span>, protein_id_transcript), <span class="co"># Extract everything behind ".v."</span></span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">chme =</span> <span class="fu">gsub</span>(<span class="st">"*KY21."</span>, <span class="st">""</span>, protein_id_transcript),</span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, chme),</span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">chr =</span> <span class="fu">gsub</span>(<span class="st">"*Chr"</span>, <span class="st">""</span>, group)</span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"UAContig"</span>, group)) <span class="sc">%&gt;%</span></span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span></span>
<span id="cb38-34"><a href="#cb38-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tally</span>(<span class="at">sort =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb38-35"><a href="#cb38-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">as.numeric</span>(chr))</span>
<span id="cb38-36"><a href="#cb38-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-37"><a href="#cb38-37" aria-hidden="true" tabindex="-1"></a>chr_freq_df <span class="ot">&lt;-</span></span>
<span id="cb38-38"><a href="#cb38-38" aria-hidden="true" tabindex="-1"></a>  prot.per.chr.df <span class="sc">%&gt;%</span></span>
<span id="cb38-39"><a href="#cb38-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(genesTot_chr, <span class="at">by =</span> <span class="st">"chr"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-40"><a href="#cb38-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span></span>
<span id="cb38-41"><a href="#cb38-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb38-42"><a href="#cb38-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">percent =</span> (n.x <span class="sc">/</span> n.y), <span class="co"># getting frequency</span></span>
<span id="cb38-43"><a href="#cb38-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">percent_long =</span> <span class="fu">paste0</span>(<span class="fu">round</span>(percent <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">0</span>), <span class="st">" %"</span>)</span>
<span id="cb38-44"><a href="#cb38-44" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb38-45"><a href="#cb38-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-46"><a href="#cb38-46" aria-hidden="true" tabindex="-1"></a>colourCount <span class="ot">&lt;-</span> <span class="dv">14</span></span>
<span id="cb38-47"><a href="#cb38-47" aria-hidden="true" tabindex="-1"></a>getPalette <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">14</span>, <span class="st">"Accent"</span>))</span>
<span id="cb38-48"><a href="#cb38-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-49"><a href="#cb38-49" aria-hidden="true" tabindex="-1"></a>plot_prot_per_chr <span class="ot">&lt;-</span> chr_freq_df <span class="sc">%&gt;%</span></span>
<span id="cb38-50"><a href="#cb38-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">chr =</span> <span class="fu">factor</span>(chr)) <span class="sc">%&gt;%</span></span>
<span id="cb38-51"><a href="#cb38-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(chr), <span class="at">y =</span> percent, <span class="at">fill =</span> <span class="fu">getPalette</span>(colourCount))) <span class="sc">+</span></span>
<span id="cb38-52"><a href="#cb38-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb38-53"><a href="#cb38-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb38-54"><a href="#cb38-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">percent_format</span>(),<span class="at">position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb38-55"><a href="#cb38-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Accent"</span>))(colourCount)) <span class="sc">+</span></span>
<span id="cb38-56"><a href="#cb38-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb38-57"><a href="#cb38-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Coverage [%]"</span>, <span class="at">x =</span> <span class="st">"Chromosome"</span>, </span>
<span id="cb38-58"><a href="#cb38-58" aria-hidden="true" tabindex="-1"></a>       <span class="co"># title = "Protein distribution per chromosome"</span></span>
<span id="cb38-59"><a href="#cb38-59" aria-hidden="true" tabindex="-1"></a>       ) <span class="sc">+</span></span>
<span id="cb38-60"><a href="#cb38-60" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_text(aes(label = percent_long), color="black") +</span></span>
<span id="cb38-61"><a href="#cb38-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb38-62"><a href="#cb38-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">14</span>,<span class="dv">1</span>))</span>
<span id="cb38-63"><a href="#cb38-63" aria-hidden="true" tabindex="-1"></a>plot_prot_per_chr</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/chromosome%20distribution-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cairo_pdf</span>(<span class="st">"prot_per_chr.pdf"</span>, <span class="at">width=</span><span class="fl">8.5</span>, <span class="at">height=</span><span class="fl">5.6</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot_prot_per_chr)</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="distribution-over-development" class="level4" data-number="1.6.3">
<h4 data-number="1.6.3" class="anchored" data-anchor-id="distribution-over-development"><span class="header-section-number">1.6.3</span> Distribution over development</h4>
</section>
<section id="protein-distribution-by-selecting-10-quantile-cutoff-x-protein" class="level4" data-number="1.6.4">
<h4 data-number="1.6.4" class="anchored" data-anchor-id="protein-distribution-by-selecting-10-quantile-cutoff-x-protein"><span class="header-section-number">1.6.4</span> Protein distribution by selecting &gt;10% quantile cutoff x protein</h4>
<p>same plot as above but selecting 10% quantile by stage not by protein row ##### 10% max per protein</p>
<ul>
<li>Switching to stacked bar plot to highlight differences in enhanced - shared - specific protein distribution over development.</li>
</ul>
<p>Genes are considered: * <code>specific</code>, if are only detected in a particular stage * <code>enhanced</code> if their abundance is at least 3x higher in a particular stage as compared to the average levels in all other timepoint * <code>core set</code> are detected in all stages * <code>shared</code> detected in all stages but did not show enhanced expression in any stage</p>
<ul>
<li>Four Groups of Elevated Proteins Tissue-enriched: at least five-fold higher than any other tissue; group-enriched: at least five-fold higher in a group of 2 to 7 tissues; enhanced: five-fold higher than the average across all other tissues; tissue-specific: for all the proteins expressed in only one organ.</li>
</ul>
</section>
<section id="high-abundance-proteins" class="level4" data-number="1.6.5">
<h4 data-number="1.6.5" class="anchored" data-anchor-id="high-abundance-proteins"><span class="header-section-number">1.6.5</span> High-Abundance Proteins</h4>
<p>We ranked the averaged protein intensities in each tissue and selected those with an abundance which is higher than the 75% quartile plus 1.5 times the interquartile range as high-abundance proteins (HAPs). About 100 HAPs were detected for each tissue type. For the same proteins in different biological replicates of one organ, mean of MS signals is calculated as the intensity of the protein.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Quantile and 10%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>prot.matrix <span class="ot">&lt;-</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  ciona_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>, <span class="dv">6</span><span class="sc">:</span><span class="dv">13</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"protein_id_transcript"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="co"># ordering each stage by rank from lowest value to highest</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a><span class="co"># rank.by.row_protein.db. &lt;-</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   as.data.frame(rowRanks(protein.db, ties.method = c("max"), useNames = T)) %&gt;%</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   rename_all(.funs = ~paste0(., "_rank")) %&gt;%</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   rownames_to_column(var="protein_id_transcript")</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>quantiles_prot.matrix <span class="ot">&lt;-</span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="fu">rowQuantiles</span>(prot.matrix, <span class="at">probs =</span> <span class="fu">c</span>(.<span class="dv">10</span>, <span class="dv">1</span>), <span class="at">useNames =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"protein_id_transcript"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="fu">ends_with</span>(<span class="st">"%"</span>), <span class="at">.fn =</span> <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_quantile"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">10%max.row.value</span><span class="st">`</span> <span class="ot">=</span> (<span class="st">`</span><span class="at">100%_quantile</span><span class="st">`</span> <span class="sc">*</span> <span class="dv">10</span>) <span class="sc">/</span> <span class="dv">100</span>)</span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>ciona_quantProt.df <span class="ot">&lt;-</span> ciona_prot.df <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(quantiles_prot.matrix, <span class="at">by =</span> <span class="st">"protein_id_transcript"</span>)</span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a>quantile_prot.df <span class="ot">&lt;-</span> ciona_quantProt.df <span class="sc">%&gt;%</span></span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(protein_id_transcript) <span class="sc">%&gt;%</span></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">protein_id_transcript =</span> protein_id_transcript,</span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">across</span>(unfE<span class="sc">:</span>larva, <span class="sc">~</span> <span class="fu">sum</span>(. <span class="sc">&gt;</span> <span class="st">`</span><span class="at">10%_quantile</span><span class="st">`</span>))</span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a>quantile_offProt <span class="ot">&lt;-</span> quantile_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise_all</span>(<span class="sc">~</span> <span class="fu">sum</span>(. <span class="sc">==</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">state =</span> <span class="st">"off"</span>)</span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a>quantile_onProt <span class="ot">&lt;-</span> quantile_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise_all</span>(<span class="sc">~</span> <span class="fu">sum</span>(. <span class="sc">==</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">state =</span> <span class="st">"on"</span>)</span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a>onOffProt_10quantile.db <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(quantile_offProt, quantile_onProt) <span class="sc">%&gt;%</span></span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(<span class="dv">10</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">9</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">protein_id_transcript =</span> <span class="st">"quantile10%"</span>)</span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a>max10_prot.df <span class="ot">&lt;-</span> ciona_quantProt.df <span class="sc">%&gt;%</span></span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(protein_id_transcript) <span class="sc">%&gt;%</span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">protein_id_transcript =</span> protein_id_transcript,</span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">across</span>(unfE<span class="sc">:</span>larva, <span class="sc">~</span> <span class="fu">sum</span>(. <span class="sc">&gt;</span> <span class="st">`</span><span class="at">10%max.row.value</span><span class="st">`</span>))</span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a>max10_offProt <span class="ot">&lt;-</span> max10_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb40-50"><a href="#cb40-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise_all</span>(<span class="sc">~</span> <span class="fu">sum</span>(. <span class="sc">==</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-51"><a href="#cb40-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">state =</span> <span class="st">"off"</span>)</span>
<span id="cb40-52"><a href="#cb40-52" aria-hidden="true" tabindex="-1"></a>max10_onProt <span class="ot">&lt;-</span> max10_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb40-53"><a href="#cb40-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise_all</span>(<span class="sc">~</span> <span class="fu">sum</span>(. <span class="sc">==</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-54"><a href="#cb40-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">state =</span> <span class="st">"on"</span>)</span>
<span id="cb40-55"><a href="#cb40-55" aria-hidden="true" tabindex="-1"></a>onOffProt_10max.db <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(max10_offProt, max10_onProt) <span class="sc">%&gt;%</span></span>
<span id="cb40-56"><a href="#cb40-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(<span class="dv">10</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">9</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-57"><a href="#cb40-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">protein_id_transcript =</span> <span class="st">"max10%"</span>)</span>
<span id="cb40-58"><a href="#cb40-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-59"><a href="#cb40-59" aria-hidden="true" tabindex="-1"></a><span class="co"># complete table</span></span>
<span id="cb40-60"><a href="#cb40-60" aria-hidden="true" tabindex="-1"></a>onOff_prot.db <span class="ot">&lt;-</span> onOffProt_10quantile.db <span class="sc">%&gt;%</span></span>
<span id="cb40-61"><a href="#cb40-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(onOffProt_10max.db) <span class="sc">%&gt;%</span></span>
<span id="cb40-62"><a href="#cb40-62" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">type =</span> protein_id_transcript)</span>
<span id="cb40-63"><a href="#cb40-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-64"><a href="#cb40-64" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(KY.TMTproC.df,file = "protein-quantile-rank-on.off.KY21.txt", sep='\t', quote = F, col.names=T)</span></span>
<span id="cb40-65"><a href="#cb40-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-66"><a href="#cb40-66" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb40-67"><a href="#cb40-67" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the number of proteins that are stage-specific and core</span></span>
<span id="cb40-68"><a href="#cb40-68" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb40-69"><a href="#cb40-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-70"><a href="#cb40-70" aria-hidden="true" tabindex="-1"></a><span class="co"># 10% quantile is too restrictive</span></span>
<span id="cb40-71"><a href="#cb40-71" aria-hidden="true" tabindex="-1"></a>stage_specific_quantile <span class="ot">&lt;-</span></span>
<span id="cb40-72"><a href="#cb40-72" aria-hidden="true" tabindex="-1"></a>  quantile_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb40-73"><a href="#cb40-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sum_count =</span> <span class="fu">rowSums</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(., <span class="sc">-</span>protein_id_transcript))) <span class="sc">%&gt;%</span></span>
<span id="cb40-74"><a href="#cb40-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sum_count <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-75"><a href="#cb40-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tally</span>() <span class="co"># 0</span></span>
<span id="cb40-76"><a href="#cb40-76" aria-hidden="true" tabindex="-1"></a>stage_all_quantile <span class="ot">&lt;-</span></span>
<span id="cb40-77"><a href="#cb40-77" aria-hidden="true" tabindex="-1"></a>  quantile_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb40-78"><a href="#cb40-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sum_count =</span> <span class="fu">rowSums</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(., <span class="sc">-</span>protein_id_transcript))) <span class="sc">%&gt;%</span></span>
<span id="cb40-79"><a href="#cb40-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sum_count <span class="sc">==</span> <span class="dv">7</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-80"><a href="#cb40-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tally</span>() <span class="co"># 7095</span></span>
<span id="cb40-81"><a href="#cb40-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-82"><a href="#cb40-82" aria-hidden="true" tabindex="-1"></a><span class="co"># 10% max expression value is good</span></span>
<span id="cb40-83"><a href="#cb40-83" aria-hidden="true" tabindex="-1"></a>stage_specific_perc <span class="ot">&lt;-</span></span>
<span id="cb40-84"><a href="#cb40-84" aria-hidden="true" tabindex="-1"></a>  max10_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb40-85"><a href="#cb40-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sum_count =</span> <span class="fu">rowSums</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(., <span class="sc">-</span>protein_id_transcript))) <span class="sc">%&gt;%</span></span>
<span id="cb40-86"><a href="#cb40-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sum_count <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-87"><a href="#cb40-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tally</span>() <span class="co"># 31</span></span>
<span id="cb40-88"><a href="#cb40-88" aria-hidden="true" tabindex="-1"></a>stage_all_perc <span class="ot">&lt;-</span></span>
<span id="cb40-89"><a href="#cb40-89" aria-hidden="true" tabindex="-1"></a>  max10_prot.df <span class="sc">%&gt;%</span></span>
<span id="cb40-90"><a href="#cb40-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sum_count =</span> <span class="fu">rowSums</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(., <span class="sc">-</span>protein_id_transcript))) <span class="sc">%&gt;%</span></span>
<span id="cb40-91"><a href="#cb40-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sum_count <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-92"><a href="#cb40-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tally</span>() <span class="co"># 6103 core protein</span></span>
<span id="cb40-93"><a href="#cb40-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-94"><a href="#cb40-94" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb40-95"><a href="#cb40-95" aria-hidden="true" tabindex="-1"></a><span class="co"># plotting the info together</span></span>
<span id="cb40-96"><a href="#cb40-96" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb40-97"><a href="#cb40-97" aria-hidden="true" tabindex="-1"></a>onOff_prot_pivot.db <span class="ot">&lt;-</span></span>
<span id="cb40-98"><a href="#cb40-98" aria-hidden="true" tabindex="-1"></a>  onOff_prot.db <span class="sc">%&gt;%</span></span>
<span id="cb40-99"><a href="#cb40-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">9</span>, <span class="at">names_to =</span> <span class="st">"stage"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-100"><a href="#cb40-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> ciona_stages))</span>
<span id="cb40-101"><a href="#cb40-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-102"><a href="#cb40-102" aria-hidden="true" tabindex="-1"></a>onOff_prot_pivot.db <span class="sc">%&gt;%</span></span>
<span id="cb40-103"><a href="#cb40-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_inorder</span>(stage), <span class="at">y =</span> counts, <span class="at">fill =</span> state)) <span class="sc">+</span></span>
<span id="cb40-104"><a href="#cb40-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"dodge"</span>, <span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb40-105"><a href="#cb40-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Protein distribution over development"</span>, <span class="at">x =</span> <span class="st">"Stages"</span>, <span class="at">y =</span> <span class="st">"Protein count"</span>) <span class="sc">+</span></span>
<span id="cb40-106"><a href="#cb40-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(type <span class="sc">~</span> state, <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">switch =</span> <span class="st">"y"</span>, ) <span class="sc">+</span></span>
<span id="cb40-107"><a href="#cb40-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">legend =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/on-off%20genes%20(1)-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>plot_a <span class="ot">&lt;-</span> onOff_prot_pivot.db <span class="sc">%&gt;%</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(state <span class="sc">==</span> <span class="st">"on"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_inorder</span>(stage), <span class="at">y =</span> counts, <span class="at">fill =</span> stage)) <span class="sc">+</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> ciona_stages_palette, <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Counts by state"</span>, <span class="at">x =</span> <span class="st">"Stage"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">legend =</span> <span class="st">"none"</span>)</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>plot_b1 <span class="ot">&lt;-</span> onOff_prot_pivot.db <span class="sc">%&gt;%</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(state <span class="sc">==</span> <span class="st">"on"</span> <span class="sc">&amp;</span> type <span class="sc">==</span> <span class="st">"quantile10%"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_inorder</span>(stage), <span class="at">y =</span> counts, <span class="at">fill =</span> type)) <span class="sc">+</span></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#DEA0FD"</span>),</span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Quantile 10%"</span>),</span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Type"</span>, <span class="at">ncol =</span> <span class="dv">1</span>)</span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Quantile 10%"</span>, <span class="at">x =</span> <span class="st">"Stage"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">legend =</span> <span class="st">"none"</span>)</span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a>plot_b2 <span class="ot">&lt;-</span> onOff_prot_pivot.db <span class="sc">%&gt;%</span></span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(state <span class="sc">==</span> <span class="st">"on"</span> <span class="sc">&amp;</span> type <span class="sc">==</span> <span class="st">"max10%"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_inorder</span>(stage), <span class="at">y =</span> counts, <span class="at">fill =</span> type)) <span class="sc">+</span></span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">6103</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#1CBE4F"</span>),</span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Max 10%"</span>),</span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Type"</span>, <span class="at">ncol =</span> <span class="dv">1</span>)</span>
<span id="cb41-30"><a href="#cb41-30" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-31"><a href="#cb41-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Max 10%"</span>, <span class="at">x =</span> <span class="st">"Stage"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb41-32"><a href="#cb41-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">legend =</span> <span class="st">"none"</span>)</span>
<span id="cb41-33"><a href="#cb41-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-34"><a href="#cb41-34" aria-hidden="true" tabindex="-1"></a>plot_b <span class="ot">&lt;-</span></span>
<span id="cb41-35"><a href="#cb41-35" aria-hidden="true" tabindex="-1"></a>  onOff_prot_pivot.db <span class="sc">%&gt;%</span></span>
<span id="cb41-36"><a href="#cb41-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(state <span class="sc">==</span> <span class="st">"on"</span> <span class="sc">&amp;</span> (type <span class="sc">==</span> <span class="st">"quantile10%"</span> <span class="sc">|</span> type <span class="sc">==</span> <span class="st">"max10%"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb41-37"><a href="#cb41-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_inorder</span>(stage), <span class="at">y =</span> counts, <span class="at">fill =</span> type)) <span class="sc">+</span></span>
<span id="cb41-38"><a href="#cb41-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb41-39"><a href="#cb41-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb41-40"><a href="#cb41-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#1CBE4F"</span>, <span class="st">"#DEA0FD"</span>),</span>
<span id="cb41-41"><a href="#cb41-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Max 10%"</span>,<span class="st">"Quantile 10%"</span>),</span>
<span id="cb41-42"><a href="#cb41-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Type"</span>, <span class="at">ncol =</span> <span class="dv">1</span>)</span>
<span id="cb41-43"><a href="#cb41-43" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-44"><a href="#cb41-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">6103</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb41-45"><a href="#cb41-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">legend =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb41-46"><a href="#cb41-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Comparison of Quantile 10% and Max 10%"</span>)</span>
<span id="cb41-47"><a href="#cb41-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-48"><a href="#cb41-48" aria-hidden="true" tabindex="-1"></a>plot_c <span class="ot">&lt;-</span> onOff_prot_pivot.db <span class="sc">%&gt;%</span></span>
<span id="cb41-49"><a href="#cb41-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(state <span class="sc">==</span> <span class="st">"on"</span> <span class="sc">&amp;</span> (type <span class="sc">==</span> <span class="st">"quantile10%"</span> <span class="sc">|</span> type <span class="sc">==</span> <span class="st">"max10%"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb41-50"><a href="#cb41-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_inorder</span>(stage), <span class="at">y =</span> counts)) <span class="sc">+</span></span>
<span id="cb41-51"><a href="#cb41-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb41-52"><a href="#cb41-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> stage, <span class="at">y =</span> counts, <span class="at">color =</span> type), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb41-53"><a href="#cb41-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> stage, <span class="at">y =</span> counts, <span class="at">color =</span> type, <span class="at">group =</span> type), <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb41-54"><a href="#cb41-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb41-55"><a href="#cb41-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#1CBE4F"</span>, <span class="st">"#DEA0FD"</span>),</span>
<span id="cb41-56"><a href="#cb41-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"10% Max expression"</span>, <span class="st">"10% Quantile"</span>),</span>
<span id="cb41-57"><a href="#cb41-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Type"</span>, <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb41-58"><a href="#cb41-58" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-59"><a href="#cb41-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">legend =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb41-60"><a href="#cb41-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Protein distribution"</span>, <span class="at">x =</span> <span class="st">"Stage"</span>, <span class="at">y =</span> <span class="st">"Count"</span>)</span>
<span id="cb41-61"><a href="#cb41-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-62"><a href="#cb41-62" aria-hidden="true" tabindex="-1"></a>plot_c <span class="sc">+</span> plot_a <span class="sc">+</span> (plot_b1 <span class="sc">/</span> plot_b2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/on-off%20genes%20(1)-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>plot_b <span class="sc">+</span> plot_c <span class="sc">+</span> plot_a</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="/Users/mariossi/Library/CloudStorage/Dropbox-Princeton/Papers/Proteomics_ciona/c.figures_analysis/01.Ciona_prot_QC/on-off%20genes%20(1)-3.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>plot_onOff <span class="ot">&lt;-</span> </span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  onOff_prot_pivot.db <span class="sc">%&gt;%</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(state <span class="sc">==</span> <span class="st">"on"</span> <span class="sc">&amp;</span> type <span class="sc">==</span> <span class="st">"max10%"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_rev</span>(<span class="fu">fct_inorder</span>(stage)), <span class="at">y =</span> counts, <span class="at">fill =</span> name)) <span class="sc">+</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">6103</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">x =</span> <span class="st">"Stage"</span>, <span class="at">y =</span> <span class="st">"Protein"</span>) <span class="sc">+</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">2000</span>, <span class="dv">4000</span>, <span class="dv">6000</span>, <span class="dv">7000</span>),<span class="at">position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> ciona_stages_palette, <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>(<span class="at">base_family =</span> <span class="st">"Avenir"</span>, <span class="at">legend =</span> <span class="st">"none"</span>,<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() </span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cairo_pdf</span>(<span class="st">"plot_onOff.pdf"</span>, <span class="at">width=</span><span class="fl">8.5</span>, <span class="at">height=</span><span class="fl">5.6</span>)</span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot_onOff)</span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



<script src="01.Ciona_prot_QC_files/libs/quarto-html/zenscroll-min.js"></script>
</body></html>